---
title: "eQTL_Stroke_MatrixEQTL_mRNA_275_052819"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Here we want to analyze eQTL on the 275 samples for only mRNA coding genes, with this model
#y=α+β1(SNP)+β2(sex)+β3(age)+β4(race)+β5(hypertension)+β6(diabetes)+β7(hypercholesterolemia)+β8(diagnosis subtype)+β9(diagnosis)+β10(SNP ✕ diagnosis)+error term.

```{r}
#GE_mRNA_19309 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/GE_mRNA_19309.txt", row.names=1) #193009 genes
#gene_expression_275_SST.RMA.GENE.FULL.Group.SST.RMA.GENE.FULL...Group.1 <- read.delim("C:/eQTL/GENE/Picked_Files_HTA2_275/gene_expression_275_SST-RMA-GENE-FULL-Group.SST-RMA-GENE-FULL - Group 1.TXT", row.names=1)
gene_expression_mRNA_19309 <- gene_expression_275_SST.RMA.GENE.FULL.Group.SST.RMA.GENE.FULL...Group.1[rownames(GE_mRNA_19309),]
gene_expression_mRNA_small_19309 <- gene_expression_mRNA_19309[,c(1:275)]
gene_expression_mRNA_19309 <- gene_expression_mRNA_19309[,c(1:275)]
#save
write.table(gene_expression_mRNA_small_19309, file = "../../Input_Output/Samples_275/GE_mRNA_19309.txt",col.names =T, sep = "\t") 
```


```{r}
SNP_calls_275_table <- read.delim("C:/eQTL/SNP/SNP_calls_275_eQTL/275_sample/SNP_calls_275_table.txt", row.names=1)#628679 20, including MAF, CR, HW and SNP id for each SNPs
#Filter based on other references, using this paper "PancanQTL: systematic identification of cis-eQTLs and trans-eQTLs in 33 cancer types"
SNP_calls_275_table_filter <- SNP_calls_275_table[SNP_calls_275_table$MinorAlleleFrequency > 0.05 & SNP_calls_275_table$CR > 95 & SNP_calls_275_table$H.W.p.Value > 0.0000005,] #I think we need to filter based on H.W (sample size), #274816  20
table(SNP_calls_275_table$MinorAlleleFrequency > 0.05)
# FALSE   TRUE 
#337917 290762 
table(SNP_calls_275_table$CR > 95)
#FALSE   TRUE 
# 15814 612865 
table(SNP_calls_275_table$H.W.p.Value > 0.0000005)
#FALSE   TRUE 
#6421 622258 
SNP_calls_275_table_filter_id <- row.names(SNP_calls_275_table_filter) 
write.table(SNP_calls_275_table_filter_id, file = "../../Input_Output/Samples_275/SNP_calls_275_table_filter_id", col.names = T, sep = "\t")
#subset snps id after filtering from SNP.txt file

#save the MAF > 0.01, CR>95 and H.W.p.Value > 0.0000005
SNP_calls_275_table_filter <- SNP_calls_275_table[SNP_calls_275_table$MinorAlleleFrequency > 0.01 & SNP_calls_275_table$CR > 95 & SNP_calls_275_table$H.W.p.Value > 0.0000005,] 
write.table(SNP_calls_275_table_filter_id, file = "../../Input_Output/Samples_275/SNP_calls_275_table_filter_id_MAFmore0.01", col.names = T, sep = "\t")
SNP <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Samples_275/SNP.txt", row.names=1)
SNP_after_filter <- SNP[SNP_calls_275_table_filter_id,] #274816 275
write.table(SNP_after_filter, file = "../../Input_Output/Samples_275/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", col.names = T, sep = "\t")

write.table(SNP_after_filter, file = "../../Input_Output/Samples_275/SNP_CRmore95_MAFmore0.01_HWEmore0.0000005.txt", col.names = T, sep = "\t") #334727  275 (including 59911 snp more than MAF more than 0.05)
```

```{r}
base.dir <- ("C:/eQTL/R_eQTL_Stroke/")
SNP_file_name = paste(base.dir, "/Input_Output/Samples_275/SNP.txt", sep="") #all snps id (628679)
SNP_file_name = paste(base.dir, "/Input_Output/Samples_275/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", sep="")
#check for SNP with MAF more than 0.01
SNP_file_name = paste(base.dir, "/Input_Output/Samples_275/SNP_CRmore95_MAFmore0.01_HWEmore0.0000005.txt", sep="")
covariates_file_name = paste(base.dir, "/Input_Output/Samples_275/Covariate_Final_NO_Risk_Factors.txt", sep="") #need sample as colnames
gene_location_file_name = paste(base.dir, "/Input_Output/Gene_Location.txt", sep="")
SNP_Location <-  paste(base.dir, "/Input_Output/SNP_Location.txt", sep="") #include all SNPs location
expression_file_name = paste(base.dir, "/Input_Output/Samples_275/GE_mRNA_19309.txt", sep="") 
```

```{r}
output_file_name_cis = tempfile()
output_file_name_tra = tempfile()
errorCovariance = numeric()
```

```{r}
## Load genotype data
snps = SlicedData$new() #SlicedData object with genotype information.
snps$fileDelimiter = "\t"      # the TAB character
snps$fileOmitCharacters = "NA" # denote missing values;
snps$fileSkipRows = 1          # one row of column labels
snps$fileSkipColumns = 1       # one column of row labels
snps$fileSliceSize = 2000    # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name) 
# Rows read: 274816 done. This is correct, we have   274816  SNPs id
```

```{r}
# #checking only filtering snps out based on MAF not other parameters
# maf.list = vector('list', length(snps))
# for(sl in 1:length(snps)) {
#   slice = snps[[sl]];
#   maf.list[[sl]] = rowMeans(slice,na.rm=TRUE)/2;
#   maf.list[[sl]] = pmin(maf.list[[sl]],1-maf.list[[sl]]);
# }
# maf = unlist(maf.list)
# 
# ## Look at the distribution of MAF
# # hist(maf[maf<0.1],seq(0,0.1,length.out=100))
# 
# cat('SNPs before filtering:',nrow(snps))
# # snps$RowReorderSimple(maf>0.05);
# snps$RowReorder(maf>0.05);
# cat('SNPs after filtering:',nrow(snps))
# #SNPs after filtering: 354013
```

```{r}
# Load gene expression data
gene = SlicedData$new(); #SlicedData object with gene expression information. Must have columns matching those of snps, create gene including row and column
gene$fileDelimiter = "\t"      # the TAB characte
gene$fileOmitCharacters = "NA" # denote missing values;
gene$fileSkipRows = 1          # one row of column labels
gene$fileSkipColumns = 1       # one column of row labels
gene$fileSliceSize = 2000      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name) 
#Rows read: 19309 done. This is correct, we have 19309 genes
```

```{r}
# Load covariates
cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t"      # the TAB character
cvrt$fileOmitCharacters = "NA" # denote missing values;
cvrt$fileSkipRows = 1         # one row of column labels
cvrt$fileSkipColumns = 1      # one column of row labels
if(length(covariates_file_name)>0) {
cvrt$LoadFile(covariates_file_name)
}
#Rows read: 8 done. correct
```

```{r}
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE) #dim  65988   4
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE) #dim 628677 3
```

```{r}
useModel =  modelLINEAR_CROSS

cisDist = 1e6
eQTL = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = output_file_name_tra,
pvOutputThreshold = 1e-10,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, 
output_file_name.cis = output_file_name_cis,
pvOutputThreshold.cis = 1e-5,
snpspos = snpspos, 
genepos = genepos,
cisDist = cisDist,
pvalue.hist = "qqplot",
min.pv.by.genesnp = FALSE,
noFDRsaveMemory = FALSE)

unlink(output_file_name_tra);
unlink(output_file_name_cis); 
```

```{r}
#Standard errors for the estimates

eQTL$cis$eqtls$beta_se <- eQTL$cis$eqtls$beta / eQTL$cis$eqtls$statistic
#save
write.csv(eQTL$cis$eqtls$beta_se,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6_se.csv")

eQTL$trans$eqtls$beta_se <- eQTL$trans$eqtls$beta / eQTL$trans$eqtls$statistic
#save
write.csv(eQTL$trans$eqtls$beta_se,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_dist_1e6_se.csv")

write.csv(eQTL$cis$eqtls,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6_withse.csv") # copy and poaste MAF in this file

eQTL$trans$eqtls$beta_se <- eQTL$trans$eqtls$beta / eQTL$trans$eqtls$statistic

write.csv(eQTL$trans$eqtls,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_dist_1e6_withse.csv")  # copy and poaste MAF in this file

```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_trans_1e6.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_cis_dist_1e6.csv")
#load("../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #43

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#43
write.csv(eQTL_cis_dist_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_cis_dist_1e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis//eQTL_trans_1e6.csv")
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #66

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#29
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_trans_1e6_merge.csv") 
```

#Try with different covariate matrix

```{r}
#try with y=α+β1(SNP)+β2(sex)+β3(age)+β4(race)+β5(diagnosis subtype)+β6(diagnosis)+β7(SNP ✕ diagnosis)+error term., with Covariate_Final_NO_PC_NO_Risk_Factors.txt
covariates_file_name = paste(base.dir, "/Input_Output/Samples_275/Covariate_Final_NO_PC_NO_Risk_Factors.txt", sep="") #5rows
```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis/eQTL_trans_1e6.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis/eQTL_cis_dist_1e6.csv")
#load("../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #49

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#43
write.csv(eQTL_cis_dist_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis/eQTL_cis_dist_1e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis/eQTL_trans_1e6.csv")
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #66

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#29
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis/eQTL_trans_1e6_merge.csv") 
```

```{r}
#calculate first 20 PCs for 275 samples with Plink to add in modelrownames(plink_v2)<- plink_v2[,1]

plink <- read.table("C:/eQTL/Plink_PCA/new_experiment_275/plink.eigenvec", quote="\"", comment.char="") #included 20 first PCs for 275 samples
plink_v2 <-plink[,-1]
rownames(plink_v2)<- plink_v2[,1] #convert sample name as a rownames
rownames(plink_v2) %>% length() #275
colnames(plink_v2)
plink_v3 <- plink_v2[,-1]
names(plink_v3)[1] <- "pc1"
names(plink_v3)[2] <- "pc2"
names(plink_v3)[3] <- "pc3"
names(plink_v3)[4] <- "pc4"
#save
write.table(plink_v3,"../../../Plink_PCA/experiment_351/plink_pc1_pc2_pc3_pc4.txt", col.names = T, sep = "\t")
#check samples name to be consistant with others list. convert these three names UCDSS.767_2 UCDSS.838_2 UCDSSTourettes to UCDSS.767.2 UCDSS.838.2 UCDSSTourettes.TSFS623CT

#to see eigenvalue
plink_eigen_value <- read.table("C:/eQTL/Plink_PCA/plink.eigenval", quote="\"", comment.char="")

#          V1
# 1  10.93030
# 2   6.16319
# 3   1.90339
# 4   1.64103
# 5   1.44360
# 6   1.40266
# 7   1.39656
# 8   1.35917
# 9   1.34444
# 10  1.33994
# 11  1.33436
# 12  1.33163
# 13  1.32852
# 14  1.31879
# 15  1.31518
# 16  1.31290
# 17  1.31203
# 18  1.30170
# 19  1.29851
#20  1.29374

sum(plink_eigen_value) #42.07164 three pcs represent 43.10% and four pcs 49.05421%

```

```{r}
#try with y=α+β1(SNP)+β2(sex)+β3(age)+β4(race)+β5(diagnosis subtype)+β6(diagnosis)+β7(pc1)+β8(pc2)_β9(pc3)+β10(pc4)+β11(SNP ✕ diagnosis)+error term., with Covariate_Final_NO_Risk_Factors.txt
covariates_file_name = paste(base.dir, "/Input_Output/Samples_275/Covariate_Final_NO_Risk_Factors.txt", sep="") #9 rows

```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_1e6.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6.csv")
#load("../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #43

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#43
write.csv(eQTL_cis_dist_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_1e6.csv")
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #66

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#29
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_1e6_merge.csv") 
```

```{r}
#saving the eQTL result for snps with MAF more than 0.01 (354013 snps)
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/MAF_more_0.01/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/MAF_more_0.01/eQTL_trans_1e6.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/MAF_more_0.01/eQTL_cis_dist_1e6.csv") #169 7
#load("../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #169 12

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#43
write.csv(eQTL_cis_dist_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/MAF_more_0.01/eQTL_cis_dist_1e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/MAF_more_0.01/eQTL_trans_1e6.csv")
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #10368

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#10368
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/MAF_more_0.01/eQTL_trans_1e6_merge.csv") 
```


```{r}
#check for coding genes for modelLINEAR
#y=α+β1(SNP)+β2(sex)+β3(age)+β4(race)+β5(pc1)+β6(pc2)+β7(pc3)+β8(pc4)+β9(diagnosis subtype)+β10(diagnosis)+error term.
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_1e6.csv")
```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6.csv") #5134    7
#load("../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE)

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#5134   19
write.csv(eQTL_cis_dist_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_1e6.csv") #117   7
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) 

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#117 19
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_1e6_merge.csv") 
```


```{r}
#start to making figure for eQTL on 275 samples

eQTL_cis_dist_1e6_merge <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_cis_dist_1e6_merge.csv", row.names=1) #38 19
eQTL_trans_1e6_merge <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/eQTL_trans_1e6_merge.csv", row.names=1) #70 19
```


```{r}
#updated merge_v3 for this analysis, additional varibales should be added as a row in covariate matrix

#SNP_CRmore95_MAFmore0.05_HWEmore0.0000005 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Samples_275/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", row.names=1) # 274816    275
SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t <- t(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005)
#dim(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#Covariate_Final <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Samples_275/Covariate_Final.txt", row.names=1) # 12 275
Covariate_Final_t <-t(Covariate_Final)
# GE_mRNA_19309 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Samples_275/GE_mRNA_19309.txt", row.names=1) #19309   275
GE_mRNA_19309_t <-t(GE_mRNA_19309)
#class(GE_mRNA_19309_t)
merge <- cbind(GE_mRNA_19309_t,SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#dim(merge) # 275 294125 (274816+19309)
merge_v2 <- cbind(merge,Covariate_Final_t)
#dim(merge_v2) #final dataset  275 294137
merge_v3 <- as.data.frame(merge_v2)
save(merge_v3, file = "../../Input_Output/RData/merge_final_275.RData")
```


```{r}
# 1 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
# AX-29907155	TC11001143.hg.1	1	5.781018764	2.10E-08	0.087925371	0.31293627	NRGN 	chr11	124609829	124617869	Coding	rs56348411	11	124974588	T	C	C	T

 #  0   1   2 
 # 13  98 164
x <- merge_v3$`AX-29907155`
#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 53	0 UCDSS.331
#8	2	5	57	0	0	0	-0.0332142	-0.0293081	-0.0111536	0.00536828	4
# 75	0 UCDSS.440
#8	1	5	51	0	0	1	-0.0320873	-0.0303784	-0.0162713	4.88E-05	4
# 77	0 UCDSS.452
#1	2	5	68	0	0	1	-0.0225055	-0.0219632	-0.0150669	0.00105871	1
# 84	0 UCDSS.473
#3	2	5	61	1	1	1	-0.0319983	-0.0308743	-0.0157616	0.00265121	1
# 122	0 UCDSS.571
#2	1	3	69	1	0	0	0.18122	-0.034571	-0.0165888	-0.102848	1
# 127	0 UCDSS.580
#1	1	5	62	1	1	1	-0.0316899	-0.0268608	-0.0236804	0.00213223	1
# 148	0 UCDSS.671
#4	2	5	61	1	0	1	-0.0330384	-0.0294158	-0.0147448	0.00302026	1
# 156	0 UCDSS.686
#3	1	5	47	1	0	1	-0.0319757	-0.0226402	-0.00941068	0.000403153	1
# 226	0 UCDSS.867
#6	2	5	69	1	1	0	-0.0318381	-0.0282704	-0.017322	0.0031588	1
# 250	0 UCSFCJ.092
#8	2	5	77	0	0	0	-0.0315985	-0.0288294	-0.0267084	-0.0157903	4
# 251	0 UCSFCJ.097
#4	1	5	60	0	0	0	-0.0320547	-0.0295992	-0.0127332	0.00131964	1
# 254	0 UCSFCJ.109
#4	1	2	53	1	1	1	-0.0323482	-0.0282733	-0.0178268	0.00268296	1
# 266	0 UofASS.003
#8	2	5	80	1	1	0	-0.0317688	-0.0271927	-0.0170305	-0.000305488	4
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC11001143.hg.1~`AX-29907155`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-29907155`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC11001143.hg.1 ~ `AX-29907155` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-29907155` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.6486 -0.4995 -0.1055  0.3224  3.1148 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.752717   0.441752  17.550  < 2e-16 ***
# `AX-29907155`           -0.800645   0.149249  -5.364 1.78e-07 ***
# age                      0.004614   0.003743   1.233 0.218790    
# race                     0.101824   0.055835   1.824 0.069340 .  
# gender                   0.217821   0.098208   2.218 0.027413 *  
# diagnosis               -0.464784   0.127724  -3.639 0.000329 ***
# diagnosissubtype         0.034893   0.054039   0.646 0.519035    
# pc1                      2.405275   0.988794   2.433 0.015661 *  
# pc2                      2.135775   0.879160   2.429 0.015797 *  
# pc3                     -1.905648   0.989943  -1.925 0.055307 .  
# pc4                     -1.120431   0.783472  -1.430 0.153880    
# `AX-29907155`:diagnosis  0.312936   0.054132   5.781 2.10e-08 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.771 on 263 degrees of freedom
# Multiple R-squared:  0.1771,	Adjusted R-squared:  0.1427 
# F-statistic: 5.145 on 11 and 263 DF,  p-value: 2.593e-07

min(merge_v3$TC11001143.hg.1) #6.298015
max(merge_v3$TC11001143.hg.1) #11.25922
```


```{r}
p1 <-ggplot(aes(y = merge_v3$TC11001143.hg.1, x = as.factor(merge_v3$`AX-29907155`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="NRGN expression", x = "rs56348411 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("NRGN")," " , "expression")))+expand_limits(y=c(6,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-29907155`), y=merge_v3$TC11001143.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="NRGN expression", x = "rs56348411 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression")))+ expand_limits(y=c(6,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-29907155`, y=merge_v3$TC11001143.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(6,12))

p2 <-p1 + labs(y="NRGN expression", x = "rs56348411 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(merge_v3,varname = "TC11001143.hg.1", groupnames =c("diagnosis","`AX-29907155`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-29907155` <- as.factor(df$`AX-29907155`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-29907155`), y=df$TC11001143.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC11001143.hg.1-sd, ymax=TC11001143.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(6,10))

p2 <-p1 + labs(y="NRGN expression", x = "rs56348411 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-29907155`), y=df$TC11001143.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC11001143.hg.1-sd, ymax=TC11001143.hg.1+sd),width=.2)+ expand_limits(y=c(6,10))

p2 <-p1 + labs(y="NRGN expression", x = "rs56348411 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL1_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
# 2 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#AX-14758778	TC04001305.hg.1	2	-5.598499386	5.43E-08	0.113844611	-0.52763463	CXCL10 	chr4	76942269	76944689	Coding	rs78046578	4	76836362	T	C	T	C


 #  0   1   2 NA
 # 243  28   2 2
x <- merge_v3$`AX-14758778`
#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 150	2 
#UCDSS.674 3	1	6	43	1	0	1	-0.0193745	0.00799197	0.0527321	0.00054771	1
# 173	2
#UCDSS.738 7	2	5	66	0	0	0	-0.0316139	-0.0284217	-0.0124966	0.00314247	4
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001305.hg.1~`AX-14758778`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-14758778`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001305.hg.1 ~ `AX-14758778` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-14758778` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.9885 -0.3702 -0.1193  0.2861  6.7600 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.251971   0.392980  10.820  < 2e-16 ***
# `AX-14758778`            2.050178   0.277475   7.389 2.01e-12 ***
# age                     -0.001336   0.003749  -0.356    0.722    
# race                     0.054615   0.056071   0.974    0.331    
# gender                   0.115867   0.099970   1.159    0.248    
# diagnosis                0.097579   0.095799   1.019    0.309    
# diagnosissubtype        -0.045107   0.054187  -0.832    0.406    
# pc1                     -1.712761   1.040768  -1.646    0.101    
# pc2                      1.226902   0.874752   1.403    0.162    
# pc3                     -1.322489   0.987002  -1.340    0.181    
# pc4                     -0.056667   0.784740  -0.072    0.942    
# `AX-14758778`:diagnosis -0.526087   0.094420  -5.572 6.27e-08 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.7706 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.1902,	Adjusted R-squared:  0.156 
# F-statistic: 5.571 on 11 and 261 DF,  p-value: 5.134e-08

min(merge_v3$TC04001305.hg.1) #3.784149
max(merge_v3$TC04001305.hg.1) #14.38171
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-14758778'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC04001305.hg.1, x = as.factor(x$`AX-14758778`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("CXCL10")," " , "expression")))+expand_limits(y=c(2,14))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-14758778`), y=x$TC04001305.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression")))+ expand_limits(y=c(2,14))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-14758778`, y=x$TC04001305.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(2,14))

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC04001305.hg.1", groupnames =c("diagnosis","`AX-14758778`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-14758778` <- as.factor(df$`AX-14758778`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-14758778`), y=df$TC04001305.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001305.hg.1-sd, ymax=TC04001305.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(2,14))

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-14758778`), y=df$TC04001305.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001305.hg.1-sd, ymax=TC04001305.hg.1+sd),width=.2)+ expand_limits(y=c(2,14))

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL2_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
# 3 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#AX-11701696	TC18000998.hg.1	3	-5.445956017	1.18E-07	0.165240725	-0.162540872	SMAD4 	chr18	48556583	48611415	Coding	rs975903	18	49306115	T	G	T	G

x<-merge_v3$`AX-11701696`
#  0   1   2 
# 234  39   2 

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 73	2 
#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
# 106	2
#UCDSS.530	3	2	5	85	1	1	1	-0.0257809	-0.0248302	-0.0143615	0.00177168	1
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC18000998.hg.1~`AX-11701696`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11701696`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC18000998.hg.1 ~ `AX-11701696` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11701696` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -2.80325 -0.10918  0.00993  0.13122  0.67186 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             10.6439834  0.1411431  75.413  < 2e-16 ***
# `AX-11701696`            0.1927835  0.0829036   2.325   0.0208 *  
# age                     -0.0004491  0.0013399  -0.335   0.7378    
# race                    -0.0315666  0.0199636  -1.581   0.1150    
# gender                  -0.0167967  0.0353592  -0.475   0.6352    
# diagnosis                0.0165986  0.0347874   0.477   0.6337    
# diagnosissubtype         0.0023839  0.0195214   0.122   0.9029    
# pc1                     -0.0036093  0.3550715  -0.010   0.9919    
# pc2                     -0.1455456  0.3117163  -0.467   0.6409    
# pc3                      0.6146027  0.3570749   1.721   0.0864 .  
# pc4                     -0.0667078  0.2814494  -0.237   0.8128    
# `AX-11701696`:diagnosis -0.1625409  0.0298462  -5.446 1.18e-07 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2772 on 263 degrees of freedom
# Multiple R-squared:  0.1702,	Adjusted R-squared:  0.1355 
# F-statistic: 4.904 on 11 and 263 DF,  p-value: 6.559e-07

min(merge_v3$TC18000998.hg.1) #6.752242
max(merge_v3$TC18000998.hg.1) #10.89588
```


```{r}
p1 <-ggplot(aes(y = merge_v3$TC18000998.hg.1, x = as.factor(merge_v3$`AX-11701696`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SMAD4 expression", x = "rs975903 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TG", "2"="GG"))+scale_y_continuous(expression(paste( italic("SMAD4")," " , "expression")))+expand_limits(y=c(6,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11701696`), y=merge_v3$TC18000998.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="SMAD4 expression", x = "rs975903 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ expand_limits(y=c(6,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11701696`, y=merge_v3$TC18000998.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(6,12))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs975903 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(merge_v3,varname = "TC18000998.hg.1", groupnames =c("diagnosis","`AX-11701696`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11701696` <- as.factor(df$`AX-11701696`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11701696`), y=df$TC18000998.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(6,12))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs975903 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11701696`), y=df$TC18000998.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd),width=.2)+ expand_limits(y=c(6,11))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs975903 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL3_scattererrorbar.jpg",width = 6, height = 4)
```



```{r}
# 4 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-34646211	TC04000145.hg.1	4	5.342514642	1.99E-07	0.208149576	0.493781483	CD38 	chr4	15779898	15851069	Coding	rs62299879	4	16448976	T	C	T	C


x<-merge_v3$`AX-34646211`
AX-39724859
#  0   1   2 NA
# 235  26   1 13

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 80	2 
#UCDSS.463	8	1	5	70	1	0	0	-0.023	-0.0217369	-0.010371	0.0020478	4 
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04000145.hg.1~`AX-34646211`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-34646211`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04000145.hg.1 ~ `AX-34646211` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-34646211` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -2.1257 -0.4381 -0.0218  0.4097  2.5265 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              8.271235   0.368436  22.450  < 2e-16 ***
# `AX-34646211`           -0.893780   0.263315  -3.394   0.0008 ***
# age                     -0.004204   0.003514  -1.196   0.2327    
# race                    -0.020003   0.051765  -0.386   0.6995    
# gender                   0.015381   0.092583   0.166   0.8682    
# diagnosis               -0.198643   0.089336  -2.224   0.0271 *  
# diagnosissubtype         0.126035   0.050982   2.472   0.0141 *  
# pc1                      0.560427   0.926936   0.605   0.5460    
# pc2                      1.586422   0.806463   1.967   0.0503 .  
# pc3                     -0.698381   0.908392  -0.769   0.4427    
# pc4                      0.749692   0.720415   1.041   0.2990    
# `AX-34646211`:diagnosis  0.492849   0.092500   5.328 2.21e-07 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.7051 on 250 degrees of freedom
#   (13 observations deleted due to missingness)
# Multiple R-squared:  0.1651,	Adjusted R-squared:  0.1284 
# F-statistic: 4.495 on 11 and 250 DF,  p-value: 3.351e-06

min(merge_v3$TC04000145.hg.1) #6.018493
max(merge_v3$TC04000145.hg.1) #12.74917
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-34646211'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC04000145.hg.1, x = as.factor(x$`AX-34646211`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CD38 expression", x = "rs62299879 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("CD38")," " , "expression")))+expand_limits(y=c(6,14))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-34646211`), y=x$TC04000145.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="CD38 expression", x = "rs62299879 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("CD38")," " , "expression")))+ expand_limits(y=c(6,14))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-34646211`, y=x$TC04000145.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(6,14))

p2 <-p1 + labs(y="CD38 expression", x = "rs62299879 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CD38")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC04000145.hg.1", groupnames =c("diagnosis","`AX-34646211`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-34646211` <- as.factor(df$`AX-34646211`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-34646211`), y=df$TC04000145.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04000145.hg.1-sd, ymax=TC04000145.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(6,14))

p2 <-p1 + labs(y="CD38 expression", x = "rs62299879 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CD38")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-34646211`), y=df$TC04000145.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04000145.hg.1-sd, ymax=TC04000145.hg.1+sd),width=.2)+ expand_limits(y=c(6,14))

p2 <-p1 + labs(y="CD38 expression", x = "rs62299879 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CD38")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL4_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
# 5 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-39724859	TC01000514.hg.1	5	5.121264987	5.87E-07	0.492350151	0.176159349	ZNF684 	chr1	40997233	41013841	Coding	rs11809423	1	41976529	T	C	C	T


x<-merge_v3$`AX-39724859`
#  0   1   2
#   4  25 246 

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 0 
#UCDSS.552	8	2	3	62	1	1	0	0.161348	-0.0344033	-0.00370047	-0.0234773	4
#UCDSS.674 3	1	6	43	1	0	1	-0.0193745	0.00799197	0.0527321	0.00054771	1
#UCDSS.773 8	1	3	63	1	0	0	0.171622	-0.0371677	-0.00054286	-0.0270997	4
#UCDSS.835 	3	1	5	46	1	0	1	-0.0321729	-0.0277639	-0.0161718	-0.00456315	1

```

```{r}
#START
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01000514.hg.1~`AX-39724859`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-39724859`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01000514.hg.1 ~ `AX-39724859` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-39724859` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.1439 -0.1468 -0.0182  0.1375  3.3450 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.492303   0.255457  29.329  < 2e-16 ***
# `AX-39724859`           -0.725165   0.100378  -7.224 5.44e-12 ***
# age                     -0.002402   0.001480  -1.623   0.1059    
# race                    -0.013893   0.022189  -0.626   0.5318    
# gender                   0.026030   0.039115   0.665   0.5063    
# diagnosis               -0.345154   0.073545  -4.693 4.33e-06 ***
# diagnosissubtype        -0.006939   0.021468  -0.323   0.7468    
# pc1                     -0.929365   0.415350  -2.238   0.0261 *  
# pc2                      0.025832   0.344212   0.075   0.9402    
# pc3                     -0.191782   0.398934  -0.481   0.6311    
# pc4                     -0.124401   0.313070  -0.397   0.6914    
# `AX-39724859`:diagnosis  0.176159   0.034398   5.121 5.87e-07 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3062 on 263 degrees of freedom
# Multiple R-squared:  0.1891,	Adjusted R-squared:  0.1551 
# F-statistic: 5.574 on 11 and 263 DF,  p-value: 4.998e-08

min(merge_v3$TC01000514.hg.1) #5.342232
max(merge_v3$TC01000514.hg.1) #10.31879
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-39724859'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC01000514.hg.1, x = as.factor(merge_v3$`AX-39724859`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZNF684 expression", x = "rs11809423 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("ZNF684")," " , "expression")))+expand_limits(y=c(5,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-39724859`), y=merge_v3$TC01000514.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ZNF684 expression", x = "rs11809423 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ expand_limits(y=c(5,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-39724859`, y=merge_v3$TC01000514.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5,12))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs11809423 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC01000514.hg.1", groupnames =c("diagnosis","`AX-39724859`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-39724859` <- as.factor(df$`AX-39724859`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-39724859`), y=df$TC01000514.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5,12))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs11809423 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-39724859`), y=df$TC01000514.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd),width=.2)+ expand_limits(y=c(6,14))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs11809423 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL5_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#No figure for this pair
# 6 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-30137055	TC01003363.hg.1	11	4.835439681	2.26E-06	0.862335513	0.070146062	FCRL4 	chr1	157543539	157567870	Coding	---	1	157485429	C	G	C	G


x<-merge_v3$`AX-30137055`
#  0   1   2
#  245  29   1 

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 2 
#UCDSS.357	3	1	3	70	1	0	0	0.162101	-0.0330967	-0.0111911	-0.0358062	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01003363.hg.1~`AX-30137055`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-30137055`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01003363.hg.1 ~ `AX-30137055` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-30137055` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.27704 -0.07278 -0.02420  0.03978  0.49008 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.6211696  0.0583988  62.008  < 2e-16 ***
# `AX-30137055`           -0.1366081  0.0397796  -3.434 0.000691 ***
# age                      0.0008415  0.0005586   1.506 0.133200    
# race                     0.0306329  0.0082906   3.695 0.000268 ***
# gender                   0.0020762  0.0147772   0.140 0.888373    
# diagnosis               -0.0059056  0.0141441  -0.418 0.676627    
# diagnosissubtype         0.0007624  0.0080754   0.094 0.924852    
# pc1                      0.2302749  0.1473734   1.563 0.119366    
# pc2                      0.1514395  0.1303755   1.162 0.246466    
# pc3                     -0.3200932  0.1464175  -2.186 0.029685 *  
# pc4                     -0.0216259  0.1182675  -0.183 0.855052    
# `AX-30137055`:diagnosis  0.0701461  0.0145067   4.835 2.26e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1152 on 263 degrees of freedom
# Multiple R-squared:  0.1422,	Adjusted R-squared:  0.1063 
# F-statistic: 3.964 on 11 and 263 DF,  p-value: 2.391e-05

min(merge_v3$TC01003363.hg.1) #3.582366
max(merge_v3$TC01003363.hg.1) #4.479516
```

```{r}
# 7 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-32754641	TC19000642.hg.1	9	4.849467074	2.12E-06	0.862335513	0.02443318	PPP1R37 	chr19	45596218	45650543	Coding	rs75608718	19	46515961	T	C	T	C



x<-merge_v3$`AX-32754641`
# 0   1   2 
#151  98  26


#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis


```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC19000642.hg.1~`AX-32754641`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-32754641`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC19000642.hg.1 ~ `AX-32754641` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-32754641` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.20195 -0.05209 -0.00404  0.04622  0.33564 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.7848159  0.0402810 118.786  < 2e-16 ***
# `AX-32754641`           -0.0437750  0.0130929  -3.343 0.000948 ***
# age                     -0.0002544  0.0003751  -0.678 0.498203    
# race                     0.0089118  0.0055920   1.594 0.112207    
# gender                   0.0043613  0.0099131   0.440 0.660334    
# diagnosis               -0.0176943  0.0101033  -1.751 0.081054 .  
# diagnosissubtype         0.0030115  0.0054720   0.550 0.582548    
# pc1                      0.0571803  0.0998271   0.573 0.567274    
# pc2                     -0.0347475  0.0892230  -0.389 0.697261    
# pc3                     -0.0713867  0.0986472  -0.724 0.469920    
# pc4                     -0.0579773  0.0788928  -0.735 0.463063    
# `AX-32754641`:diagnosis  0.0244332  0.0050383   4.849 2.12e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.07768 on 263 degrees of freedom
# Multiple R-squared:  0.1144,	Adjusted R-squared:  0.07738 
# F-statistic: 3.089 on 11 and 263 DF,  p-value: 0.0006347

min(merge_v3$TC19000642.hg.1) #4.542393
max(merge_v3$TC19000642.hg.1) #5.142106
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-39724859'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC19000642.hg.1, x = as.factor(merge_v3$`AX-32754641`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PPP1R37 expression", x = "rs75608718 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("PPP1R37")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-32754641`), y=merge_v3$TC19000642.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="PPP1R37 expression", x = "rs75608718 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("PPP1R37")," " , "expression")))+ expand_limits(y=c(4,5))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-32754641`, y=merge_v3$TC19000642.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4.5,5.5))

p2 <-p1 + labs(y="PPP1R37 expression", x = "rs75608718 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PPP1R37")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC19000642.hg.1", groupnames =c("diagnosis","`AX-32754641`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-32754641` <- as.factor(df$`AX-32754641`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-32754641`), y=df$TC19000642.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC19000642.hg.1-sd, ymax=TC19000642.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,5))

p2 <-p1 + labs(y="PPP1R37 expression", x = "rs75608718 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PPP1R37")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-32754641`), y=df$TC19000642.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC19000642.hg.1-sd, ymax=TC19000642.hg.1+sd),width=.2)+ expand_limits(y=c(4.5,5))

p2 <-p1 + labs(y="PPP1R37 expression", x = "rs75608718 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PPP1R37")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL7_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
# 19 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-11190683	TC19002694.hg.1	19	4.695530851	4.29E-06	0.945503491	0.146163515	IGFLR1 	chr19	36230151	36233520	Coding	rs12110	19	35660508	A	G	G	A


x<-merge_v3$`AX-11190683`
# 0   1   2 
#198  72   5


#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis
# UCDSS.241	3	1	7	77	0	0	0	-0.0242015	-0.0141061	0.0189637	-0.00609588	1
# UCDSS.251	1	2	5	79	1	0	0	-0.032348	-0.0233221	-0.0195041	-0.00537922	1
# UCDSS.386	4	2	5	82	1	0	1	-0.0327022	-0.0298716	-0.0168187	0.000230636	1
# UCDSS.739	8	1	5	48	1	0	0	-0.0327341	-0.0304762	-0.0108814	0.00640304	4
# UCDSS.826	3	2	5	50	1	1	0	-0.0346912	-0.0276195	-0.00740617	-0.00138206	1


```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC19002694.hg.1~`AX-11190683`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11190683`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC19002694.hg.1 ~ `AX-11190683` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11190683` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.04206 -0.20990  0.05225  0.22218  0.95455 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.447789   0.192217  33.544  < 2e-16 ***
# `AX-11190683`           -0.367535   0.086524  -4.248 3.00e-05 ***
# age                     -0.001118   0.001839  -0.608    0.544    
# race                    -0.043229   0.027312  -1.583    0.115    
# gender                   0.040551   0.048316   0.839    0.402    
# diagnosis               -0.060305   0.047242  -1.277    0.203    
# diagnosissubtype         0.009940   0.026560   0.374    0.709    
# pc1                     -0.270933   0.487454  -0.556    0.579    
# pc2                     -0.350484   0.425809  -0.823    0.411    
# pc3                      0.446019   0.482274   0.925    0.356    
# pc4                     -0.220892   0.384870  -0.574    0.566    
# `AX-11190683`:diagnosis  0.146164   0.031128   4.696 4.29e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3792 on 263 degrees of freedom
# Multiple R-squared:  0.09431,	Adjusted R-squared:  0.05643 
# F-statistic:  2.49 on 11 and 263 DF,  p-value: 0.005477

min(merge_v3$TC19002694.hg.1) #5.115492
max(merge_v3$TC19002694.hg.1) #7.087374
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-39724859'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC19002694.hg.1, x = as.factor(merge_v3$`AX-11190683`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="IGFLR1 expression", x = "rs12110 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("IGFLR1")," " , "expression")))+expand_limits(y=c(5,7))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11190683`), y=merge_v3$TC19002694.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="IGFLR1 expression", x = "rs12110 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("IGFLR1")," " , "expression")))+ expand_limits(y=c(5,7))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11190683`, y=merge_v3$TC19002694.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5,7))

p2 <-p1 + labs(y="IGFLR1 expression", x = "rs12110 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("IGFLR1")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC19002694.hg.1", groupnames =c("diagnosis","`AX-11190683`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11190683` <- as.factor(df$`AX-11190683`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11190683`), y=df$TC19002694.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC19002694.hg.1-sd, ymax=TC19002694.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5,7))

p2 <-p1 + labs(y="IGFLR1 expression", x = "rs12110 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("IGFLR1")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11190683`), y=df$TC19002694.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC19002694.hg.1-sd, ymax=TC19002694.hg.1+sd),width=.2)+ expand_limits(y=c(5,7))

p2 <-p1 + labs(y="IGFLR1 expression", x = "rs12110 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("IGFLR1")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL19_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
# 22 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-42740371	TC0X000794.hg.1	22	-4.655676036	5.13E-06	0.976902943	-0.111721399	GTPBP6 	chrX	220013	230887	Coding	rs2738360	X	302966	A	G	G	A


x<-merge_v3$`AX-42740371`
# 0   1   2 
#2  27   246


#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis
#UCDSS.624	8	1	5	46	0	0	1	-0.028159	-0.0294293	-0.0219248	0.00980667	4
#UCDSS.678	8	1	5	47	0	0	1	-0.028244	-0.0291217	-0.0224977	0.00993752	4

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC0X000794.hg.1~`AX-42740371`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-42740371`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC0X000794.hg.1 ~ `AX-42740371` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-42740371` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.49083 -0.12520 -0.01314  0.12324  0.76564 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.8323987  0.1688329  28.622  < 2e-16 ***
# `AX-42740371`            0.2515192  0.0752658   3.342 0.000953 ***
# age                     -0.0017710  0.0009505  -1.863 0.063532 .  
# race                     0.0114887  0.0141261   0.813 0.416784    
# gender                   0.0110024  0.0250479   0.439 0.660839    
# diagnosis                0.2225047  0.0541554   4.109 5.32e-05 ***
# diagnosissubtype        -0.0046892  0.0139749  -0.336 0.737482    
# pc1                      0.2695578  0.2529181   1.066 0.287496    
# pc2                     -0.2346367  0.2214216  -1.060 0.290261    
# pc3                     -0.2026883  0.2497887  -0.811 0.417847    
# pc4                     -0.3428628  0.1992664  -1.721 0.086495 .  
# `AX-42740371`:diagnosis -0.1117214  0.0239968  -4.656 5.13e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1963 on 263 degrees of freedom
# Multiple R-squared:  0.1252,	Adjusted R-squared:  0.08861 
# F-statistic: 3.422 on 11 and 263 DF,  p-value: 0.0001846

min(merge_v3$TC0X000794.hg.1) #4.775781
max(merge_v3$TC0X000794.hg.1) #6.415451
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-39724859'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC0X000794.hg.1, x = as.factor(merge_v3$`AX-42740371`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="GTPBP6 expression", x = "rs2738360 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("GTPBP6")," " , "expression")))+expand_limits(y=c(4,7))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-42740371`), y=merge_v3$TC0X000794.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="GTPBP6 expression", x = "rs2738360 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("GTPBP6")," " , "expression")))+ expand_limits(y=c(4,7))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-42740371`, y=merge_v3$TC0X000794.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,7))

p2 <-p1 + labs(y="GTPBP6 expression", x = "rs2738360 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("GTPBP6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC0X000794.hg.1", groupnames =c("diagnosis","`AX-42740371`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-42740371` <- as.factor(df$`AX-42740371`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-42740371`), y=df$TC0X000794.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC0X000794.hg.1-sd, ymax=TC0X000794.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,7))

p2 <-p1 + labs(y="GTPBP6 expression", x = "rs2738360 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("GTPBP6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-42740371`), y=df$TC0X000794.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC0X000794.hg.1-sd, ymax=TC0X000794.hg.1+sd),width=.2)+ expand_limits(y=c(5,7))

p2 <-p1 + labs(y="GTPBP6 expression", x = "rs2738360 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("GTPBP6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL22_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
# 30 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-30509799	TC12002016.hg.1	29	-4.566833373	7.61E-06	1	-0.10479546	WSB2 	chr12	118470492	118499979	Coding	rs11068369	12	117586896	T	G	T	G


x<-merge_v3$`AX-30509799`
# 0   1   2 
#2  27   246

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC12002016.hg.1~`AX-30509799`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-30509799`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC12002016.hg.1 ~ `AX-30509799` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-30509799` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.98632 -0.24423 -0.00835  0.19803  1.71677 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.866264   0.207009  38.000  < 2e-16 ***
# `AX-30509799`            0.273523   0.067856   4.031 7.28e-05 ***
# age                      0.002010   0.001931   1.041   0.2988    
# race                    -0.003564   0.028720  -0.124   0.9013    
# gender                   0.018414   0.051665   0.356   0.7218    
# diagnosis                0.093057   0.053869   1.727   0.0853 .  
# diagnosissubtype        -0.014825   0.027980  -0.530   0.5967    
# pc1                     -0.138810   0.522333  -0.266   0.7906    
# pc2                     -0.061906   0.450021  -0.138   0.8907    
# pc3                      0.564860   0.517422   1.092   0.2760    
# pc4                     -0.500439   0.405462  -1.234   0.2182    
# `AX-30509799`:diagnosis -0.104795   0.022947  -4.567 7.61e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3989 on 263 degrees of freedom
# Multiple R-squared:  0.09491,	Adjusted R-squared:  0.05705 
# F-statistic: 2.507 on 11 and 263 DF,  p-value: 0.005154

min(merge_v3$TC12002016.hg.1) #7.120933
max(merge_v3$TC12002016.hg.1) #10.08423
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-39724859'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC12002016.hg.1, x = as.factor(merge_v3$`AX-30509799`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="WSB2 expression", x = "rs11068369 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TG", "2"="GG"))+scale_y_continuous(expression(paste( italic("WSB2")," " , "expression")))+expand_limits(y=c(7,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-30509799`), y=merge_v3$TC12002016.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="WSB2 expression", x = "rs11068369 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))+scale_y_continuous(expression(paste(italic("WSB2")," " , "expression")))+ expand_limits(y=c(7,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-30509799`, y=merge_v3$TC12002016.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(7,10))

p2 <-p1 + labs(y="WSB2 expression", x = "rs11068369 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("WSB2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC12002016.hg.1", groupnames =c("diagnosis","`AX-30509799`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-30509799` <- as.factor(df$`AX-30509799`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-30509799`), y=df$TC12002016.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC12002016.hg.1-sd, ymax=TC12002016.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(7,10))

p2 <-p1 + labs(y="WSB2 expression", x = "rs11068369 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("WSB2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-30509799`), y=df$TC12002016.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC12002016.hg.1-sd, ymax=TC12002016.hg.1+sd),width=.2)+ expand_limits(y=c(7,10))

p2 <-p1 + labs(y="WSB2 expression", x = "rs11068369 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("WSB2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL30_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
# 33 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
# AX-38492059	TC09000417.hg.1	35	4.527901449	9.04E-06	1	0.128617875	SECISBP2 	chr9	91933412	91974561	Coding	rs76287022	9	91627100	T	C	C	T


x<-merge_v3$`AX-38492059`
# 0   1   2 NA
#  2  32 240  

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis
#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.602	3	1	3	51	1	1	0	0.119142	-0.0404059	-0.0206839	-0.0306819	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC09000417.hg.1~`AX-38492059`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38492059`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC09000417.hg.1 ~ `AX-38492059` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38492059` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.37628 -0.12872  0.00321  0.13634  0.89404 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             10.020309   0.191745  52.259  < 2e-16 ***
# `AX-38492059`           -0.227233   0.081655  -2.783  0.00578 ** 
# age                     -0.001244   0.001210  -1.028  0.30504    
# race                     0.022798   0.018148   1.256  0.21015    
# gender                   0.039981   0.031980   1.250  0.21234    
# diagnosis               -0.238269   0.061338  -3.885  0.00013 ***
# diagnosissubtype        -0.003147   0.017617  -0.179  0.85836    
# pc1                      0.436661   0.324590   1.345  0.17970    
# pc2                      0.198100   0.285169   0.695  0.48787    
# pc3                     -0.222997   0.319786  -0.697  0.48621    
# pc4                     -0.043163   0.255743  -0.169  0.86611    
# `AX-38492059`:diagnosis  0.128631   0.028462   4.519 9.39e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2508 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.1016,	Adjusted R-squared:  0.06392 
# F-statistic: 2.695 on 11 and 262 DF,  p-value: 0.002657

min(merge_v3$TC09000417.hg.1) #7.800171
max(merge_v3$TC09000417.hg.1) #10.56844
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-38492059'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC09000417.hg.1, x = as.factor(x$`AX-38492059`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SECISBP2 expression", x = "rs76287022 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("SECISBP2")," " , "expression")))+expand_limits(y=c(7,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38492059`), y=x$TC09000417.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="SECISBP2 expression", x = "rs76287022 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("SECISBP2")," " , "expression")))+ expand_limits(y=c(7,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38492059`, y=x$TC09000417.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(7,10))

p2 <-p1 + labs(y="SECISBP2 expression", x = "rs76287022 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SECISBP2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC09000417.hg.1", groupnames =c("diagnosis","`AX-38492059`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38492059` <- as.factor(df$`AX-38492059`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38492059`), y=df$TC09000417.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC09000417.hg.1-sd, ymax=TC09000417.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(7,10))

p2 <-p1 + labs(y="SECISBP2 expression", x = "rs76287022 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SECISBP2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38492059`), y=df$TC09000417.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC09000417.hg.1-sd, ymax=TC09000417.hg.1+sd),width=.2)+ expand_limits(y=c(7,10))

p2 <-p1 + labs(y="SECISBP2 expression", x = "rs76287022 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SECISBP2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL33_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
# 37 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#AX-42518859	TC09000560.hg.1	36	4.523302406	9.22E-06	1	0.196688919	UGCG 	chr9	114659046	114697649	Coding	rs1326895	9	113678096	T	C	C	T


x<-merge_v3$`AX-42518859`
# 0   1   2 
#  5  62 208   

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis
#UCDSS.518	8	2	6	48	1	0	0	-0.0104733	0.0373449	0.100049	-0.000678858	4
#UCDSS.743	8	1	2	74	1	0	0	0.0120923	0.174417	-0.0507614	0.000402591	4
#UCSFCJ.031	3	1	2	68	0	0	1	0.0118368	0.176311	-0.0726385	-0.00112801	1
#UCSFCJ.128	8	1	6	91	1	0	1	-0.01013	0.00301463	0.0420307	-0.00780142	4
#UofASS.002	4	1	5	68	1	0	1	-0.0317127	-0.0293228	-0.0169263	0.000271324	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC09000560.hg.1~`AX-42518859`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-42518859`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC09000560.hg.1 ~ `AX-42518859` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-42518859` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.01599 -0.25008 -0.06967  0.13301  2.84331 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              9.7284127  0.3337929  29.145  < 2e-16 ***
# `AX-42518859`           -0.6691268  0.1314339  -5.091 6.79e-07 ***
# age                     -0.0007065  0.0024551  -0.288 0.773761    
# race                    -0.0334438  0.0366260  -0.913 0.362017    
# gender                   0.0988719  0.0654356   1.511 0.131994    
# diagnosis               -0.3545137  0.0988811  -3.585 0.000401 ***
# diagnosissubtype        -0.0356339  0.0356215  -1.000 0.318061    
# pc1                     -0.7886407  0.6556553  -1.203 0.230125    
# pc2                     -0.7410748  0.5851291  -1.267 0.206449    
# pc3                      0.1010564  0.6502506   0.155 0.876616    
# pc4                     -0.1107753  0.5161671  -0.215 0.830237    
# `AX-42518859`:diagnosis  0.1966889  0.0434835   4.523 9.22e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.5083 on 263 degrees of freedom
# Multiple R-squared:  0.1399,	Adjusted R-squared:  0.1039 
# F-statistic: 3.889 on 11 and 263 DF,  p-value: 3.171e-05

min(merge_v3$TC09000560.hg.1) #7.226324
max(merge_v3$TC09000560.hg.1) #11.38889
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-38492059'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC09000560.hg.1, x = as.factor(merge_v3$`AX-42518859`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="UGCG expression", x = "rs1326895 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("UGCG")," " , "expression")))+expand_limits(y=c(7,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-42518859`), y=merge_v3$TC09000560.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="UGCG expression", x = "rs1326895 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ expand_limits(y=c(7,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-42518859`, y=merge_v3$TC09000560.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(7,11))

p2 <-p1 + labs(y="UGCG expression", x = "rs1326895 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC09000560.hg.1", groupnames =c("diagnosis","`AX-42518859`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-42518859` <- as.factor(df$`AX-42518859`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-42518859`), y=df$TC09000560.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC09000560.hg.1-sd, ymax=TC09000560.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(7,11))

p2 <-p1 + labs(y="UGCG expression", x = "rs1326895 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-42518859`), y=df$TC09000560.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC09000560.hg.1-sd, ymax=TC09000560.hg.1+sd),width=.2)+ expand_limits(y=c(7,10))

p2 <-p1 + labs(y="UGCG expression", x = "rs1326895 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL37_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
# 21 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
# 21	AX-11608684	TC11001143.hg.1		-4.659384975	5.04E-06	0.976902943	-0.214862121	NRGN 5.04E-06	0.976902943	124609829	124617869	Coding	rs7129315	11	124977280	T	C	T	C


 #  0   1   2 NA
 # 86 136  52 1
x <- merge_v3$`AX-11608684`
#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis
# 4	2
# 10	2
# 13	2
# 18	2
# 53	2
# 55	2
# 62	2
# 63	2
# 75	2
# 77	2
# 80	2
# 84	2
# 93	2
# 108	2
# 109	2
# 110	2
# 113	2
# 116	2
# 117	2
# 122	2
# 127	2
# 145	2
# 146	2
# 148	2
# 149	2
# 150	2
# 153	2
# 155	2
# 156	2
# 158	2
# 165	2
# 171	2
# 172	2
# 176	2
# 179	2
# 184	2
# 193	2
# 211	2
# 218	2
# 226	2
# 228	2
# 230	2
# 239	2
# 250	2
# 251	2
# 254	2
# 260	2
# 266	2
# 267	2
# 268	2
# 269	2
# 275	2
y <- merge_v3$`AX-29907155`== 0 & merge_v3$`AX-11608684` == 2 #both CC genotype and NRGN gene
View(y)
# 53
# TRUE
# 75
# TRUE
# 77
# TRUE
# 84
# TRUE
# 122
# TRUE
# 127
# TRUE
# 148
# TRUE
# 156
# TRUE
# 226
# TRUE
# 250
# TRUE
# 251
# TRUE
# 254
# TRUE
# 266
# TRUE

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC11001143.hg.1~`AX-11608684`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11608684`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC11001143.hg.1 ~ `AX-11608684` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11608684` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.6258 -0.4998 -0.1073  0.3531  3.0649 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.207289   0.418061  14.848  < 2e-16 ***
# `AX-11608684`            0.577648   0.134406   4.298 2.43e-05 ***
# age                      0.003041   0.003839   0.792   0.4290    
# race                     0.079232   0.057349   1.382   0.1683    
# gender                   0.193034   0.100714   1.917   0.0564 .  
# diagnosis                0.172366   0.101220   1.703   0.0898 .  
# diagnosissubtype         0.057786   0.055817   1.035   0.3015    
# pc1                      2.306371   1.015560   2.271   0.0240 *  
# pc2                      1.790532   0.892404   2.006   0.0458 *  
# pc3                     -1.530965   1.010523  -1.515   0.1310    
# pc4                     -0.879045   0.805497  -1.091   0.2761    
# `AX-11608684`:diagnosis -0.214472   0.046178  -4.644 5.40e-06 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.7884 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.1425,	Adjusted R-squared:  0.1065 
# F-statistic: 3.959 on 11 and 262 DF,  p-value: 2.446e-05

min(merge_v3$TC11001143.hg.1) #6.298015
max(merge_v3$TC11001143.hg.1) #11.25922
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-11608684'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC11001143.hg.1, x = as.factor(x$`AX-11608684`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="NRGN expression", x = "rs7129315 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("NRGN")," " , "expression")))+expand_limits(y=c(6,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-11608684`), y=x$TC11001143.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="NRGN expression", x = "rs7129315 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression")))+ expand_limits(y=c(6,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-11608684`, y=x$TC11001143.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(6,12))

p2 <-p1 + labs(y="NRGN expression", x = "rs7129315 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(x,varname = "TC11001143.hg.1", groupnames =c("diagnosis","`AX-11608684`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11608684` <- as.factor(df$`AX-11608684`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11608684`), y=df$TC11001143.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC11001143.hg.1-sd, ymax=TC11001143.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(6,10))

p2 <-p1 + labs(y="NRGN expression", x = "rs7129315 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11608684`), y=df$TC11001143.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC11001143.hg.1-sd, ymax=TC11001143.hg.1+sd),width=.2)+ expand_limits(y=c(6,10))

p2 <-p1 + labs(y="NRGN expression", x = "rs7129315 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("NRGN")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/cis-eQTL21_scattererrorbar.jpg",width = 6, height = 4)
```



```{r}

# merge_v3$`AX-39724859`
# #  0   1   2 
# #  4  25 246
# 
# merge_v3$`AX-30137055`
# #  0   1   2 
# #245  29   1 
# 
# merge_v3$`AX-32754641`
# #  0   1   2 
# #151  98  26 
# 
# merge_v3$`AX-34244929`
# # 0   1   2 
# #  5  83 187
# 
# merge_v3$`AX-36092663`
# # 0   1   2 
# #  1  46 227 
# 
# merge_v3$`AX-42418945`
# # 0   1   2 
# # 29 108 136 
# 
# merge_v3$`AX-12560974`
# #  0   1   2 
# #235  26   1 
# 
# merge_v3$`AX-15683169`
# #  0   1   2 
# # 235  26   1 
# 
# merge_v3$`AX-32776075`
# # 0   1   2 
# #  2  31 241
# 
# merge_v3$`AX-40461039`
# #  0   1   2 
# # 69 130  76 
# 
# merge_v3$`AX-40552141`
# #  0   1   2 
# #132 109  33
# 
# merge_v3$`AX-42558879`
# #  0   1   2 
# #  1  42 232 
# 
# merge_v3$`AX-83130777`
# #  0   1   2 
# #  2  45 228
# 
# merge_v3$`AX-11190683`
# #  0   1   2 
# #198  72   5 
# 
# merge_v3$`AX-32558153`
# #  0   1   2 
# #  1  31 232 
# 
# merge_v3$`AX-11608684`
# # 0   1   2 
# # 86 136  52
# 
# merge_v3$`AX-42740371`
#   0   1   2 
#   2  27 246 
# 
# merge_v3$`AX-14538617`
# 
#   0   1   2 
# 239  35   1 
# 
# merge_v3$`AX-11643502`
#   0   1   2 
# 212  61   1 
# 
# merge_v3$`AX-42740371`
# 0   1   2 
#   2  27 246 
#   
# merge_v3$`AX-11133338`
# 
#   0   1   2 
#  23 109 141 
# 
# merge_v3$`AFFX-SNP-001119`
#   0   1   2 
# 127 121  27 
# 
# merge_v3$`AX-11374277`
#   0   1   2 
#  37  97 141 
# 
# merge_v3$`AX-13098879`
#   0   1   2 
#   2  48 225 
#   
# 
# merge_v3$`AX-14612074`
#   0   1   2 
# 141 112  22
# 
# merge_v3$`AX-30509799`
#   0   1   2 
#  85 131  59 
# 
# merge_v3$`AX-32813941`
#  0   1   2 
#   1  44 230 
# merge_v3$`AX-34997671`
# 1   2 
#  31 244 
# 
# merge_v3$`AX-38492059`
#  0   1   2 
#   2  32 240 
# 
# merge_v3$`AX-40475333`
#  0   1   2 
#   4  54 217 
# merge_v3$`AX-40592209`
#  0   1   2 
#   2  40 233 
# merge_v3$`AX-40901009`
#   0   1   2 
#   1  53 220 
# merge_v3$`AX-42518859`
# 
#   0   1   2 
#   5  62 208
# 
# merge_v3$`AX-83404926`
#  0   1   2 
#  66 125  84 

```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

# 1 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-38686743	TC0X000829.hg.1	1	12.44456982	2.90E-28	1.54E-18	0.349323792	ANOS1 	chrX	8496915	8700227	Coding	rs148791848	X	93386861	T	C	T	C


x<-merge_v3$`AX-38686743`

#  0   1   2  NA
# 257  12   3 3

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 129	2 
#UCDSS.586	8	2	5	67	1	0	1	-0.00434061	0.0623294	0.13238	-0.00514716	4
#162 2
#UCDSS.709	3	1	6	55	1	1	1	-0.00616543	0.0499376	0.114078	0.00200624	1
#206 2
#UCDSS.808	3	1	6	54	1	1	1	-0.00964104	0.0365048	0.106183	-0.00234253	1
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC0X000829.hg.1~`AX-38686743`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38686743`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC0X000829.hg.1 ~ `AX-38686743` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38686743` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.80595 -0.08841 -0.02613  0.07390  0.67158 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.237368   0.083192  50.935  < 2e-16 ***
# `AX-38686743`           -0.403775   0.058718  -6.877 4.54e-11 ***
# age                      0.001555   0.000798   1.949   0.0524 .  
# race                     0.011277   0.011861   0.951   0.3426    
# gender                  -0.130300   0.021126  -6.168 2.63e-09 ***
# diagnosis               -0.016706   0.020146  -0.829   0.4077    
# diagnosissubtype         0.004052   0.011508   0.352   0.7251    
# pc1                     -0.008672   0.211590  -0.041   0.9673    
# pc2                      0.095860   0.186024   0.515   0.6068    
# pc3                      0.133069   0.209444   0.635   0.5258    
# pc4                     -0.043000   0.166367  -0.258   0.7963    
# `AX-38686743`:diagnosis  0.349147   0.028046  12.449  < 2e-16 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1638 on 260 degrees of freedom
#   (3 observations deleted due to missingness)
# Multiple R-squared:  0.4565,	Adjusted R-squared:  0.4335 
# F-statistic: 19.85 on 11 and 260 DF,  p-value: < 2.2e-16

min(merge_v3$TC0X000829.hg.1) #3.764966
max(merge_v3$TC0X000829.hg.1) #6.515363
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-38686743'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC0X000829.hg.1, x = as.factor(x$`AX-38686743`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ANOS1 expression", x = "rs148791848 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("ANOS1")," " , "expression")))+expand_limits(y=c(3,7))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL.jpg",width = 6, height = 4)

```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38686743`), y=x$TC0X000829.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ANOS1 expression", x = "rs148791848 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ expand_limits(y=c(3,7))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38686743`, y=x$TC0X000829.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs148791848 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC0X000829.hg.1", groupnames =c("diagnosis","`AX-38686743`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38686743` <- as.factor(df$`AX-38686743`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38686743`), y=df$TC0X000829.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC0X000829.hg.1-sd, ymax=TC0X000829.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(2,8))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs148791848 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38686743`), y=df$TC0X000829.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC0X000829.hg.1-sd, ymax=TC0X000829.hg.1+sd),width=.2)+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs148791848 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL_scattererrorbar.jpg",width = 6, height = 4)
```



```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

# 3 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-11393384	TC17001827.hg.1	3	-8.607949313	6.91E-16	1.22E-06	-0.395706241	ABCA6 	chr17	67074847	67138015	Coding	rs2464504	4	48232441	T	C	C	T


x<-merge_v3$`AX-11393384`

#  0   1   2
# 250  22   3 

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

# 37	2 
#UCDSS.272	8	2	5	57	1	0	1	-0.0273933	-0.0257722	-0.0115635	0.000843584	4
#54 2
#UCDSS.343	8	2	5	63	0	0	0	-0.0319583	-0.0307143	-0.0203161	0.000106389	4
#252 2
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17001827.hg.1~`AX-11393384`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11393384`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17001827.hg.1 ~ `AX-11393384` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11393384` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.3099 -0.0992 -0.0126  0.0799  4.3581 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.961965   0.185915  21.311  < 2e-16 ***
# `AX-11393384`            1.599420   0.149245  10.717  < 2e-16 ***
# age                      0.002833   0.001771   1.600   0.1109    
# race                     0.021442   0.026406   0.812   0.4175    
# gender                  -0.063740   0.046872  -1.360   0.1750    
# diagnosis                0.081691   0.045274   1.804   0.0723 .  
# diagnosissubtype        -0.040535   0.025726  -1.576   0.1163    
# pc1                      0.360143   0.473693   0.760   0.4478    
# pc2                      0.006198   0.415132   0.015   0.9881    
# pc3                     -0.113489   0.466706  -0.243   0.8081    
# pc4                     -0.055876   0.372220  -0.150   0.8808    
# `AX-11393384`:diagnosis -0.395706   0.045970  -8.608 6.91e-16 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3668 on 263 degrees of freedom
# Multiple R-squared:  0.3248,	Adjusted R-squared:  0.2965 
# F-statistic:  11.5 on 11 and 263 DF,  p-value: < 2.2e-16

min(merge_v3$TC17001827.hg.1) #3.87316
max(merge_v3$TC17001827.hg.1) #10.95106
```


```{r}
p1 <-ggplot(aes(y = merge_v3$TC17001827.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(2,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2.jpg",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/sample.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11393384`), y=merge_v3$TC17001827.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(2,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11393384`, y=merge_v3$TC17001827.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT2L_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC17001827.hg.1", groupnames =c("diagnosis","`AX-11393384`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11393384` <- as.factor(df$`AX-11393384`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11393384`), y=df$TC17001827.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11393384`), y=df$TC17001827.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd),width=.2)+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT2L_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

# 2 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-16372367	TC17001827.hg.1	2	8.990203601	4.97E-17	1.32E-07	0.478932822	ABCA6 	chr17	67074847	67138015	Coding	rs950391	X	86454329	A	G	G	A


x<-merge_v3$`AX-16372367`

#  0   1   2 NA 
# 250  22   3 1

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

#UCDSS.513	8	1	2	71	1	1	1	-0.01459	0.0134688	-0.0145828	-0.00139512	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17001827.hg.1~`AX-16372367`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372367`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17001827.hg.1 ~ `AX-16372367` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372367` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.4519 -0.1189 -0.0174  0.1162  3.8908 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              8.073517   0.378437  21.334  < 2e-16 ***
# `AX-16372367`           -1.895385   0.159992 -11.847  < 2e-16 ***
# age                      0.001478   0.001715   0.862   0.3897    
# race                    -0.018047   0.025902  -0.697   0.4866    
# gender                  -0.093244   0.045489  -2.050   0.0414 *  
# diagnosis               -0.888094   0.111959  -7.932 6.26e-14 ***
# diagnosissubtype        -0.035783   0.024891  -1.438   0.1517    
# pc1                     -0.977714   0.458368  -2.133   0.0339 *  
# pc2                     -0.140586   0.404425  -0.348   0.7284    
# pc3                      0.334883   0.451588   0.742   0.4590    
# pc4                      0.087393   0.364611   0.240   0.8108    
# `AX-16372367`:diagnosis  0.479085   0.053373   8.976  < 2e-16 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3547 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.3709,	Adjusted R-squared:  0.3445 
# F-statistic: 14.05 on 11 and 262 DF,  p-value: < 2.2e-16

min(merge_v3$TC17001827.hg.1) #3.87316
max(merge_v3$TC17001827.hg.1) #10.95106
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17001827.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(2,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2.jpg",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/sample.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372367`), y=x$TC17001827.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(2,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372367`, y=x$TC17001827.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT2L_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC17001827.hg.1", groupnames =c("diagnosis","`AX-16372367`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372367` <- as.factor(df$`AX-16372367`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372367`), y=df$TC17001827.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372367`), y=df$TC17001827.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd),width=.2)+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL2_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT2L_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

# 2 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-11173316	TC17001827.hg.1	4	-8.551433325	1.01E-15	1.34E-06	-0.392840406	ABCA6 	chr17	67074847	67138015	Coding	rs11758921	6	166247384	A	G	A	G



x<-merge_v3$`AX-11173316`

#  0   1   2 
# 244  29   2

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

#UCDSS.791	8	1	5	70	0	0	0	-0.032299	-0.0281154	-0.0130974	0.00054316	4

#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17001827.hg.1~`AX-11173316`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11173316`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17001827.hg.1 ~ `AX-11173316` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11173316` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.2571 -0.0954 -0.0073  0.0909  4.5203 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.986047   0.188494  21.147  < 2e-16 ***
# `AX-11173316`            1.547101   0.149304  10.362  < 2e-16 ***
# age                      0.001341   0.001797   0.746   0.4561    
# race                     0.024388   0.026746   0.912   0.3627    
# gender                  -0.093416   0.047402  -1.971   0.0498 *  
# diagnosis               -0.025353   0.045798  -0.554   0.5803    
# diagnosissubtype         0.027148   0.026489   1.025   0.3064    
# pc1                      0.350637   0.477480   0.734   0.4634    
# pc2                      0.149484   0.418383   0.357   0.7212    
# pc3                     -0.288250   0.473438  -0.609   0.5432    
# pc4                      0.049551   0.376774   0.132   0.8955    
# `AX-11173316`:diagnosis -0.392840   0.045939  -8.551 1.01e-15 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3712 on 263 degrees of freedom
# Multiple R-squared:  0.3085,	Adjusted R-squared:  0.2796 
# F-statistic: 10.67 on 11 and 263 DF,  p-value: 3.323e-16

min(merge_v3$TC17001827.hg.1) #3.87316
max(merge_v3$TC17001827.hg.1) #10.95106
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC17001827.hg.1, x = as.factor(merge_v3$`AX-11173316`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(2,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11173316`), y=merge_v3$TC17001827.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(2,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11173316`, y=merge_v3$TC17001827.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC17001827.hg.1", groupnames =c("diagnosis","`AX-11173316`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11173316` <- as.factor(df$`AX-11173316`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11173316`), y=df$TC17001827.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11173316`), y=df$TC17001827.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd),width=.2)+ expand_limits(y=c(2,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL4_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

# 5 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-11173316	TC01002971.hg.1	5	-8.249400762	7.70E-15	8.17E-06	-0.27328946	SLC16A4 	chr1	110905470	110933704	Coding	rs11758921	6	166247384	A	G	A	G


x<-merge_v3$`AX-11173316`

#  0   1   2 
# 244  29   2

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

#UCDSS.791	8	1	5	70	0	0	0	-0.032299	-0.0281154	-0.0130974	0.00054316	4

#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01002971.hg.1~`AX-11173316`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11173316`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01002971.hg.1 ~ `AX-11173316` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11173316` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.95214 -0.10685 -0.01447  0.10427  2.77880 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              5.1506437  0.1359318  37.891  < 2e-16 ***
# `AX-11173316`            1.1210191  0.1076700  10.412  < 2e-16 ***
# age                      0.0004547  0.0012956   0.351    0.726    
# race                     0.0053880  0.0192875   0.279    0.780    
# gender                  -0.0487341  0.0341836  -1.426    0.155    
# diagnosis               -0.0467624  0.0330272  -1.416    0.158    
# diagnosissubtype         0.0286007  0.0191023   1.497    0.136    
# pc1                     -0.1638224  0.3443331  -0.476    0.635    
# pc2                     -0.0688714  0.3017152  -0.228    0.820    
# pc3                     -0.2016977  0.3414177  -0.591    0.555    
# pc4                     -0.0600998  0.2717093  -0.221    0.825    
# `AX-11173316`:diagnosis -0.2732895  0.0331284  -8.249  7.7e-15 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2677 on 263 degrees of freedom
# Multiple R-squared:  0.3249,	Adjusted R-squared:  0.2967 
# F-statistic: 11.51 on 11 and 263 DF,  p-value: < 2.2e-16

min(merge_v3$TC01002971.hg.1) #4.827518
max(merge_v3$TC01002971.hg.1) #9.654251
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC01002971.hg.1, x = as.factor(merge_v3$`AX-11173316`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SLC16A4 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("SLC16A4")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11173316`), y=merge_v3$TC01002971.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11173316`, y=merge_v3$TC01002971.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC01002971.hg.1", groupnames =c("diagnosis","`AX-11173316`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11173316` <- as.factor(df$`AX-11173316`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11173316`), y=df$TC01002971.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11173316`), y=df$TC01002971.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd),width=.2)+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL5_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#6 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#	AX-14058740	TC15001756.hg.1	6	-8.188402103	1.15E-14	1.02E-05	-0.122452891	AP3B2 	chr15	83328033	83378666	Coding	rs72944885	3	102311450	A	G	G	A



x<-merge_v3$`AX-14058740`

#  0   1   2 NA
# 6  21 247 1

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

#UCDSS.357	3	1	3	70	1	0	0	0.162101	-0.0330967	-0.0111911	-0.0358062	1
#UCDSS.693	8	2	3	48	1	0	0	0.141	-0.0354487	-0.0212148	-0.0503794	4
#UCDSS.708	8	2	3	41	1	1	1	0.161344	-0.0350806	0.0073334	-0.0420488	4
#UCDSS.730	4	1	3	55	1	1	1	0.160227	-0.0337266	0.0066151	-0.0470337	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.017	2	1	3	63	1	0	0	0.163721	-0.0349639	-0.0122536	-0.0369332	1


```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-14058740`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-14058740`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-14058740` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-14058740` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.43107 -0.08183 -0.02156  0.04799  0.60005 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.0514066  0.1212443  33.415  < 2e-16 ***
# `AX-14058740`            0.0679189  0.0526063   1.291   0.1978    
# age                      0.0009485  0.0006643   1.428   0.1545    
# race                     0.0124178  0.0099030   1.254   0.2110    
# gender                  -0.0044243  0.0175555  -0.252   0.8012    
# diagnosis                0.2551831  0.0325697   7.835 1.18e-13 ***
# diagnosissubtype        -0.0119812  0.0096067  -1.247   0.2135    
# pc1                     -0.4382727  0.2466616  -1.777   0.0768 .  
# pc2                      0.1360912  0.1605190   0.848   0.3973    
# pc3                     -0.0201180  0.1736865  -0.116   0.9079    
# pc4                     -0.2772196  0.1396912  -1.985   0.0482 *  
# `AX-14058740`:diagnosis -0.1226158  0.0150061  -8.171 1.31e-14 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1367 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.3273,	Adjusted R-squared:  0.2991 
# F-statistic: 11.59 on 11 and 262 DF,  p-value: < 2.2e-16

min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-14058740'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-14058740`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs72944885 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-14058740`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72944885 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-14058740`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72944885 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-14058740`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-14058740` <- as.factor(df$`AX-14058740`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-14058740`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72944885 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-14058740`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72944885 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL6_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#7 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#AX-16372367	TC04001036.hg.1	7	8.132859817	1.67E-14	1.26E-05	0.267929515	CLNK 	chr4	10488019	10686489	Coding	rs950391	X	86454329	A	G	G	A


x<-merge_v3$`AX-16372367`

#  0   1   2 NA
# 2  14 258 1

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1
#UCDSS.513	8	1	2	71	1	1	1	-0.01459	0.0134688	-0.0145828	-0.00139512	4
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001036.hg.1~`AX-16372367`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372367`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001036.hg.1 ~ `AX-16372367` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372367` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.89888 -0.08895 -0.01187  0.08770  2.15001 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.3907433  0.2338370  27.330  < 2e-16 ***
# `AX-16372367`           -1.0285771  0.0988591 -10.404  < 2e-16 ***
# age                     -0.0005815  0.0010598  -0.549    0.584    
# race                     0.0017482  0.0160047   0.109    0.913    
# gender                  -0.0425327  0.0281079  -1.513    0.131    
# diagnosis               -0.4945724  0.0691796  -7.149 8.69e-12 ***
# diagnosissubtype        -0.0172554  0.0153801  -1.122    0.263    
# pc1                      0.0763734  0.2832263   0.270    0.788    
# pc2                     -0.1811198  0.2498948  -0.725    0.469    
# pc3                      0.1045543  0.2790371   0.375    0.708    
# pc4                      0.2180740  0.2252940   0.968    0.334    
# `AX-16372367`:diagnosis  0.2676457  0.0329793   8.116 1.89e-14 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2192 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.3129,	Adjusted R-squared:  0.2841 
# F-statistic: 10.85 on 11 and 262 DF,  p-value: < 2.2e-16

min(merge_v3$TC04001036.hg.1) #4.00259
max(merge_v3$TC04001036.hg.1) #7.940702
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC04001036.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CLNK expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372367`), y=x$TC04001036.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="CLNK expression", x = "rs950391 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372367`, y=x$TC04001036.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC04001036.hg.1", groupnames =c("diagnosis","`AX-16372367`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372367` <- as.factor(df$`AX-16372367`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372367`), y=df$TC04001036.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372367`), y=df$TC04001036.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL7_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#8 snps	gene	X	statistic	pvalue	FDR	beta	Gene.Symbol	Chromosome.x	Start	Stop	locus.type	dbSNP.RS.ID	Chromosome.y	Physical.Position	Allele.A	Allele.B	Ref.Allele	Alt.Allele
#AX-32623127	TC15001756.hg.1	8	8.098890891	2.08E-14	1.38E-05	0.131705875	AP3B2 	chr15	83328033	83378666	Coding	rs73507341	19	13135197	T	C	T	C


x<-merge_v3$`AX-32623127`

#  0   1   2 NA
#243  25   6  1

#diagnosissubtype	gender	race	age	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia	pc1	pc2	pc3	pc4	diagnosis

#UCDSS.519	7	1	3	43	0	0	0	0.15934	-0.0349643	-0.0133984	-0.0445161	4
#UCDSS.782	6	1	3	42	0	0	0	0.164189	-0.0371417	0.00127105	-0.0370388	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.009	3	1	6	77	0	0	0	-0.0047954	0.0331163	0.0958979	-0.00147604	1
#UCSFCJ.017	2	1	3	63	1	0	0	0.163721	-0.0349639	-0.0122536	-0.0369332	1
#UCSFCJ.050	3	2	3	59	1	0	1	0.170806	-0.0337208	-0.0114983	-0.042197	1


```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-32623127`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-32623127`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-32623127` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-32623127` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.76713 -0.07447 -0.01757  0.04509  0.68678 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.2215649  0.0750142  56.277  < 2e-16 ***
# `AX-32623127`           -0.2110673  0.0403886  -5.226 3.54e-07 ***
# age                      0.0009426  0.0007116   1.325    0.186    
# race                     0.0036388  0.0105708   0.344    0.731    
# gender                   0.0122417  0.0187451   0.653    0.514    
# diagnosis                0.0010865  0.0183466   0.059    0.953    
# diagnosissubtype        -0.0072795  0.0103417  -0.704    0.482    
# pc1                      0.2320050  0.1989546   1.166    0.245    
# pc2                     -0.0830774  0.1672464  -0.497    0.620    
# pc3                      0.0465471  0.1862780   0.250    0.803    
# pc4                     -0.1304620  0.1509999  -0.864    0.388    
# `AX-32623127`:diagnosis  0.1317948  0.0162728   8.099 2.11e-14 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1466 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2265,	Adjusted R-squared:  0.1941 
# F-statistic: 6.977 on 11 and 262 DF,  p-value: 2.423e-10

min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-32623127'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-32623127`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs73507341 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-32623127`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs73507341 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-32623127`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs73507341 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-32623127`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-32623127` <- as.factor(df$`AX-32623127`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-32623127`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs73507341 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-32623127`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs73507341 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL8_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#9	AX-16372367	TC02000318.hg.1	9	7.960308899	5.16E-14	3.04E-05	0.325174108	EML6 	chr2	54950636	55199157	Coding	rs950391	X	86454329	A	G	G	A


x<-merge_v3$`AX-16372367`

#  0   1   2 NA
#  2  14 258 1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC02000318.hg.1~`AX-16372367`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372367`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC02000318.hg.1 ~ `AX-16372367` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372367` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.07358 -0.11597 -0.02183  0.10826  2.65410 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.478309   0.290034  25.784  < 2e-16 ***
# `AX-16372367`           -1.313693   0.122618 -10.714  < 2e-16 ***
# age                     -0.001027   0.001315  -0.781    0.435    
# race                    -0.006791   0.019851  -0.342    0.733    
# gender                  -0.044421   0.034863  -1.274    0.204    
# diagnosis               -0.598122   0.085805  -6.971 2.56e-11 ***
# diagnosissubtype        -0.029660   0.019076  -1.555    0.121    
# pc1                     -0.178471   0.351293  -0.508    0.612    
# pc2                      0.078030   0.309951   0.252    0.801    
# pc3                      0.073035   0.346097   0.211    0.833    
# pc4                     -0.010956   0.279438  -0.039    0.969    
# `AX-16372367`:diagnosis  0.324884   0.040905   7.942 5.87e-14 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2718 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.3281,	Adjusted R-squared:  0.2999 
# F-statistic: 11.63 on 11 and 262 DF,  p-value: < 2.2e-16


min(merge_v3$TC02000318.hg.1) #4.35876
max(merge_v3$TC02000318.hg.1) #9.330151
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC02000318.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="EML6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372367`), y=x$TC02000318.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="EML6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372367`, y=x$TC02000318.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="EML6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC02000318.hg.1", groupnames =c("diagnosis","`AX-16372367`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372367` <- as.factor(df$`AX-16372367`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372367`), y=df$TC02000318.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="EML6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372367`), y=df$TC02000318.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="EML6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL9_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#10	AX-50823541	TC08001664.hg.1	10	7.79880739	1.47E-13	7.79E-05	0.148685051	ZFAT 	chr8	135490031	135725292	Coding	rs12833155	X	42486482	A	C	A	C


x<-merge_v3$`AX-50823541`

#  0   1   2 
#  246  18  11 

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC08001664.hg.1~`AX-50823541`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-50823541`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC08001664.hg.1 ~ `AX-50823541` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-50823541` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.67164 -0.09647  0.00511  0.09437  1.58495 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.3826719  0.1040727  61.329  < 2e-16 ***
# `AX-50823541`           -0.1462957  0.0484637  -3.019  0.00279 ** 
# age                     -0.0001573  0.0009873  -0.159  0.87351    
# race                     0.0114574  0.0147099   0.779  0.43675    
# gender                  -0.0572561  0.0260027  -2.202  0.02854 *  
# diagnosis                0.0279697  0.0250616   1.116  0.26542    
# diagnosissubtype        -0.0118407  0.0143583  -0.825  0.41032    
# pc1                      0.0906122  0.2620143   0.346  0.72975    
# pc2                     -0.0173652  0.2296816  -0.076  0.93979    
# pc3                     -0.1052896  0.2595747  -0.406  0.68535    
# pc4                      0.1084572  0.2070363   0.524  0.60082    
# `AX-50823541`:diagnosis  0.1486851  0.0190651   7.799 1.47e-13 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.204 on 263 degrees of freedom
# Multiple R-squared:  0.2975,	Adjusted R-squared:  0.2681 
# F-statistic: 10.13 on 11 and 263 DF,  p-value: 2.273e-15


min(merge_v3$TC08001664.hg.1) #5.858995
max(merge_v3$TC08001664.hg.1) #8.907337
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC08001664.hg.1, x = as.factor(merge_v3$`AX-50823541`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZFAT expression", x = "rs12833155 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AC", "2"="CC"))+scale_y_continuous(expression(paste( italic("ZFAT")," " , "expression")))+expand_limits(y=c(5,9))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-50823541`), y=merge_v3$TC08001664.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ZFAT expression", x = "rs12833155 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AC", "CC"))+scale_y_continuous(expression(paste(italic("ZFAT")," " , "expression")))+ expand_limits(y=c(5,9))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-50823541`, y=merge_v3$TC08001664.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5,9))

p2 <-p1 + labs(y="ZFAT expression", x = "rs12833155 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZFAT")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC08001664.hg.1", groupnames =c("diagnosis","`AX-50823541`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-50823541` <- as.factor(df$`AX-50823541`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-50823541`), y=df$TC08001664.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC08001664.hg.1-sd, ymax=TC08001664.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5,9))

p2 <-p1 + labs(y="ZFAT expression", x = "rs12833155 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZFAT")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-50823541`), y=df$TC08001664.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC08001664.hg.1-sd, ymax=TC08001664.hg.1+sd),width=.2)+ expand_limits(y=c(5,9))

p2 <-p1 + labs(y="ZFAT expression", x = "rs12833155 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZFAT")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL10_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#11	AX-11173316	TC04001036.hg.1	11	-7.638788067	4.09E-13	0.00018386	-0.213074118	CLNK 	chr4	10488019	10686489	Coding	rs11758921	6	166247384	A	G	A	G


x<-merge_v3$`AX-11173316`

#  0   1   2 
#244  29   2 

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001036.hg.1~`AX-11173316`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11173316`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001036.hg.1 ~ `AX-11173316` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11173316` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.68236 -0.08660 -0.00400  0.08058  2.45916 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.1751086  0.1144529  36.479  < 2e-16 ***
# `AX-11173316`            0.8491283  0.0906568   9.366  < 2e-16 ***
# age                     -0.0006912  0.0010908  -0.634  0.52684    
# race                     0.0250542  0.0162398   1.543  0.12409    
# gender                  -0.0450760  0.0287822  -1.566  0.11853    
# diagnosis               -0.0125189  0.0278085  -0.450  0.65295    
# diagnosissubtype         0.0175764  0.0160839   1.093  0.27548    
# pc1                      0.7939814  0.2899242   2.739  0.00659 ** 
# pc2                      0.0247904  0.2540405   0.098  0.92234    
# pc3                     -0.2463295  0.2874695  -0.857  0.39229    
# pc4                      0.1840677  0.2287759   0.805  0.42179    
# `AX-11173316`:diagnosis -0.2130741  0.0278937  -7.639 4.09e-13 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2254 on 263 degrees of freedom
# Multiple R-squared:  0.2712,	Adjusted R-squared:  0.2407 
# F-statistic: 8.897 on 11 and 263 DF,  p-value: 1.903e-13


min(merge_v3$TC04001036.hg.1) #4.00259
max(merge_v3$TC04001036.hg.1) #7.940702
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC04001036.hg.1, x = as.factor(merge_v3$`AX-11173316`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CLNK expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11173316`), y=merge_v3$TC04001036.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="CLNK expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_point.jpg",width = 6, height = 4)

```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11173316`, y=merge_v3$TC04001036.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC04001036.hg.1", groupnames =c("diagnosis","`AX-11173316`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11173316` <- as.factor(df$`AX-11173316`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11173316`), y=df$TC04001036.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11173316`), y=df$TC04001036.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL11_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#12	AX-16372292	TC17001827.hg.1	12	-7.635906573	4.16E-13	0.00018386	-0.316079099	ABCA6 	chr17	67074847	67138015	Coding	rs2369519	X	86392534	A	G	G	A


x<-merge_v3$`AX-16372292`

#  0   1   2 NA
#248  18  7 2

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17001827.hg.1~`AX-16372292`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372292`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17001827.hg.1 ~ `AX-16372292` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372292` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.5897 -0.1301 -0.0230  0.1157  4.6922 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.226952   0.193577  21.836  < 2e-16 ***
# `AX-16372292`            1.325720   0.139192   9.524  < 2e-16 ***
# age                      0.001861   0.001841   1.011  0.31286    
# race                    -0.009999   0.027769  -0.360  0.71908    
# gender                  -0.101158   0.048860  -2.070  0.03940 *  
# diagnosis                0.059680   0.047180   1.265  0.20702    
# diagnosissubtype        -0.031377   0.026808  -1.170  0.24288    
# pc1                     -1.480144   0.516263  -2.867  0.00448 ** 
# pc2                     -0.011042   0.437926  -0.025  0.97990    
# pc3                      0.244589   0.485016   0.504  0.61448    
# pc4                     -0.611234   0.395157  -1.547  0.12312    
# `AX-16372292`:diagnosis -0.316457   0.041542  -7.618 4.76e-13 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3811 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2765,	Adjusted R-squared:  0.2461 
# F-statistic:  9.07 on 11 and 261 DF,  p-value: 1.055e-13


min(merge_v3$TC17001827.hg.1) #3.87316
max(merge_v3$TC17001827.hg.1) #10.95106
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372292'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17001827.hg.1, x = as.factor(x$`AX-16372292`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(3,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372292`), y=x$TC17001827.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(3,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372292`, y=x$TC17001827.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC17001827.hg.1", groupnames =c("diagnosis","`AX-16372292`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372292` <- as.factor(df$`AX-16372292`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372292`), y=df$TC17001827.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372292`), y=df$TC17001827.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd),width=.2)+ expand_limits(y=c(3,12))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL12_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#13	AX-39865613	TC15001756.hg.1	13	-7.556448071	6.88E-13	0.00028072	-0.124492903	AP3B2 	chr15	83328033	83378666	Coding	rs11853524	15	25508955	T	G	G	T


x<-merge_v3$`AX-39865613`

#  0   1   2 NA
#  5  18 251 1
#UCDSS.571	2	1	3	69	1	0	0	0.18122	-0.034571	-0.0165888	-0.102848	1
#UCDSS.693	8	2	3	48	1	0	0	0.141	-0.0354487	-0.0212148	-0.0503794	4
#UCDSS.716	1	1	3	56	1	0	0	0.144306	-0.0374005	-0.0204159	-0.0239198	1
#UCDSS.845	3	1	3	56	0	1	0	0.131643	-0.0324731	-0.00850289	-0.023962	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-39865613`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-39865613`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-39865613` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-39865613` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.44147 -0.07827 -0.02133  0.05205  0.61388 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.9986125  0.1136383  35.187  < 2e-16 ***
# `AX-39865613`            0.1042077  0.0462486   2.253   0.0251 *  
# age                      0.0009828  0.0006924   1.419   0.1569    
# race                     0.0040346  0.0103487   0.390   0.6970    
# gender                   0.0121593  0.0182181   0.667   0.5051    
# diagnosis                0.2594109  0.0366256   7.083 1.30e-11 ***
# diagnosissubtype        -0.0121603  0.0100245  -1.213   0.2262    
# pc1                     -0.2503213  0.2065459  -1.212   0.2266    
# pc2                     -0.1311586  0.1607699  -0.816   0.4153    
# pc3                      0.1166988  0.1825164   0.639   0.5231    
# pc4                     -0.0244336  0.1466120  -0.167   0.8678    
# `AX-39865613`:diagnosis -0.1245538  0.0165010  -7.548 7.32e-13 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1425 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2692,	Adjusted R-squared:  0.2386 
# F-statistic: 8.776 on 11 and 262 DF,  p-value: 3.021e-13


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-39865613'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-39865613`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GT", "2"="TT"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-39865613`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11853524 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GT", "TT"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-39865613`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11853524 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-39865613`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-39865613` <- as.factor(df$`AX-39865613`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-39865613`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11853524 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-39865613`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs11853524 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL13_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#14 AX-38679391	TC03000198.hg.1	14	7.524152225	8.44E-13	0.000319535	0.150130069	TTC21A 	chr3	39149152	39180394	Coding	rs139929471	X	88063578	A	G	G	A



x<-merge_v3$`AX-38679391`

#  0   1   2 NA
#  14  11 249  1


```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC03000198.hg.1~`AX-38679391`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38679391`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC03000198.hg.1 ~ `AX-38679391` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38679391` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.98983 -0.06291 -0.01049  0.05607  2.04595 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.0965706  0.1635725  37.271  < 2e-16 ***
# `AX-38679391`           -0.6120008  0.0715918  -8.548 1.05e-15 ***
# age                     -0.0005415  0.0009215  -0.588  0.55729    
# race                     0.0224985  0.0134995   1.667  0.09678 .  
# gender                  -0.0144045  0.0238972  -0.603  0.54718    
# diagnosis               -0.2868074  0.0438915  -6.534 3.31e-10 ***
# diagnosissubtype        -0.0027207  0.0130827  -0.208  0.83542    
# pc1                      0.2337647  0.2405073   0.972  0.33196    
# pc2                      0.5516198  0.2090069   2.639  0.00881 ** 
# pc3                     -0.4888905  0.2609077  -1.874  0.06207 .  
# pc4                      0.0094193  0.1892974   0.050  0.96035    
# `AX-38679391`:diagnosis  0.1501380  0.0199781   7.515 9.01e-13 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1862 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2404,	Adjusted R-squared:  0.2085 
# F-statistic: 7.538 on 11 and 262 DF,  p-value: 2.939e-11



min(merge_v3$TC03000198.hg.1) #4.719301
max(merge_v3$TC03000198.hg.1) #7.918954
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-38679391'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC03000198.hg.1, x = as.factor(x$`AX-38679391`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="TTC21A expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("TTC21A")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38679391`), y=x$TC03000198.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="TTC21A expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("TTC21A")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38679391`, y=x$TC03000198.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="TTC21A expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TTC21A")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC03000198.hg.1", groupnames =c("diagnosis","`AX-38679391`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38679391` <- as.factor(df$`AX-38679391`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38679391`), y=df$TC03000198.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC03000198.hg.1-sd, ymax=TC03000198.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="TTC21A expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TTC21A")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38679391`), y=df$TC03000198.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC03000198.hg.1-sd, ymax=TC03000198.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="TTC21A expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TTC21A")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL14_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#15 AX-33587459	TC02000318.hg.1	15	-7.505122813	9.51E-13	0.000336164	-0.270441634	EML6 	chr2	54950636	55199157	Coding	rs79434685	22	29556745	C	G	C	G


x<-merge_v3$`AX-33587459`

#  0   1   2 
#  246  28   1 

```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#16 AX-14631945	TC15001756.hg.1	16	7.482865078	1.09E-12	0.00036244	0.130956399	AP3B2 	chr15	83328033	83378666	Coding	rs7664829	4	21791787	A	G	A	G



x<-merge_v3$`AX-14631945`

#  0   1   2 NA
# 250  18   5 2

#UCDSS.602	3	1	3	51	1	1	0	0.119142	-0.0404059	-0.0206839	-0.0306819	1
#UCDSS.681	4	2	7	70	1	1	1	0.176038	-0.0336108	0.0226537	-0.0244828	1
#UCDSS.693	8	2	3	48	1	0	0	0.141	-0.0354487	-0.0212148	-0.0503794	4
#UCDSS.708	8	2	3	41	1	1	1	0.161344	-0.0350806	0.0073334	-0.0420488	4
#UCSFCJ.050	3	2	3	59	1	0	1	0.170806	-0.0337208	-0.0114983	-0.042197	1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-14631945`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-14631945`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-14631945` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-14631945` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.43755 -0.07633 -0.02314  0.04200  1.00550 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.2142421  0.0742420  56.764  < 2e-16 ***
# `AX-14631945`           -0.1432189  0.0497562  -2.878  0.00433 ** 
# age                      0.0012459  0.0007153   1.742  0.08272 .  
# race                     0.0043035  0.0106746   0.403  0.68716    
# gender                  -0.0091939  0.0188956  -0.487  0.62698    
# diagnosis                0.0026470  0.0182249   0.145  0.88463    
# diagnosissubtype        -0.0067855  0.0103754  -0.654  0.51369    
# pc1                     -0.1798565  0.2337959  -0.769  0.44242    
# pc2                      0.0077597  0.1667131   0.047  0.96291    
# pc3                      0.0779088  0.1861667   0.418  0.67593    
# pc4                     -0.0045117  0.1494857  -0.030  0.97595    
# `AX-14631945`:diagnosis  0.1311773  0.0175492   7.475 1.17e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.146 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2345,	Adjusted R-squared:  0.2022 
# F-statistic: 7.268 on 11 and 261 DF,  p-value: 8.195e-11



min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-14631945'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-14631945`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs79434685 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-14631945`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs79434685 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-14631945`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs79434685 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-14631945`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-14631945` <- as.factor(df$`AX-14631945`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-14631945`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs79434685 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-14631945`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs79434685 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL16_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#17	AX-15355229	TC01001640.hg.1	18	7.460585989	1.26E-12	0.000370463	0.27607835	PTPRC 	chr1	198607801	198726545	Coding	rs1063632	6	31378510	A	G	G	A



x<-merge_v3$`AX-15355229`

#  0   1   2 NA
# 2  31 240 2

#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.844	3	2	6	47	1	0	0	-0.00837881	0.0458948	0.108759	-0.00399842	1


```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01001640.hg.1~`AX-15355229`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-15355229`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01001640.hg.1 ~ `AX-15355229` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-15355229` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -3.14721 -0.10769  0.01530  0.09419  0.92136 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             18.591530   0.228314  81.430  < 2e-16 ***
# `AX-15355229`           -0.297312   0.090419  -3.288  0.00115 ** 
# age                      0.002348   0.001492   1.574  0.11664    
# race                    -0.034007   0.022093  -1.539  0.12496    
# gender                  -0.024756   0.038808  -0.638  0.52409    
# diagnosis               -0.561602   0.078457  -7.158 8.29e-12 ***
# diagnosissubtype         0.008692   0.021334   0.407  0.68403    
# pc1                     -0.162070   0.386872  -0.419  0.67562    
# pc2                      0.013967   0.345531   0.040  0.96779    
# pc3                      0.933235   0.384623   2.426  0.01593 *  
# pc4                      0.140942   0.306554   0.460  0.64607    
# `AX-15355229`:diagnosis  0.276819   0.037078   7.466 1.24e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.302 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2707,	Adjusted R-squared:  0.2399 
# F-statistic: 8.805 on 11 and 261 DF,  p-value: 2.763e-13


min(merge_v3$TC01001640.hg.1) #13.08913
max(merge_v3$TC01001640.hg.1) #18.45755
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-15355229'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC01001640.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PTPRC expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("PTPRC")," " , "expression")))+expand_limits(y=c(13,19))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-15355229`), y=x$TC01001640.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="PTPRC expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ expand_limits(y=c(13,19))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-15355229`, y=x$TC01001640.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC01001640.hg.1", groupnames =c("diagnosis","`AX-15355229`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-15355229` <- as.factor(df$`AX-15355229`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-15355229`), y=df$TC01001640.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01001640.hg.1-sd, ymax=TC01001640.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-15355229`), y=df$TC01001640.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01001640.hg.1-sd, ymax=TC01001640.hg.1+sd),width=.2)+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL17_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#18 AX-51270082	TC01001640.hg.1	17	-7.466976322	1.21E-12	0.000370463	-0.273366826	PTPRC 	chr1	198607801	198726545	Coding	rs9779183	X	13009957	T	C	T	C




x<-merge_v3$`AX-51270082`

#  0   1   2 NA
# 258   8   7 2

#2
#UCDSS.424	1	2	3	67	1	0	0	0.161859	-0.035007	0.0250816	0.712585	1
#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.517	8	2	3	45	1	0	0	0.126155	-0.0357504	0.025814	0.649375	4
#UCDSS.845	3	1	3	56	0	1	0	0.131643	-0.0324731	-0.00850289	-0.023962	1
#UCSFCJ.017	2	1	3	63	1	0	0	0.163721	-0.0349639	-0.0122536	-0.0369332	1
#UCSFCJ.101	2	1	5	46	1	0	1	-0.031927	-0.0306845	-0.0165275	0.00071694	1
#UCSFCJ.109	4	1	2	53	1	1	1	-0.0323482	-0.0282733	-0.0178268	0.00268296	1 (Hemorrhagic Conversion/HTA1)

#1
#UCDSS.201	8	2	5	72	1	0	0	-0.0303633	-0.0330236	-0.0209832	-0.00285422	4
#UCDSS.281	8	2	5	64	1	1	0	-0.0328373	-0.0284304	-0.0233561	-0.00423923	4
#UCDSS.510	8	2	5	63	1	0	1	-0.0295969	-0.0287967	-0.0210884	-0.000399126	4
#UCDSS.525	4	2	5	60	0	0	1	-0.0319856	-0.0308957	-0.0116315	0.00586204	1
#UCDSS.555	8	2	5	85	1	0	1	-0.0315911	-0.0276138	-0.0168431	7.76E-06	4
#UCSFCJ.005	3	2	3	53	1	1	0	0.102322	-0.0350191	-0.0147273	-0.0238308	1
#UCSFCJ.091	6	2	5	43	0	0	0	0.00300048	0.0634869	0.136817	-0.00680118	1
#UCSFCJ.124	2	2	5	83	0	0	0	-0.031664	-0.0305188	-0.0216808	-0.00661383	1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01001640.hg.1~`AX-51270082`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-51270082`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01001640.hg.1 ~ `AX-51270082` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-51270082` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -2.85842 -0.11346  0.02363  0.09963  0.98609 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             18.063556   0.154498 116.918  < 2e-16 ***
# `AX-51270082`            0.166150   0.091136   1.823  0.06943 .  
# age                      0.002235   0.001462   1.529  0.12754    
# race                    -0.057861   0.021837  -2.650  0.00855 ** 
# gender                   0.020204   0.037968   0.532  0.59509    
# diagnosis               -0.024850   0.036525  -0.680  0.49687    
# diagnosissubtype         0.015335   0.020879   0.734  0.46331    
# pc1                     -0.070971   0.381436  -0.186  0.85254    
# pc2                     -0.303638   0.345160  -0.880  0.37983    
# pc3                      0.936664   0.383514   2.442  0.01526 *  
# pc4                      1.422734   0.338828   4.199 3.68e-05 ***
# `AX-51270082`:diagnosis -0.272243   0.036680  -7.422 1.63e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2951 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.3046,	Adjusted R-squared:  0.2753 
# F-statistic: 10.39 on 11 and 261 DF,  p-value: 9.257e-16


min(merge_v3$TC01001640.hg.1) #13.08913
max(merge_v3$TC01001640.hg.1) #18.45755
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-51270082'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC01001640.hg.1, x = as.factor(x$`AX-51270082`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PTPRC expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("PTPRC")," " , "expression")))+expand_limits(y=c(13,19))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-51270082`), y=x$TC01001640.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="PTPRC expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ expand_limits(y=c(13,19))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-51270082`, y=x$TC01001640.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC01001640.hg.1", groupnames =c("diagnosis","`AX-51270082`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-51270082` <- as.factor(df$`AX-51270082`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-51270082`), y=df$TC01001640.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01001640.hg.1-sd, ymax=TC01001640.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-51270082`), y=df$TC01001640.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01001640.hg.1-sd, ymax=TC01001640.hg.1+sd),width=.2)+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL18_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#19	AX-11393384	TC04001036.hg.1	19	-7.450029432	1.34E-12	0.000374945	-0.211053993	CLNK 	chr4	10488019	10686489	Coding	rs2464504	4	48232441	T	C	C	T



x<-merge_v3$`AX-11393384`

#  0   1   2
# 250  22   3

#UCDSS.272	8	2	5	57	1	0	1	-0.0273933	-0.0257722	-0.0115635	0.000843584	4
#UCDSS.343	8	2	5	63	0	0	0	-0.0319583	-0.0307143	-0.0203161	0.000106389	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001036.hg.1~`AX-11393384`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11393384`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001036.hg.1 ~ `AX-11393384` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11393384` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.73202 -0.08758 -0.01073  0.08935  2.41487 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.1630956  0.1145714  36.336  < 2e-16 ***
# `AX-11393384`            0.8508078  0.0919733   9.251  < 2e-16 ***
# age                      0.0001326  0.0010913   0.121   0.9034    
# race                     0.0234102  0.0162732   1.439   0.1515    
# gender                  -0.0287785  0.0288850  -0.996   0.3200    
# diagnosis                0.0459640  0.0279002   1.647   0.1007    
# diagnosissubtype        -0.0193666  0.0158536  -1.222   0.2230    
# pc1                      0.7907358  0.2919172   2.709   0.0072 ** 
# pc2                     -0.0619534  0.2558286  -0.242   0.8088    
# pc3                     -0.1528054  0.2876113  -0.531   0.5957    
# pc4                      0.1258309  0.2293833   0.549   0.5838    
# `AX-11393384`:diagnosis -0.2110540  0.0283293  -7.450 1.34e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.226 on 263 degrees of freedom
# Multiple R-squared:  0.2669,	Adjusted R-squared:  0.2362 
# F-statistic: 8.705 on 11 and 263 DF,  p-value: 3.854e-13


min(merge_v3$TC04001036.hg.1) #4.00259
max(merge_v3$TC04001036.hg.1) #7.940702
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-11393384'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC04001036.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CLNK expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11393384`), y=merge_v3$TC04001036.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="CLNK expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11393384`, y=merge_v3$TC04001036.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC04001036.hg.1", groupnames =c("diagnosis","`AX-11393384`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11393384` <- as.factor(df$`AX-11393384`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11393384`), y=df$TC04001036.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11393384`), y=df$TC04001036.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL19_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#20	AX-16372367	TC0X001190.hg.1	21	7.413179123	1.69E-12	0.000427075	0.30116745	POF1B 	chrX	84532395	84634748	Coding	rs950391	X	86454329	A	G	G	A



x<-merge_v3$`AX-16372367`

# 0   1   2 NA
#  2  14 258  1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC0X001190.hg.1~`AX-16372367`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372367`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC0X001190.hg.1 ~ `AX-16372367` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372367` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.91603 -0.13529 -0.02245  0.09446  2.17587 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.573922   0.286909  22.913  < 2e-16 ***
# `AX-16372367`           -1.157538   0.121296  -9.543  < 2e-16 ***
# age                      0.002591   0.001300   1.992   0.0474 *  
# race                    -0.004267   0.019637  -0.217   0.8282    
# gender                   0.028177   0.034487   0.817   0.4146    
# diagnosis               -0.574968   0.084881  -6.774 8.23e-11 ***
# diagnosissubtype        -0.012514   0.018871  -0.663   0.5078    
# pc1                     -0.218588   0.347507  -0.629   0.5299    
# pc2                     -0.021323   0.306611  -0.070   0.9446    
# pc3                      0.103726   0.342367   0.303   0.7622    
# pc4                      0.303370   0.276427   1.097   0.2734    
# `AX-16372367`:diagnosis  0.300275   0.040464   7.421 1.63e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2689 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.288,	Adjusted R-squared:  0.2581 
# F-statistic: 9.633 on 11 and 262 DF,  p-value: 1.353e-14


min(merge_v3$TC0X001190.hg.1) #4.030025
max(merge_v3$TC0X001190.hg.1) #8.337179
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC0X001190.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="POF1B expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("POF1B")," " , "expression")))+expand_limits(y=c(4,9))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372367`), y=x$TC0X001190.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="POF1B expression", x = "rs950391 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("POF1B")," " , "expression")))+ expand_limits(y=c(4,9))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372367`, y=x$TC0X001190.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,9))

p2 <-p1 + labs(y="POF1B expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("POF1B")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC0X001190.hg.1", groupnames =c("diagnosis","`AX-16372367`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372367` <- as.factor(df$`AX-16372367`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372367`), y=df$TC0X001190.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC0X001190.hg.1-sd, ymax=TC0X001190.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,9))

p2 <-p1 + labs(y="POF1B expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("POF1B")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372367`), y=df$TC0X001190.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC0X001190.hg.1-sd, ymax=TC0X001190.hg.1+sd),width=.2)+ expand_limits(y=c(4,9))

p2 <-p1 + labs(y="POF1B expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("POF1B")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL20_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT20_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#21	AX-38679391	TC01000514.hg.1	20	7.414250469	1.68E-12	0.000427075	0.239309544	ZNF684 	chr1	40997233	41013841	Coding	rs139929471	X	88063578	A	G	G	A


x<-merge_v3$`AX-38679391`

# 0   1   2 NA
# 14  11 249 1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01000514.hg.1~`AX-38679391`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38679391`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01000514.hg.1 ~ `AX-38679391` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38679391` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.29689 -0.14317 -0.01421  0.12658  3.04412 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.759770   0.264756  29.309  < 2e-16 ***
# `AX-38679391`           -0.940237   0.115878  -8.114 1.91e-14 ***
# age                     -0.001356   0.001491  -0.909    0.364    
# race                     0.015401   0.021850   0.705    0.482    
# gender                   0.001801   0.038680   0.047    0.963    
# diagnosis               -0.479252   0.071042  -6.746 9.69e-11 ***
# diagnosissubtype        -0.001857   0.021175  -0.088    0.930    
# pc1                      0.222381   0.389282   0.571    0.568    
# pc2                      0.038503   0.338296   0.114    0.909    
# pc3                     -0.611833   0.422302  -1.449    0.149    
# pc4                      0.140445   0.306394   0.458    0.647    
# `AX-38679391`:diagnosis  0.239314   0.032336   7.401 1.84e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3014 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2159,	Adjusted R-squared:  0.183 
# F-statistic: 6.558 on 11 and 262 DF,  p-value: 1.179e-09


min(merge_v3$TC01000514.hg.1) #5.342232
max(merge_v3$TC01000514.hg.1) #10.31879
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-38679391'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC01000514.hg.1, x = as.factor(x$`AX-38679391`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZNF684 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ZNF684")," " , "expression")))+expand_limits(y=c(5,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38679391`), y=x$TC01000514.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ZNF684 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ expand_limits(y=c(5,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38679391`, y=x$TC01000514.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5,11))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC01000514.hg.1", groupnames =c("diagnosis","`AX-38679391`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38679391` <- as.factor(df$`AX-38679391`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38679391`), y=df$TC01000514.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5,11))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38679391`), y=df$TC01000514.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd),width=.2)+ expand_limits(y=c(5,11))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL21_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT21_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#22	39	AX-35735671	TC01001640.hg.1	23	7.358157746	2.38E-12	0.000536009	0.27801048	PTPRC 	chr1	198607801	198726545	Coding	rs1051785	6	31378388	A	G	G	A


x<-merge_v3$`AX-35735671`

# 0   1   2 
# 1  32 240 

```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#23	AX-38686587	TC0X000829.hg.1	22	-7.365592398	2.27E-12	0.000536009	-0.296445063	ANOS1 	chrX	8496915	8700227	Coding	rs149957475	X	93351607	T	C	C	T


x<-merge_v3$`AX-38686587`

# 0   1   2 
# 3  16 256 
#UCDSS.347	3	1	3	75	1	0	1	0.105963	-0.0356412	-0.0275675	-0.0349564	1
#UCDSS.709	3	1	6	55	1	1	1	-0.00616543	0.0499376	0.114078	0.00200624	1
#UCDSS.845	3	1	3	56	0	1	0	0.131643	-0.0324731	-0.00850289	-0.023962	1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC0X000829.hg.1~`AX-38686587`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38686587`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC0X000829.hg.1 ~ `AX-38686587` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38686587` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.84731 -0.09938 -0.01420  0.06943  1.50539 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.6158858  0.1627013  22.224  < 2e-16 ***
# `AX-38686587`            0.3463659  0.0665872   5.202 3.98e-07 ***
# age                      0.0019183  0.0009363   2.049   0.0415 *  
# race                    -0.0077140  0.0139639  -0.552   0.5811    
# gender                  -0.1260939  0.0248565  -5.073 7.41e-07 ***
# diagnosis                0.5776771  0.0824925   7.003 2.09e-11 ***
# diagnosissubtype         0.0021699  0.0135417   0.160   0.8728    
# pc1                     -0.2560791  0.2490604  -1.028   0.3048    
# pc2                      0.0161468  0.2170340   0.074   0.9408    
# pc3                      0.4924665  0.2451700   2.009   0.0456 *  
# pc4                     -0.0680970  0.1962075  -0.347   0.7288    
# `AX-38686587`:diagnosis -0.2964451  0.0402473  -7.366 2.27e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1931 on 263 degrees of freedom
# Multiple R-squared:  0.2429,	Adjusted R-squared:  0.2112 
# F-statistic: 7.671 on 11 and 263 DF,  p-value: 1.765e-11


min(merge_v3$TC0X000829.hg.1) #3.764966
max(merge_v3$TC0X000829.hg.1) #6.515363
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-38686587'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC0X000829.hg.1, x = as.factor(merge_v3$`AX-38686587`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ANOS1 expression", x = "rs149957475 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("ANOS1")," " , "expression")))+expand_limits(y=c(3,7))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-38686587`), y=merge_v3$TC0X000829.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ANOS1 expression", x = "rs149957475 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ expand_limits(y=c(3,7))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-38686587`, y=merge_v3$TC0X000829.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs149957475 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC0X000829.hg.1", groupnames =c("diagnosis","`AX-38686587`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38686587` <- as.factor(df$`AX-38686587`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38686587`), y=df$TC0X000829.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC0X000829.hg.1-sd, ymax=TC0X000829.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs149957475 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38686587`), y=df$TC0X000829.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC0X000829.hg.1-sd, ymax=TC0X000829.hg.1+sd),width=.2)+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs149957475 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL23_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT23_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#24 AX-41205499	TC15001756.hg.1	24	7.355220367	2.43E-12	0.000536009	0.117810999	AP3B2 	chr15	83328033	83378666	Coding	rs9847733	3	23242050	A	G	A	G



x<-merge_v3$`AX-41205499`

# 0   1   2  NA
# 245  20   6 4
#UCDSS.693	8	2	3	48	1	0	0	0.141	-0.0354487	-0.0212148	-0.0503794	4
#UCDSS.716	1	1	3	56	1	0	0	0.144306	-0.0374005	-0.0204159	-0.0239198	1
#UCDSS.730	4	1	3	55	1	1	1	0.160227	-0.0337266	0.0066151	-0.0470337	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.050	3	2	3	59	1	0	1	0.170806	-0.0337208	-0.0114983	-0.042197	1
#UCSFCJ.091	6	2	5	43	0	0	0	0.00300048	0.0634869	0.136817	-0.00680118	1


```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-41205499`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-41205499`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-41205499` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-41205499` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.47679 -0.07844 -0.02253  0.05391  0.74322 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.2439703  0.0763901  55.557  < 2e-16 ***
# `AX-41205499`           -0.1611257  0.0461781  -3.489 0.000569 ***
# age                      0.0007706  0.0007271   1.060 0.290220    
# race                    -0.0013740  0.0107441  -0.128 0.898341    
# gender                   0.0141988  0.0190181   0.747 0.455986    
# diagnosis                0.0024471  0.0183431   0.133 0.893976    
# diagnosissubtype        -0.0077693  0.0104017  -0.747 0.455785    
# pc1                     -0.0990902  0.2162105  -0.458 0.647118    
# pc2                     -0.0916909  0.1687918  -0.543 0.587446    
# pc3                      0.0253857  0.1896725   0.134 0.893633    
# pc4                     -0.0576577  0.1534381  -0.376 0.707394    
# `AX-41205499`:diagnosis  0.1178416  0.0160781   7.329 2.95e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1482 on 259 degrees of freedom
#   (4 observations deleted due to missingness)
# Multiple R-squared:  0.2146,	Adjusted R-squared:  0.1812 
# F-statistic: 6.432 on 11 and 259 DF,  p-value: 1.963e-09


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-41205499'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-41205499`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ANOS1 expression", x = "rs9847733 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("ANOS1")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-41205499`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ANOS1 expression", x = "rs9847733 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-41205499`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs9847733 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-41205499`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-41205499` <- as.factor(df$`AX-41205499`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-41205499`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs9847733 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-41205499`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="ANOS1 expression", x = "rs9847733 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ANOS1")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL24_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT24_scattererrorbar.jpg",width = 6, height = 4)
```



```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#25 AX-15355229	TC18000998.hg.1	25	7.339935063	2.67E-12	0.000565758	0.241326569	SMAD4 	chr18	48556583	48611415	Coding	rs1063632	6	31378510	A	G	G	A



x<-merge_v3$`AX-15355229`

# 0   1   2  NA
#  2  31 240 2

#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.844	3	2	6	47	1	0	0	-0.00837881	0.0458948	0.108759	-0.00399842	1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC18000998.hg.1~`AX-15355229`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-15355229`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC18000998.hg.1 ~ `AX-15355229` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-15355229` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -2.42912 -0.11114  0.00069  0.12900  0.81591 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             11.224899   0.202866  55.332  < 2e-16 ***
# `AX-15355229`           -0.289590   0.080341  -3.605 0.000375 ***
# age                     -0.001640   0.001325  -1.237 0.217017    
# race                    -0.013807   0.019631  -0.703 0.482485    
# gender                  -0.034226   0.034482  -0.993 0.321833    
# diagnosis               -0.477451   0.069712  -6.849 5.32e-11 ***
# diagnosissubtype         0.007643   0.018956   0.403 0.687145    
# pc1                      0.070942   0.343752   0.206 0.836659    
# pc2                     -0.097677   0.307019  -0.318 0.750629    
# pc3                      0.061169   0.341753   0.179 0.858088    
# pc4                      0.030681   0.272386   0.113 0.910406    
# `AX-15355229`:diagnosis  0.241946   0.032945   7.344 2.65e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2683 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2271,	Adjusted R-squared:  0.1945 
# F-statistic: 6.972 on 11 and 261 DF,  p-value: 2.495e-10


min(merge_v3$TC18000998.hg.1) #6.752242
max(merge_v3$TC18000998.hg.1) #10.89588
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-15355229'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC18000998.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SMAD4 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("SMAD4")," " , "expression")))+expand_limits(y=c(6,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-15355229`), y=x$TC18000998.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="SMAD4 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ expand_limits(y=c(6,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-15355229`, y=x$TC18000998.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(6,11))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC18000998.hg.1", groupnames =c("diagnosis","`AX-15355229`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-15355229` <- as.factor(df$`AX-15355229`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-15355229`), y=df$TC18000998.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(6,11))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-15355229`), y=df$TC18000998.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd),width=.2)+ expand_limits(y=c(6,11))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL25_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT25_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#26	AX-42755741	TC09000560.hg.1	26	7.32798207	2.87E-12	0.000585826	0.376896665	UGCG 	chr9	114659046	114697649	Coding	rs12399124	X	3544089	A	G	G	A


x<-merge_v3$`AX-42755741`

# 0   1   2  NA
#  7  13 254 1

#UCDSS.245	2	1	5	63	1	0	0	-0.0321194	-0.0298252	-0.0169677	0.000904222	1
#UCDSS.573	7	1	5	74	0	0	0	-0.0330446	-0.0322942	-0.0143157	0.00197388	4
#UCDSS.711	8	1	5	60	0	0	0	-0.0329261	-0.0298791	-0.023036	-0.00107866	4
#UCDSS.758	8	1	5	60	1	0	1	-0.0313821	-0.0289029	-0.0154322	-0.00246663	4
#UCDSS.871	8	1	5	58	0	0	1	-0.0335037	-0.0262638	-0.0132783	0.00269816	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1
#UofASS.002	4	1	5	68	1	0	1	-0.0317127	-0.0293228	-0.0169263	0.000271324	1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC09000560.hg.1~`AX-42755741`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-42755741`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC09000560.hg.1 ~ `AX-42755741` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-42755741` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.43388 -0.23315 -0.04675  0.12846  2.86302 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             11.427967   0.389568  29.335  < 2e-16 ***
# `AX-42755741`           -1.468025   0.163640  -8.971  < 2e-16 ***
# age                     -0.001642   0.002254  -0.728    0.467    
# race                    -0.029689   0.033521  -0.886    0.377    
# gender                   0.071009   0.059347   1.197    0.233    
# diagnosis               -0.776882   0.117232  -6.627 1.94e-10 ***
# diagnosissubtype        -0.010301   0.033077  -0.311    0.756    
# pc1                     -0.813258   0.598123  -1.360    0.175    
# pc2                     -0.180076   0.526144  -0.342    0.732    
# pc3                     -0.170813   0.591516  -0.289    0.773    
# pc4                     -0.024950   0.472823  -0.053    0.958    
# `AX-42755741`:diagnosis  0.376980   0.051524   7.317 3.11e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.4658 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2799,	Adjusted R-squared:  0.2497 
# F-statistic: 9.258 on 11 and 262 DF,  p-value: 5.227e-14


min(merge_v3$TC09000560.hg.1) #7.226324
max(merge_v3$TC09000560.hg.1) #11.38889
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-42755741'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC09000560.hg.1, x = as.factor(x$`AX-42755741`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="UGCG expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("UGCG")," " , "expression")))+expand_limits(y=c(7,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-42755741`), y=x$TC09000560.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="UGCG expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ expand_limits(y=c(7,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-42755741`, y=x$TC09000560.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(7,12))

p2 <-p1 + labs(y="UGCG expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC09000560.hg.1", groupnames =c("diagnosis","`AX-42755741`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-42755741` <- as.factor(df$`AX-42755741`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-42755741`), y=df$TC09000560.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC09000560.hg.1-sd, ymax=TC09000560.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(7,12))

p2 <-p1 + labs(y="UGCG expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-42755741`), y=df$TC09000560.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC09000560.hg.1-sd, ymax=TC09000560.hg.1+sd),width=.2)+ expand_limits(y=c(7,12))

p2 <-p1 + labs(y="UGCG expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL26_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT26_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#27	AX-38679391	TC03002054.hg.1	27	7.318023286	3.06E-12	0.000600007	0.372996531	LAMP3 	chr3	182840001	182881627	Coding	rs139929471	X	88063578	A	G	G	A


x<-merge_v3$`AX-38679391`

# 0   1   2  NA
# 14  11 249 1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC03002054.hg.1~`AX-38679391`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38679391`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC03002054.hg.1 ~ `AX-38679391` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38679391` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -2.3498 -0.1536 -0.0568  0.0800  5.1013 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.1000369  0.4157840  17.076  < 2e-16 ***
# `AX-38679391`           -1.5706367  0.1819787  -8.631 6.00e-16 ***
# age                     -0.0005357  0.0023423  -0.229   0.8193    
# race                     0.0625690  0.0343142   1.823   0.0694 .  
# gender                   0.0240045  0.0607441   0.395   0.6930    
# diagnosis               -0.7595918  0.1115677  -6.808 6.72e-11 ***
# diagnosissubtype         0.0185792  0.0332548   0.559   0.5768    
# pc1                      1.0455820  0.6113444   1.710   0.0884 .  
# pc2                      0.7702245  0.5312737   1.450   0.1483    
# pc3                     -1.4332722  0.6631999  -2.161   0.0316 *  
# pc4                      0.1223175  0.4811741   0.254   0.7995    
# `AX-38679391`:diagnosis  0.3730560  0.0507823   7.346 2.59e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.4734 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2365,	Adjusted R-squared:  0.2045 
# F-statistic: 7.379 on 11 and 262 DF,  p-value: 5.333e-11


min(merge_v3$TC03002054.hg.1) #4.017943
max(merge_v3$TC03002054.hg.1) #11.78421
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-38679391'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC03002054.hg.1, x = as.factor(x$`AX-38679391`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="LAMP3 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("LAMP3")," " , "expression")))+expand_limits(y=c(4,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38679391`), y=x$TC03002054.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="LAMP3 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("LAMP3")," " , "expression")))+ expand_limits(y=c(4,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38679391`, y=x$TC03002054.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,12))

p2 <-p1 + labs(y="LAMP3 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("LAMP3")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC03002054.hg.1", groupnames =c("diagnosis","`AX-38679391`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38679391` <- as.factor(df$`AX-38679391`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38679391`), y=df$TC03002054.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC03002054.hg.1-sd, ymax=TC03002054.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,12))

p2 <-p1 + labs(y="LAMP3 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("LAMP3")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38679391`), y=df$TC03002054.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC03002054.hg.1-sd, ymax=TC03002054.hg.1+sd),width=.2)+ expand_limits(y=c(4,12))

p2 <-p1 + labs(y="LAMP3 expression", x = "rs139929471 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("LAMP3")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL27_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT27_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#28 AX-16372292	TC02000318.hg.1	28	-7.288652883	3.66E-12	0.000693752	-0.225949853	EML6 	chr2	54950636	55199157	Coding	rs2369519	X	86392534	A	G	G	A


x<-merge_v3$`AX-16372292`

# 0   1   2  NA
# 248  18   7 2

#2
#UCDSS.147	7	1	3	55	0	0	0	0.135238	-0.0306445	-0.0170027	-0.0934562	4
#UCDSS.448	7	1	3	59	0	0	0	0.160812	-0.0388213	-0.0185052	-0.061743	4
#UCDSS.513	8	1	2	71	1	1	1	-0.01459	0.0134688	-0.0145828	-0.00139512	4
#UCDSS.517	8	2	3	45	1	0	0	0.126155	-0.0357504	0.025814	0.649375	4
#UCDSS.519	7	1	3	43	0	0	0	0.15934	-0.0349643	-0.0133984	-0.0445161	4
#UCDSS.840	4	1	5	60	0	0	1	0.166113	-0.0373338	0.00370748	-0.0316622	1
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC02000318.hg.1~`AX-16372292`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372292`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC02000318.hg.1 ~ `AX-16372292` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372292` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.25090 -0.12212 -0.02917  0.09869  3.15800 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.8148239  0.1448810  33.233  < 2e-16 ***
# `AX-16372292`            0.9586578  0.1041770   9.202  < 2e-16 ***
# age                     -0.0007532  0.0013776  -0.547   0.5850    
# race                    -0.0022768  0.0207832  -0.110   0.9129    
# gender                  -0.0504918  0.0365689  -1.381   0.1685    
# diagnosis                0.0448832  0.0353114   1.271   0.2048    
# diagnosissubtype        -0.0264829  0.0200639  -1.320   0.1880    
# pc1                     -0.5823102  0.3863920  -1.507   0.1330    
# pc2                      0.1705069  0.3277615   0.520   0.6034    
# pc3                      0.0176354  0.3630056   0.049   0.9613    
# pc4                     -0.5071238  0.2957511  -1.715   0.0876 .  
# `AX-16372292`:diagnosis -0.2259134  0.0310916  -7.266 4.28e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2852 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.263,	Adjusted R-squared:  0.2319 
# F-statistic: 8.466 on 11 and 261 DF,  p-value: 9.555e-13


min(merge_v3$TC02000318.hg.1) #4.35876
max(merge_v3$TC02000318.hg.1) #9.330151
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372292'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC02000318.hg.1, x = as.factor(x$`AX-16372292`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="EML6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28.jpg",width = 6, height = 4)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372292`), y=x$TC02000318.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="EML6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372292`, y=x$TC02000318.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC02000318.hg.1", groupnames =c("diagnosis","`AX-16372292`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372292` <- as.factor(df$`AX-16372292`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372292`), y=df$TC02000318.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372292`), y=df$TC02000318.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd),width=.2)+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL28_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT28_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#29 AX-33017431	TC17001827.hg.1	29	7.272873901	4.04E-12	0.00073831	0.295083636	ABCA6 	chr17	67074847	67138015	Coding	rs17409498	20	56044855	T	C	C	T


x<-merge_v3$`AX-33017431`

# 0   1   2  NA
#  6  24 244 1
#UCDSS.283	8	2	5	88	1	0	0	-0.0303396	-0.0286512	-0.0267102	-0.0159477	4
#UCDSS.344	8	2	5	73	1	0	1	-0.033754	-0.0290867	-0.0153624	0.00425484	4
#UCDSS.590	8	2	5	66	1	0	0	-0.0322412	-0.0313669	-0.017775	-0.00106227	4
#UCDSS.871	8	1	5	58	0	0	1	-0.0335037	-0.0262638	-0.0132783	0.00269816	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1
#UofASS.008	8	1	5	67	1	0	1	-0.0343674	-0.0322281	-0.0130704	0.00195825	4

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17001827.hg.1~`AX-33017431`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-33017431`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17001827.hg.1 ~ `AX-33017431` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-33017431` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -0.8940 -0.0973 -0.0149  0.0920  4.9775 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.199050   0.298957  20.736  < 2e-16 ***
# `AX-33017431`           -1.204024   0.131784  -9.136  < 2e-16 ***
# age                      0.003066   0.001906   1.608    0.109    
# race                     0.042844   0.027784   1.542    0.124    
# gender                  -0.038806   0.049254  -0.788    0.431    
# diagnosis               -0.506267   0.088659  -5.710 3.05e-08 ***
# diagnosissubtype        -0.040169   0.026969  -1.489    0.138    
# pc1                      0.303107   0.494467   0.613    0.540    
# pc2                      0.387930   0.437079   0.888    0.376    
# pc3                     -0.090278   0.487463  -0.185    0.853    
# pc4                      0.015774   0.389396   0.041    0.968    
# `AX-33017431`:diagnosis  0.295314   0.040639   7.267 4.23e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3836 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2641,	Adjusted R-squared:  0.2332 
# F-statistic: 8.548 on 11 and 262 DF,  p-value: 6.953e-13


min(merge_v3$TC17001827.hg.1) #3.87316
max(merge_v3$TC17001827.hg.1) #10.95106
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-33017431'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17001827.hg.1, x = as.factor(x$`AX-33017431`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs17409498 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(3,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-33017431`), y=x$TC17001827.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs17409498 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(3,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-33017431`, y=x$TC17001827.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,10))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs17409498 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC17001827.hg.1", groupnames =c("diagnosis","`AX-33017431`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-33017431` <- as.factor(df$`AX-33017431`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-33017431`), y=df$TC17001827.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs17409498 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-33017431`), y=df$TC17001827.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd),width=.2)+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs17409498 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL29_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT29_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#30 8	AX-11393384	TC02000318.hg.1	30	-7.263330564	4.28E-12	0.00075693	-0.257966962	EML6 	chr2	54950636	55199157	Coding	rs2464504	4	48232441	T	C	C	T


x<-merge_v3$`AX-11393384`

# 0   1   2  
# 250  22   3

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC02000318.hg.1~`AX-11393384`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11393384`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC02000318.hg.1 ~ `AX-11393384` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11393384` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.95800 -0.10790 -0.02155  0.10829  3.04065 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.635e+00  1.436e-01  32.266  < 2e-16 ***
# `AX-11393384`            1.068e+00  1.153e-01   9.262  < 2e-16 ***
# age                     -5.371e-05  1.368e-03  -0.039   0.9687    
# race                     1.936e-02  2.040e-02   0.949   0.3434    
# gender                  -2.490e-02  3.621e-02  -0.687   0.4924    
# diagnosis                5.718e-02  3.498e-02   1.635   0.1033    
# diagnosissubtype        -3.173e-02  1.988e-02  -1.596   0.1116    
# pc1                      7.549e-01  3.660e-01   2.063   0.0401 *  
# pc2                      2.101e-01  3.207e-01   0.655   0.5130    
# pc3                     -2.384e-01  3.606e-01  -0.661   0.5090    
# pc4                     -9.395e-02  2.876e-01  -0.327   0.7442    
# `AX-11393384`:diagnosis -2.580e-01  3.552e-02  -7.263 4.28e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2834 on 263 degrees of freedom
# Multiple R-squared:  0.2676,	Adjusted R-squared:  0.237 
# F-statistic: 8.738 on 11 and 263 DF,  p-value: 3.41e-13 


min(merge_v3$TC02000318.hg.1) #4.35876
max(merge_v3$TC02000318.hg.1) #9.330151
```


```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-33017431'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC02000318.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="EML6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11393384`), y=merge_v3$TC02000318.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="EML6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11393384`, y=merge_v3$TC02000318.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC02000318.hg.1", groupnames =c("diagnosis","`AX-11393384`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11393384` <- as.factor(df$`AX-11393384`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11393384`), y=df$TC02000318.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11393384`), y=df$TC02000318.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd),width=.2)+ expand_limits(y=c(3,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL30_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT30_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#No Figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#31 AX-33587459	TC01002971.hg.1	32	-7.225632115	5.40E-12	0.000894767	-0.260834444	SLC16A4 	chr1	110905470	110933704	Coding	rs79434685	22	29556745	C	G	C	G


x<-merge_v3$`AX-33587459`

# 0   1   2  
# 246  28   1

```

```{r}
#No Figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#32	AX-35735671	TC18000998.hg.1	31	7.227165395	5.35E-12	0.000894767	0.242620608	SMAD4 	chr18	48556583	48611415	Coding	rs1051785	6	31378388	A	G	G	A


x<-merge_v3$`AX-35735671`
# 0   1   2  
# 1  32 240  

```

```{r}
#No Figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#33	AX-33587459	TC17001827.hg.1	33	-7.214935197	5.77E-12	0.000926518	-0.353326338	ABCA6 	chr17	67074847	67138015	Coding	rs79434685	22	29556745	C	G	C	G


x<-merge_v3$`AX-33587459`
# 0   1   2  
#  246  28   1 

```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#34	AX-41232349	TC15001756.hg.1	35	7.190968964	6.68E-12	0.001011753	0.118990642	AP3B2 	chr15	83328033	83378666	Coding	rs6787784	3	40486470	T	C	T	C


x<-merge_v3$`AX-41232349`
# 0   1   2  NA 
# 241  28   4 2
#UCDSS.357	3	1	3	70	1	0	0	0.162101	-0.0330967	-0.0111911	-0.0358062	1
#UCDSS.708	8	2	3	41	1	1	1	0.161344	-0.0350806	0.0073334	-0.0420488	4
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.050	3	2	3	59	1	0	1	0.170806	-0.0337208	-0.0114983	-0.042197	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-41232349`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-41232349`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-41232349` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-41232349` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.41333 -0.08266 -0.02429  0.05795  0.80188 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.1471257  0.0765731  54.159  < 2e-16 ***
# `AX-41232349`           -0.2030628  0.0487037  -4.169 4.16e-05 ***
# age                      0.0011819  0.0007324   1.614    0.108    
# race                     0.0176666  0.0110318   1.601    0.110    
# gender                   0.0096247  0.0190975   0.504    0.615    
# diagnosis                0.0053399  0.0183721   0.291    0.772    
# diagnosissubtype        -0.0107183  0.0104484  -1.026    0.306    
# pc1                      0.2842155  0.2188410   1.299    0.195    
# pc2                     -0.1193391  0.1683616  -0.709    0.479    
# pc3                     -0.0276193  0.1926792  -0.143    0.886    
# pc4                     -0.0917620  0.1535102  -0.598    0.551    
# `AX-41232349`:diagnosis  0.1189510  0.0165977   7.167 7.87e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1491 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2018,	Adjusted R-squared:  0.1682 
#F-statistic:     6 on 11 and 261 DF,  p-value: 9.973e-09 


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```


```{r}
x <-merge_v3[!is.na(merge_v3$'AX-41232349'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-41232349`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-41232349`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-41232349`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-41232349`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-41232349` <- as.factor(df$`AX-41232349`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-41232349`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-41232349`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="EML6 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL34_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT34_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#35 AX-51270082	TC0M000010.hg.1	34	-7.192006904	6.64E-12	0.001011753	-0.440890362	---	chrM	14857	15888	Coding	rs9779183	X	13009957	T	C	T	C


x<-merge_v3$`AX-51270082`
# 0   1   2  NA 
# 258   8   7 2

#2
#UCDSS.424	1	2	3	67	1	0	0	0.161859	-0.035007	0.0250816	0.712585	1
#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.517	8	2	3	45	1	0	0	0.126155	-0.0357504	0.025814	0.649375	4
#UCDSS.845	3	1	3	56	0	1	0	0.131643	-0.0324731	-0.00850289	-0.023962	1
#UCSFCJ.017	2	1	3	63	1	0	0	0.163721	-0.0349639	-0.0122536	-0.0369332	1
#UCSFCJ.101	2	1	5	46	1	0	1	-0.031927	-0.0306845	-0.0165275	0.00071694	1
#UCSFCJ.109	4	1	2	53	1	1	1	-0.0323482	-0.0282733	-0.0178268	0.00268296	1 (HT1)

#1
#UCDSS.201	8	2	5	72	1	0	0	-0.0303633	-0.0330236	-0.0209832	-0.00285422	4
#UCDSS.281	8	2	5	64	1	1	0	-0.0328373	-0.0284304	-0.0233561	-0.00423923	4
#UCDSS.510	8	2	5	63	1	0	1	-0.0295969	-0.0287967	-0.0210884	-0.000399126	4
#UCDSS.525	4	2	5	60	0	0	1	-0.0319856	-0.0308957	-0.0116315	0.00586204	1
#UCDSS.555	8	2	5	85	1	0	1	-0.0315911	-0.0276138	-0.0168431	7.76E-06	4
#UCSFCJ.005	3	2	3	53	1	1	0	0.102322	-0.0350191	-0.0147273	-0.0238308	1
#UCSFCJ.091	6	2	5	43	0	0	0	0.00300048	0.0634869	0.136817	-0.00680118	1
#UCSFCJ.124	2	2	5	83	0	0	0	-0.031664	-0.0305188	-0.0216808	-0.00661383	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC0M000010.hg.1~`AX-51270082`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-51270082`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC0M000010.hg.1 ~ `AX-51270082` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-51270082` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -4.4093 -0.0234  0.0851  0.1674  1.5332 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             19.829303   0.259239  76.490  < 2e-16 ***
# `AX-51270082`            0.336480   0.152922   2.200 0.028659 *  
# age                     -0.001289   0.002453  -0.525 0.599780    
# race                    -0.039604   0.036641  -1.081 0.280756    
# gender                   0.021648   0.063707   0.340 0.734272    
# diagnosis               -0.020075   0.061286  -0.328 0.743500    
# diagnosissubtype         0.040650   0.035034   1.160 0.246979    
# pc1                      0.298113   0.640029   0.466 0.641761    
# pc2                     -0.187816   0.579161  -0.324 0.745978    
# pc3                      0.230992   0.643516   0.359 0.719921    
# pc4                      2.119842   0.568535   3.729 0.000236 ***
# `AX-51270082`:diagnosis -0.440318   0.061546  -7.154 8.49e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.4952 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2454,	Adjusted R-squared:  0.2136 
# F-statistic: 7.718 on 11 and 261 DF,  p-value: 1.523e-11 


min(merge_v3$TC0M000010.hg.1) #12.51037
max(merge_v3$TC0M000010.hg.1) #19.92878
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-51270082'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC0M000010.hg.1, x = as.factor(x$`AX-51270082`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("TC0M000010.hg.1")," " , "expression")))+expand_limits(y=c(12,20))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-51270082`), y=x$TC0M000010.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ expand_limits(y=c(12,20))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-51270082`, y=x$TC0M000010.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC0M000010.hg.1", groupnames =c("diagnosis","`AX-51270082`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-51270082` <- as.factor(df$`AX-51270082`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-51270082`), y=df$TC0M000010.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC0M000010.hg.1-sd, ymax=TC0M000010.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-51270082`), y=df$TC0M000010.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC0M000010.hg.1-sd, ymax=TC0M000010.hg.1+sd),width=.2)+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL35_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT35_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#36 AX-38475535	TC04001036.hg.1	36	7.183896995	6.97E-12	0.001027148	0.201570749	CLNK 	chr4	10488019	10686489	Coding	rs117781420	9	19355687	A	G	G	A

x<-merge_v3$`AX-38475535`
# 0   1   2  NA 
#  2  27 245  1

#UCDSS.196	8	2	5	56	1	0	1	-0.0311587	-0.0281442	-0.0114177	0.000141421	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001036.hg.1~`AX-38475535`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38475535`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001036.hg.1 ~ `AX-38475535` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38475535` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.62212 -0.09098 -0.01516  0.08707  2.63753 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              5.6081069  0.1959134  28.625  < 2e-16 ***
# `AX-38475535`           -0.7490268  0.0868504  -8.624 6.27e-16 ***
# age                      0.0003157  0.0011233   0.281   0.7789    
# race                     0.0157065  0.0165858   0.947   0.3445    
# gender                   0.0036660  0.0296837   0.124   0.9018    
# diagnosis               -0.3926068  0.0621987  -6.312 1.17e-09 ***
# diagnosissubtype         0.0034858  0.0162388   0.215   0.8302    
# pc1                      0.6928094  0.2974129   2.329   0.0206 *  
# pc2                      0.0163752  0.2612905   0.063   0.9501    
# pc3                      0.0051602  0.2936628   0.018   0.9860    
# pc4                      0.1354214  0.2338957   0.579   0.5631    
# `AX-38475535`:diagnosis  0.2011943  0.0280437   7.174 7.45e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2305 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2382,	Adjusted R-squared:  0.2062 
# F-statistic: 7.448 on 11 and 262 DF,  p-value: 4.115e-11

min(merge_v3$TC04001036.hg.1) #4.00259
max(merge_v3$TC04001036.hg.1) #7.940702
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-38475535'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC04001036.hg.1, x = as.factor(x$`AX-38475535`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CLNK expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38475535`), y=x$TC04001036.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="CLNK expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38475535`, y=x$TC04001036.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC04001036.hg.1", groupnames =c("diagnosis","`AX-38475535`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38475535` <- as.factor(df$`AX-38475535`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38475535`), y=df$TC04001036.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38475535`), y=df$TC04001036.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL36_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT36_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#37 AX-38692681	TC04000912.hg.1	37	-7.176935102	7.28E-12	0.001042852	-0.351340598	TLR3 	chr4	186990306	187006255	Coding	rs140580619	X	98022257	T	C	C	T

x<-merge_v3$`AX-38692681`
# 0   1   2  
# 7  12 256 

#UCDSS.089	2	1	5	54	1	0	1	-0.0264045	-0.00915837	-0.0208476	0.00082716	1
#UCDSS.313	4	1	5	50	0	1	1	-0.0330535	-0.0304682	-0.0169296	-5.32E-05	1
#UCDSS.330	1	1	5	83	1	0	1	-0.0329765	-0.0297038	-0.0239384	0.000434691	1
#UCDSS.563	1	1	5	53	1	1	1	-0.0320841	-0.0291427	-0.0214086	-0.00452532	1
#UCDSS.601	4	1	5	54	1	0	1	-0.0326527	-0.0289111	-0.0207153	1.10E-05	1
#UCDSS.709	3	1	6	55	1	1	1	-0.00616543	0.0499376	0.114078	0.00200624	1
#UCDSS.847	8	1	5	56	0	0	0	-0.0327132	-0.0285476	-0.0173752	0.00250807	4

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04000912.hg.1~`AX-38692681`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38692681`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04000912.hg.1 ~ `AX-38692681` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38692681` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.42916 -0.18460 -0.03315  0.15147  3.14107 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.101871   0.276151  14.854  < 2e-16 ***
# `AX-38692681`            0.271049   0.107833   2.514   0.0125 *  
# age                      0.003906   0.001881   2.077   0.0388 *  
# race                    -0.006828   0.027951  -0.244   0.8072    
# gender                  -0.060151   0.049661  -1.211   0.2269    
# diagnosis                0.664458   0.106364   6.247 1.67e-09 ***
# diagnosissubtype         0.028309   0.027260   1.038   0.3000    
# pc1                      0.777591   0.500878   1.552   0.1218    
# pc2                      0.383119   0.437820   0.875   0.3823    
# pc3                      0.567047   0.493585   1.149   0.2517    
# pc4                     -0.465644   0.394124  -1.181   0.2385    
# `AX-38692681`:diagnosis -0.351341   0.048954  -7.177 7.28e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3884 on 263 degrees of freedom
# Multiple R-squared:  0.2538,	Adjusted R-squared:  0.2226 
# F-statistic: 8.132 on 11 and 263 DF,  p-value: 3.181e-12

min(merge_v3$TC04000912.hg.1) #4.182154
max(merge_v3$TC04000912.hg.1) #10.20431
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-38692681'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC04000912.hg.1, x = as.factor(merge_v3$`AX-38692681`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="TLR3 expression", x = "rs140580619 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("TLR3")," " , "expression")))+expand_limits(y=c(4,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-38692681`), y=merge_v3$TC04000912.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="TLR3 expression", x = "rs140580619 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("TLR3")," " , "expression")))+ expand_limits(y=c(4,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-38692681`, y=merge_v3$TC04000912.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,11))

p2 <-p1 + labs(y="TLR3 expression", x = "rs140580619 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TLR3")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC04000912.hg.1", groupnames =c("diagnosis","`AX-38692681`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38692681` <- as.factor(df$`AX-38692681`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38692681`), y=df$TC04000912.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04000912.hg.1-sd, ymax=TC04000912.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,11))

p2 <-p1 + labs(y="TLR3 expression", x = "rs140580619 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TLR3")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38692681`), y=df$TC04000912.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04000912.hg.1-sd, ymax=TC04000912.hg.1+sd),width=.2)+ expand_limits(y=c(4,11))

p2 <-p1 + labs(y="TLR3 expression", x = "rs140580619 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TLR3")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL37_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT37_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#38 AX-37250277	TC09000899.hg.1	38	7.159985121	8.07E-12	0.001126179	0.210051969	GLDC 	chr9	6532464	6645692	Coding	rs73178117	X	3466525	T	C	T	C


x<-merge_v3$`AX-37250277`
# 0   1   2  NA 
# 252  15   7 1

#UCDSS.357	3	1	3	70	1	0	0	0.162101	-0.0330967	-0.0111911	-0.0358062	1
#UCDSS.463	8	1	5	70	1	0	0	-0.023	-0.0217369	-0.010371	0.0020478	4
#UCDSS.621	4	1	5	73	1	0	1	-0.0331423	-0.0305817	-0.0200513	0.00038171	1
#UCDSS.672	4	2	5	69	1	0	1	-0.0318989	-0.0291652	-0.0182134	0.00128585	1
#UCDSS.774	6	1	5	58	0	0	1	-0.0294796	-0.0289014	-0.0121061	-0.000155022	1
#UCSFCJ.007	4	1	5	49	0	0	0	-0.030481	-0.0272547	-0.0140486	0.00257031	1
#UCSFCJ.072	3	1	5	73	1	0	1	-0.0324272	-0.0284671	-0.0230238	0.000460096	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC09000899.hg.1~`AX-37250277`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-37250277`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC09000899.hg.1 ~ `AX-37250277` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-37250277` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.70791 -0.11764 -0.04192  0.06256  1.83641 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.9423796  0.1197720  32.916  < 2e-16 ***
# `AX-37250277`           -0.2602532  0.0641031  -4.060 6.49e-05 ***
# age                      0.0005549  0.0011526   0.481  0.63059    
# race                     0.0242727  0.0169936   1.428  0.15438    
# gender                  -0.0815469  0.0301853  -2.702  0.00735 ** 
# diagnosis               -0.0648931  0.0296480  -2.189  0.02950 *  
# diagnosissubtype         0.0358933  0.0167588   2.142  0.03313 *  
# pc1                      0.6679458  0.3025164   2.208  0.02811 *  
# pc2                      0.4035660  0.2665328   1.514  0.13120    
# pc3                     -0.2987556  0.3007883  -0.993  0.32151    
# pc4                     -0.0075466  0.2396215  -0.031  0.97490    
# `AX-37250277`:diagnosis  0.2103243  0.0293745   7.160 8.13e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2361 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2147,	Adjusted R-squared:  0.1817 
# F-statistic:  6.51 on 11 and 262 DF,  p-value: 1.415e-09

min(merge_v3$TC09000899.hg.1) #3.751397
max(merge_v3$TC09000899.hg.1) #7.026065
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-37250277'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC09000899.hg.1, x = as.factor(x$`AX-37250277`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="GLDC expression", x = "rs73178117 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("GLDC")," " , "expression")))+expand_limits(y=c(3,7))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-37250277`), y=x$TC09000899.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="GLDC expression", x = "rs73178117 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("GLDC")," " , "expression")))+ expand_limits(y=c(3,7))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-37250277`, y=merge_v3$TC09000899.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="GLDC expression", x = "rs73178117 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("GLDC")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC09000899.hg.1", groupnames =c("diagnosis","`AX-37250277`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-37250277` <- as.factor(df$`AX-37250277`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-37250277`), y=df$TC09000899.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC09000899.hg.1-sd, ymax=TC09000899.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="GLDC expression", x = "rs73178117 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("GLDC")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-37250277`), y=df$TC09000899.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC09000899.hg.1-sd, ymax=TC09000899.hg.1+sd),width=.2)+ expand_limits(y=c(3,7))

p2 <-p1 + labs(y="GLDC expression", x = "rs73178117 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("GLDC")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL38_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT38_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#39 AX-37092389	TC11001026.hg.1	39	-7.145100983	8.84E-12	0.00120159	-0.180661294	ZBTB16 	chr11	113930315	114121398	Coding	rs1172922	9	93488534	A	C	A	C


x<-merge_v3$`AX-37092389`
# 0   1   2  
#  3  48 224

#UCDSS.498	8	2	5	69	1	0	1	-0.0329975	-0.0310356	-0.019201	0.00256623	4
#UofASS.026	4	2	5	72	1	1	0	-0.0322463	-0.0265595	-0.0157869	0.0033596	1
#UofASS.031	8	2	5	61	1	1	0	-0.0338622	-0.0285684	-0.0157328	0.00646103	4

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC11001026.hg.1~`AX-37092389`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-37092389`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC11001026.hg.1 ~ `AX-37092389` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-37092389` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.58323 -0.14518 -0.04196  0.10966  1.53320 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              5.3907346  0.1861077  28.966  < 2e-16 ***
# `AX-37092389`            0.3577859  0.0765835   4.672 4.77e-06 ***
# age                      0.0008718  0.0012591   0.692   0.4893    
# race                     0.0000209  0.0187157   0.001   0.9991    
# gender                   0.0016760  0.0331455   0.051   0.9597    
# diagnosis                0.2813769  0.0520685   5.404 1.46e-07 ***
# diagnosissubtype         0.0357309  0.0185159   1.930   0.0547 .  
# pc1                     -0.6506621  0.3340694  -1.948   0.0525 .  
# pc2                      0.0821443  0.2944529   0.279   0.7805    
# pc3                     -0.4325894  0.3305684  -1.309   0.1918    
# pc4                      0.0149139  0.2631209   0.057   0.9548    
# `AX-37092389`:diagnosis -0.1806613  0.0252846  -7.145 8.84e-12 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2593 on 263 degrees of freedom
# Multiple R-squared:  0.2262,	Adjusted R-squared:  0.1939 
# F-statistic:  6.99 on 11 and 263 DF,  p-value: 2.278e-10

min(merge_v3$TC11001026.hg.1) #5.704674
max(merge_v3$TC11001026.hg.1) #8.418521
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-37092389'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC11001026.hg.1, x = as.factor(merge_v3$`AX-37092389`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZBTB16 expression", x = "rs1172922 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AC", "2"="CC"))+scale_y_continuous(expression(paste( italic("ZBTB16")," " , "expression")))+expand_limits(y=c(5,9))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-37092389`), y=merge_v3$TC11001026.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ZBTB16 expression", x = "rs1172922 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AC", "CC"))+scale_y_continuous(expression(paste(italic("ZBTB16")," " , "expression")))+ expand_limits(y=c(5,9))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-37092389`, y=merge_v3$TC11001026.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5,9))

p2 <-p1 + labs(y="ZBTB16 expression", x = "rs1172922 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZBTB16")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC11001026.hg.1", groupnames =c("diagnosis","`AX-37092389`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-37092389` <- as.factor(df$`AX-37092389`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-37092389`), y=df$TC11001026.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC11001026.hg.1-sd, ymax=TC11001026.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5,9))

p2 <-p1 + labs(y="ZBTB16 expression", x = "rs1172922 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZBTB16")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-37092389`), y=df$TC11001026.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC11001026.hg.1-sd, ymax=TC11001026.hg.1+sd),width=.2)+ expand_limits(y=c(5,9))

p2 <-p1 + labs(y="ZBTB16 expression", x = "rs1172922 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZBTB16")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL39_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT39_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#40 AX-11393384	TC01002971.hg.1	40	-7.11298277	1.07E-11	0.001420772	-0.24903497	SLC16A4 	chr1	110905470	110933704	Coding	rs2464504	4	48232441	T	C	C	T


x<-merge_v3$`AX-11393384`
# 0   1   2  
# 250  22   3 

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01002971.hg.1~`AX-11393384`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11393384`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01002971.hg.1 ~ `AX-11393384` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11393384` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.01341 -0.10333 -0.01041  0.09680  2.88314 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              5.140930   0.141595  36.307  < 2e-16 ***
# `AX-11393384`            1.016677   0.113667   8.944  < 2e-16 ***
# age                      0.001561   0.001349   1.157    0.248    
# race                     0.003242   0.020112   0.161    0.872    
# gender                  -0.028071   0.035698  -0.786    0.432    
# diagnosis                0.028183   0.034481   0.817    0.414    
# diagnosissubtype        -0.019356   0.019593  -0.988    0.324    
# pc1                     -0.188264   0.360772  -0.522    0.602    
# pc2                     -0.199292   0.316171  -0.630    0.529    
# pc3                     -0.077106   0.355450  -0.217    0.828    
# pc4                     -0.137339   0.283488  -0.484    0.628    
# `AX-11393384`:diagnosis -0.249035   0.035011  -7.113 1.07e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2794 on 263 degrees of freedom
# Multiple R-squared:  0.2647,	Adjusted R-squared:  0.2339 
#F-statistic: 8.606 on 11 and 263 DF,  p-value: 5.534e-13

min(merge_v3$TC01002971.hg.1) #4.827518
max(merge_v3$TC01002971.hg.1) #9.654251
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-37092389'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC01002971.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SLC16A4 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("SLC16A4")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11393384`), y=merge_v3$TC01002971.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11393384`, y=merge_v3$TC01002971.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC01002971.hg.1", groupnames =c("diagnosis","`AX-11393384`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11393384` <- as.factor(df$`AX-11393384`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11393384`), y=df$TC01002971.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11393384`), y=df$TC01002971.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd),width=.2)+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL40_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT40_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#41 AX-30210493	TC15001756.hg.1	41	-7.10935283	1.10E-11	0.001420772	-0.119577645	AP3B2 	chr15	83328033	83378666	Coding	rs72906031	1	16845719	T	G	G	T

x<-merge_v3$`AX-30210493`
# 0   1   2  
# 4  20 251 

#UCDSS.517	8	2	3	45	1	0	0	0.126155	-0.0357504	0.025814	0.649375	4
#UCDSS.693	8	2	3	48	1	0	0	0.141	-0.0354487	-0.0212148	-0.0503794	4
#UCDSS.798	3	2	3	34	1	1	1	0.131794	-0.0329576	-0.00276457	-0.0250158	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-30210493`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-30210493`*diagnosis, data = merge_v3)
summary (fit)
#Call:
# lm(formula = TC15001756.hg.1 ~ `AX-30210493` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-30210493` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.43619 -0.07603 -0.02240  0.05197  0.56153 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.1031760  0.1196220  34.301  < 2e-16 ***
# `AX-30210493`            0.0585903  0.0541935   1.081    0.281    
# age                      0.0010914  0.0006757   1.615    0.107    
# race                     0.0029931  0.0101167   0.296    0.768    
# gender                  -0.0015622  0.0178419  -0.088    0.930    
# diagnosis                0.2584587  0.0362490   7.130 9.68e-12 ***
# diagnosissubtype        -0.0159198  0.0097609  -1.631    0.104    
# pc1                     -0.3436615  0.2086108  -1.647    0.101    
# pc2                     -0.0008821  0.1594713  -0.006    0.996    
# pc3                      0.0194432  0.1779316   0.109    0.913    
# pc4                     -0.6048333  0.1478354  -4.091 5.71e-05 ***
# `AX-30210493`:diagnosis -0.1195776  0.0168198  -7.109 1.10e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1392 on 263 degrees of freedom
# Multiple R-squared:  0.3008,	Adjusted R-squared:  0.2716 
# F-statistic: 10.29 on 11 and 263 DF,  p-value: 1.281e-15


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-37092389'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC15001756.hg.1, x = as.factor(merge_v3$`AX-30210493`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs72906031 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GT", "2"="TT"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-30210493`), y=merge_v3$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72906031 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GT", "TT"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-30210493`, y=merge_v3$TC15001756.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72906031 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-30210493`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-30210493` <- as.factor(df$`AX-30210493`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-30210493`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72906031 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-30210493`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs72906031 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL41_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#42 AX-12721174	TC15001756.hg.1	43	-7.099941191	1.16E-11	0.001434384	-0.125908312	AP3B2 	chr15	83328033	83378666	Coding	rs57764234	14	21897616	T	C	C	T


x<-merge_v3$`AX-12721174`
# 0   1   2  
# 3  29 243 

#UCDSS.683	3	2	3	51	1	0	0	0.122311	-0.0234439	0.00907122	-0.0285038	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.005	3	2	3	53	1	1	0	0.102322	-0.0350191	-0.0147273	-0.0238308	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-12721174`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-12721174`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-12721174` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-12721174` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.52027 -0.07824 -0.02285  0.04939  0.66802 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.9474715  0.1100123  35.882  < 2e-16 ***
# `AX-12721174`            0.1303205  0.0462802   2.816  0.00523 ** 
# age                      0.0006828  0.0007188   0.950  0.34305    
# race                     0.0082104  0.0109025   0.753  0.45208    
# gender                   0.0125123  0.0190147   0.658  0.51109    
# diagnosis                0.2617553  0.0378009   6.925 3.34e-11 ***
# diagnosissubtype        -0.0121164  0.0104158  -1.163  0.24577    
# pc1                     -0.1869675  0.2144229  -0.872  0.38403    
# pc2                      0.0529952  0.1738344   0.305  0.76071    
# pc3                     -0.0071032  0.1891839  -0.038  0.97008    
# pc4                     -0.2842261  0.1519011  -1.871  0.06244 .  
# `AX-12721174`:diagnosis -0.1259083  0.0177337  -7.100 1.16e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1485 on 263 degrees of freedom
# Multiple R-squared:  0.2034,	Adjusted R-squared:  0.1701 
# F-statistic: 6.106 on 11 and 263 DF,  p-value: 6.538e-09


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-37092389'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC15001756.hg.1, x = as.factor(merge_v3$`AX-12721174`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs57764234 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-12721174`), y=merge_v3$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs57764234 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-12721174`, y=merge_v3$TC15001756.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs57764234 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-12721174`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-12721174` <- as.factor(df$`AX-12721174`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-12721174`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs57764234 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-12721174`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs57764234 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL42_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#43	AX-41232325	TC15001756.hg.1	42	7.103752479	1.14E-11	0.001434384	0.111598105	AP3B2 	chr15	83328033	83378666	Coding	rs9873394	3	40468206	T	G	T	G


x<-merge_v3$`AX-41232325`
# 0   1   2  
# 237  33   5 

#UCDSS.357	3	1	3	70	1	0	0	0.162101	-0.0330967	-0.0111911	-0.0358062	1
#UCDSS.571	2	1	3	69	1	0	0	0.18122	-0.034571	-0.0165888	-0.102848	1
#UCDSS.708	8	2	3	41	1	1	1	0.161344	-0.0350806	0.0073334	-0.0420488	4
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.050	3	2	3	59	1	0	1	0.170806	-0.0337208	-0.0114983	-0.042197	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-41232325`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-41232325`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-41232325` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-41232325` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.39414 -0.08005 -0.01678  0.05805  0.82507 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.1547900  0.0763988  54.383  < 2e-16 ***
# `AX-41232325`           -0.1914060  0.0453473  -4.221 3.35e-05 ***
# age                      0.0011982  0.0007306   1.640    0.102    
# race                     0.0168128  0.0110439   1.522    0.129    
# gender                   0.0069802  0.0190846   0.366    0.715    
# diagnosis                0.0029784  0.0185178   0.161    0.872    
# diagnosissubtype        -0.0095374  0.0104816  -0.910    0.364    
# pc1                      0.3491132  0.2302465   1.516    0.131    
# pc2                     -0.1548149  0.1681104  -0.921    0.358    
# pc3                     -0.0650343  0.1918261  -0.339    0.735    
# pc4                     -0.0474735  0.1525675  -0.311    0.756    
# `AX-41232325`:diagnosis  0.1115981  0.0157097   7.104 1.14e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1495 on 263 degrees of freedom
# Multiple R-squared:  0.1935,	Adjusted R-squared:  0.1597 
# F-statistic: 5.735 on 11 and 263 DF,  p-value: 2.704e-08


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-37092389'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC15001756.hg.1, x = as.factor(merge_v3$`AX-41232325`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs9873394 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TG", "2"="GG"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-41232325`), y=merge_v3$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9873394 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-41232325`, y=merge_v3$TC15001756.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9873394 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-41232325`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-41232325` <- as.factor(df$`AX-41232325`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-41232325`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9873394 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-41232325`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9873394 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL43_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#44 AX-38475535	TC02000318.hg.1	44	7.048621115	1.59E-11	0.001912842	0.247553871	EML6 	chr2	54950636	55199157	Coding	rs117781420	9	19355687	A	G	G	A


x<-merge_v3$`AX-38475535`
# 0   1   2  NA
# 2  27 245 1

#UCDSS.196	8	2	5	56	1	0	1	-0.0311587	-0.0281442	-0.0114177	0.000141421	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC02000318.hg.1~`AX-38475535`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38475535`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC02000318.hg.1 ~ `AX-38475535` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38475535` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -0.7791 -0.1027 -0.0222  0.1042  3.3037 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.4562278  0.2458293  26.263  < 2e-16 ***
# `AX-38475535`           -0.9512352  0.1089787  -8.729 3.08e-16 ***
# age                      0.0003152  0.0014094   0.224   0.8232    
# race                     0.0095161  0.0208116   0.457   0.6479    
# gender                   0.0187067  0.0372467   0.502   0.6159    
# diagnosis               -0.4833101  0.0780461  -6.193 2.28e-09 ***
# diagnosissubtype        -0.0024834  0.0203762  -0.122   0.9031    
# pc1                      0.6336505  0.3731894   1.698   0.0907 .  
# pc2                      0.3034042  0.3278636   0.925   0.3556    
# pc3                     -0.0454963  0.3684839  -0.123   0.9018    
# pc4                     -0.0861646  0.2934890  -0.294   0.7693    
# `AX-38475535`:diagnosis  0.2474947  0.0351889   7.033 1.75e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2892 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2403,	Adjusted R-squared:  0.2084 
# F-statistic: 7.532 on 11 and 262 DF,  p-value: 3.007e-11


min(merge_v3$TC02000318.hg.1) #4.35876
max(merge_v3$TC02000318.hg.1) #9.330151
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-38475535'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC02000318.hg.1, x = as.factor(x$`AX-38475535`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="EML6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38475535`), y=x$TC02000318.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="EML6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38475535`, y=x$TC02000318.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC02000318.hg.1", groupnames =c("diagnosis","`AX-38475535`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38475535` <- as.factor(df$`AX-38475535`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38475535`), y=df$TC02000318.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38475535`), y=df$TC02000318.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC02000318.hg.1-sd, ymax=TC02000318.hg.1+sd),width=.2)+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="EML6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("EML6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL44_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#45 AX-16372367	TC07000736.hg.1	45	7.033833317	1.74E-11	0.002045038	0.246993073	WNT16 	chr7	120965421	120981158	Coding	rs950391	X	86454329	A	G	G	A


x<-merge_v3$`AX-16372367`
# 0   1   2  NA
# 2  14 258 1

#UCDSS.513	8	1	2	71	1	1	1	-0.01459	0.0134688	-0.0145828	-0.00139512	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC07000736.hg.1~`AX-16372367`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372367`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC07000736.hg.1 ~ `AX-16372367` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372367` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.92158 -0.11090 -0.01671  0.11278  2.02421 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.300e+00  2.494e-01  25.258  < 2e-16 ***
# `AX-16372367`           -9.249e-01  1.055e-01  -8.771 2.31e-16 ***
# age                     -4.358e-06  1.131e-03  -0.004    0.997    
# race                     1.889e-02  1.707e-02   1.107    0.269    
# gender                  -2.931e-02  2.998e-02  -0.978    0.329    
# diagnosis               -4.392e-01  7.379e-02  -5.951 8.50e-09 ***
# diagnosissubtype        -2.427e-02  1.641e-02  -1.479    0.140    
# pc1                     -1.298e-01  3.021e-01  -0.430    0.668    
# pc2                      8.539e-03  2.666e-01   0.032    0.974    
# pc3                      1.292e-01  2.977e-01   0.434    0.665    
# pc4                     -1.141e-02  2.403e-01  -0.047    0.962    
# `AX-16372367`:diagnosis  2.471e-01  3.518e-02   7.024 1.85e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2338 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2505,	Adjusted R-squared:  0.2191 
# F-statistic: 7.962 on 11 and 262 DF,  p-value: 6.066e-12


min(merge_v3$TC07000736.hg.1) #4.170041
max(merge_v3$TC07000736.hg.1) #7.903534
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC07000736.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="WNT16 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("WNT16")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372367`), y=x$TC07000736.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="WNT16 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("WNT16")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372367`, y=x$TC07000736.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="WNT16 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("WNT16")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC07000736.hg.1", groupnames =c("diagnosis","`AX-16372367`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372367` <- as.factor(df$`AX-16372367`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372367`), y=df$TC07000736.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC07000736.hg.1-sd, ymax=TC07000736.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="WNT16 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("WNT16")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372367`), y=df$TC07000736.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC07000736.hg.1-sd, ymax=TC07000736.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="WNT16 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("WNT16")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL45_scattererrorbar.jpg",width = 6, height = 4)
```



```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#46 AX-38913193	TC03000198.hg.1	46	7.019271668	1.89E-11	0.002184206	0.163531314	TTC21A 	chr3	39149152	39180394	Coding	rs7081076	10	97174537	A	C	C	A


x<-merge_v3$`AX-38913193`
# 0   1   2  
# 1 26  248

```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#47	AX-16372292	TC04001036.hg.1	47	-6.996061566	2.18E-11	0.002458306	-0.177003544	CLNK 	chr4	10488019	10686489	Coding	rs2369519	X	86392534	A	G	G	A


x<-merge_v3$`AX-16372292`
# 0   1   2  NA 
# 248  18   7 2


#UCDSS.147	7	1	3	55	0	0	0	0.135238	-0.0306445	-0.0170027	-0.0934562	4
#UCDSS.448	7	1	3	59	0	0	0	0.160812	-0.0388213	-0.0185052	-0.061743	4
#UCDSS.513	8	1	2	71	1	1	1	-0.01459	0.0134688	-0.0145828	-0.00139512	4
#UCDSS.517	8	2	3	45	1	0	0	0.126155	-0.0357504	0.025814	0.649375	4
#UCDSS.519	7	1	3	43	0	0	0	0.15934	-0.0349643	-0.0133984	-0.0445161	4
#UCDSS.840	4	1	5	60	0	0	1	0.166113	-0.0373338	0.00370748	-0.0316622	1
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001036.hg.1~`AX-16372292`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-16372292`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001036.hg.1 ~ `AX-16372292` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-16372292` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.02778 -0.09246 -0.01808  0.08401  2.60611 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.3021042  0.1182666  36.376  < 2e-16 ***
# `AX-16372292`            0.7073049  0.0850399   8.317 5.03e-15 ***
# age                     -0.0003907  0.0011246  -0.347    0.729    
# race                     0.0060719  0.0169654   0.358    0.721    
# gender                  -0.0458644  0.0298513  -1.536    0.126    
# diagnosis                0.0362401  0.0288248   1.257    0.210    
# diagnosissubtype        -0.0149731  0.0163782  -0.914    0.361    
# pc1                     -0.1334928  0.3154126  -0.423    0.672    
# pc2                     -0.1213708  0.2675524  -0.454    0.650    
# pc3                      0.0537511  0.2963223   0.181    0.856    
# pc4                     -0.1458247  0.2414223  -0.604    0.546    
# `AX-16372292`:diagnosis -0.1766825  0.0253801  -6.961 2.72e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2328 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.227,	Adjusted R-squared:  0.1944 
#F-statistic: 6.967 on 11 and 261 DF,  p-value: 2.54e-10


min(merge_v3$TC04001036.hg.1) #4.00259
max(merge_v3$TC04001036.hg.1) #7.940702
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-16372292'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC04001036.hg.1, x = as.factor(x$`AX-16372292`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CLNK expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-16372292`), y=x$TC04001036.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="CLNK expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-16372292`, y=x$TC04001036.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC04001036.hg.1", groupnames =c("diagnosis","`AX-16372292`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-16372292` <- as.factor(df$`AX-16372292`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-16372292`), y=df$TC04001036.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-16372292`), y=df$TC04001036.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL47_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#48	AX-15355229	TC17000375.hg.1	48	6.982469661	2.36E-11	0.002611967	0.358066163	ZNF207 	chr17	30677128	30714780	Coding	rs1063632	6	31378510	A	G	G	A



x<-merge_v3$`AX-15355229`
# 0   1   2  NA 
# 2  31 240 2


#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.844	3	2	6	47	1	0	0	-0.00837881	0.0458948	0.108759	-0.00399842	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17000375.hg.1~`AX-15355229`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-15355229`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17000375.hg.1 ~ `AX-15355229` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-15355229` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -3.6370 -0.1555  0.0736  0.2038  1.2439 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             13.4630456  0.3160837  42.593  < 2e-16 ***
# `AX-15355229`           -0.4714890  0.1251786  -3.767 0.000205 ***
# age                     -0.0008388  0.0020650  -0.406 0.684928    
# race                    -0.0409836  0.0305867  -1.340 0.181437    
# gender                  -0.0177004  0.0537262  -0.329 0.742075    
# diagnosis               -0.6805679  0.1086180  -6.266 1.52e-09 ***
# diagnosissubtype        -0.0040214  0.0295349  -0.136 0.891801    
# pc1                     -0.1764159  0.5355965  -0.329 0.742131    
# pc2                      0.0449575  0.4783625   0.094 0.925195    
# pc3                      0.2782476  0.5324823   0.523 0.601732    
# pc4                      0.2907700  0.4244017   0.685 0.493870    
# `AX-15355229`:diagnosis  0.3585889  0.0513311   6.986 2.35e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.4181 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2064,	Adjusted R-squared:  0.1729 
# F-statistic:  6.17 on 11 and 261 DF,  p-value: 5.217e-09


min(merge_v3$TC17000375.hg.1) #6.735869
max(merge_v3$TC17000375.hg.1) #12.87503
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-15355229'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17000375.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZNF207 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ZNF207")," " , "expression")))+expand_limits(y=c(6,13))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-15355229`), y=x$TC17000375.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ZNF207 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("ZNF207")," " , "expression")))+ expand_limits(y=c(6,13))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-15355229`, y=x$TC17000375.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(6,13))

p2 <-p1 + labs(y="ZNF207 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF207")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC17000375.hg.1", groupnames =c("diagnosis","`AX-15355229`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-15355229` <- as.factor(df$`AX-15355229`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-15355229`), y=df$TC17000375.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17000375.hg.1-sd, ymax=TC17000375.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(6,12))

p2 <-p1 + labs(y="ZNF207 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF207")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-15355229`), y=df$TC17000375.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17000375.hg.1-sd, ymax=TC17000375.hg.1+sd),width=.2)+ expand_limits(y=c(6,12))

p2 <-p1 + labs(y="ZNF207 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF207")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL48_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#49 AX-34261733	TC15001756.hg.1	49	-6.961135619	2.69E-11	0.002908068	-0.110360449	AP3B2 	chr15	83328033	83378666	Coding	rs9812616	3	23237608	T	C	C	T


x<-merge_v3$`AX-34261733`
# 0   1   2  
# 6  22 247 
#UCDSS.693	8	2	3	48	1	0	0	0.141	-0.0354487	-0.0212148	-0.0503794	4
#UCDSS.716	1	1	3	56	1	0	0	0.144306	-0.0374005	-0.0204159	-0.0239198	1
#UCDSS.730	4	1	3	55	1	1	1	0.160227	-0.0337266	0.0066151	-0.0470337	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.050	3	2	3	59	1	0	1	0.170806	-0.0337208	-0.0114983	-0.042197	1
#UCSFCJ.091	6	2	5	43	0	0	0	0.00300048	0.0634869	0.136817	-0.00680118	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-34261733`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-34261733`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-34261733` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-34261733` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.45369 -0.08032 -0.02204  0.05488  0.78002 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.9429023  0.1097877  35.914  < 2e-16 ***
# `AX-34261733`            0.1464936  0.0459436   3.189   0.0016 ** 
# age                      0.0008081  0.0007232   1.117   0.2648    
# race                    -0.0007427  0.0108012  -0.069   0.9452    
# gender                   0.0157260  0.0189810   0.829   0.4081    
# diagnosis                0.2207498  0.0338595   6.520 3.59e-10 ***
# diagnosissubtype        -0.0066515  0.0104655  -0.636   0.5256    
# pc1                     -0.1094758  0.2198038  -0.498   0.6189    
# pc2                     -0.0777269  0.1697956  -0.458   0.6475    
# pc3                      0.0238382  0.1904869   0.125   0.9005    
# pc4                     -0.0361211  0.1549617  -0.233   0.8159    
# `AX-34261733`:diagnosis -0.1103604  0.0158538  -6.961 2.69e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.149 on 263 degrees of freedom
# Multiple R-squared:  0.1981,	Adjusted R-squared:  0.1646 
# F-statistic: 5.907 on 11 and 263 DF,  p-value: 1.398e-08


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-34261733'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC15001756.hg.1, x = as.factor(merge_v3$`AX-34261733`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs9812616 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-34261733`), y=merge_v3$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9812616 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-34261733`, y=merge_v3$TC15001756.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9812616 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-34261733`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-34261733` <- as.factor(df$`AX-34261733`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-34261733`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9812616 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-34261733`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs9812616
               genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL49_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#50 AX-35735671	TC17000375.hg.1	50	6.951363364	2.85E-11	0.003021746	0.363794293	ZNF207 	chr17	30677128	30714780	Coding	rs1051785	6	31378388	A	G	G	A


x<-merge_v3$`AX-35735671`
# 0   1   2  
#  1  32 240  

```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#51 AX-38578625	TC01000514.hg.1	51	6.944846641	2.96E-11	0.003080359	0.239353878	ZNF684 	chr1	40997233	41013841	Coding	rs148991762	X	13461054	A	C	C	A



x<-merge_v3$`AX-38578625`
# 0   1   2  
# 7  11 257 
#UCDSS.393	7	1	7	52	0	0	0	-0.0270024	-0.0155441	-0.00966672	0.00209958	4
#UCDSS.571	2	1	3	69	1	0	0	0.18122	-0.034571	-0.0165888	-0.102848	1
#UCDSS.584	8	1	5	87	0	1	1	-0.032718	-0.0295884	-0.017582	-3.20E-05	4
#UCDSS.591	8	2	3	79	1	1	1	0.149133	-0.037464	-0.0256132	-0.0202513	4
#UCDSS.619	8	1	2	60	1	0	0	0.0115403	0.17502	-0.0635817	0.000366905	4
#UCDSS.674	3	1	6	43	1	0	1	-0.0193745	0.00799197	0.0527321	0.00054771	1
#UCDSS.806	1	1	5	82	0	0	1	-0.032926	-0.0299422	-0.0214166	2.75E-05	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01000514.hg.1~`AX-38578625`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38578625`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01000514.hg.1 ~ `AX-38578625` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38578625` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.2365 -0.1375 -0.0173  0.1120  3.1152 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.702155   0.256612  30.015  < 2e-16 ***
# `AX-38578625`           -0.888000   0.109433  -8.115 1.88e-14 ***
# age                     -0.002619   0.001456  -1.799   0.0732 .  
# race                     0.009948   0.021736   0.458   0.6476    
# gender                   0.026834   0.038391   0.699   0.4852    
# diagnosis               -0.526380   0.079312  -6.637 1.82e-10 ***
# diagnosissubtype         0.024649   0.021309   1.157   0.2484    
# pc1                     -0.097787   0.391860  -0.250   0.8031    
# pc2                      0.103018   0.338298   0.305   0.7610    
# pc3                     -0.434387   0.383359  -1.133   0.2582    
# pc4                      0.262674   0.306107   0.858   0.3916    
# `AX-38578625`:diagnosis  0.239354   0.034465   6.945 2.96e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3009 on 263 degrees of freedom
# Multiple R-squared:  0.2167,	Adjusted R-squared:  0.1839 
# F-statistic: 6.614 on 11 and 263 DF,  p-value: 9.457e-10


min(merge_v3$TC01000514.hg.1) #5.342232
max(merge_v3$TC01000514.hg.1) #10.31879
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-38578625'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC01000514.hg.1, x = as.factor(merge_v3$`AX-38578625`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZNF684 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ZNF684")," " , "expression")))+expand_limits(y=c(5,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-38578625`), y=merge_v3$TC01000514.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ZNF684 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CA", "AA"))+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ expand_limits(y=c(5,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-38578625`, y=merge_v3$TC01000514.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5,11))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT51_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC01000514.hg.1", groupnames =c("diagnosis","`AX-38578625`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38578625` <- as.factor(df$`AX-38578625`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38578625`), y=df$TC01000514.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5,11))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38578625`), y=df$TC01000514.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd),width=.2)+ expand_limits(y=c(5,11))

p2 <-p1 + labs(y="ZNF684 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ZNF684")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL51_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#52 AX-11577079	TC17001827.hg.1	52	-6.936201272	3.12E-11	0.003181487	-0.286877009	ABCA6 	chr17	67074847	67138015	Coding	rs6665585	1	61090200	A	G	A	G


x<-merge_v3$`AX-11577079`
# 0   1   2  
# 235  37   3 
#UCDSS.636	8	2	7	52	1	1	0	-0.00798334	0.0432846	0.0985264	-0.00891958	4
#UCDSS.739	8	1	5	48	1	0	0	-0.0327341	-0.0304762	-0.0108814	0.00640304	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17001827.hg.1~`AX-11577079`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11577079`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17001827.hg.1 ~ `AX-11577079` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11577079` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.0868 -0.0975 -0.0137  0.0883  5.0785 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.967227   0.197654  20.072  < 2e-16 ***
# `AX-11577079`            1.154421   0.133388   8.655 5.03e-16 ***
# age                      0.002330   0.001876   1.242    0.215    
# race                     0.018312   0.028006   0.654    0.514    
# gender                  -0.087477   0.049604  -1.763    0.079 .  
# diagnosis                0.012132   0.047880   0.253    0.800    
# diagnosissubtype         0.004458   0.027447   0.162    0.871    
# pc1                      0.243874   0.499861   0.488    0.626    
# pc2                      0.150135   0.439013   0.342    0.733    
# pc3                     -0.078639   0.493354  -0.159    0.873    
# pc4                      0.020138   0.394552   0.051    0.959    
# `AX-11577079`:diagnosis -0.286877   0.041359  -6.936 3.12e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3887 on 263 degrees of freedom
# Multiple R-squared:  0.2419,	Adjusted R-squared:  0.2102 
#F-statistic: 7.628 on 11 and 263 DF,  p-value: 2.074e-11


min(merge_v3$TC17001827.hg.1) #3.87316
max(merge_v3$TC17001827.hg.1) #10.95106
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-11577079'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC17001827.hg.1, x = as.factor(merge_v3$`AX-11577079`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(3,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11577079`), y=merge_v3$TC17001827.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(3,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11577079`, y=merge_v3$TC17001827.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,11))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT52_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC17001827.hg.1", groupnames =c("diagnosis","`AX-11577079`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11577079` <- as.factor(df$`AX-11577079`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11577079`), y=df$TC17001827.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3,11))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11577079`), y=df$TC17001827.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd),width=.2)+ expand_limits(y=c(3,11))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL52_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#53 AX-40361177	TC15001756.hg.1	53	-6.914663557	3.55E-11	0.003493401	-0.11227898	AP3B2 	chr15	83328033	83378666	Coding	rs627635	18	66904739	T	C	T	C

x<-merge_v3$`AX-40361177`
# 0   1   2  NA
# 4  26 241 4
#UCDSS.602	3	1	3	51	1	1	0	0.119142	-0.0404059	-0.0206839	-0.0306819	1
#UCDSS.693	8	2	3	48	1	0	0	0.141	-0.0354487	-0.0212148	-0.0503794	4
#UCDSS.708	8	2	3	41	1	1	1	0.161344	-0.0350806	0.0073334	-0.0420488	4
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-40361177`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-40361177`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC15001756.hg.1 ~ `AX-40361177` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-40361177` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.43842 -0.07671 -0.02287  0.05674  0.80739 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.8812466  0.1209980  32.077  < 2e-16 ***
# `AX-40361177`            0.1712737  0.0499892   3.426 0.000712 ***
# age                      0.0005787  0.0007072   0.818 0.413959    
# race                     0.0012275  0.0104052   0.118 0.906184    
# gender                   0.0284424  0.0187210   1.519 0.129912    
# diagnosis                0.2216745  0.0336778   6.582 2.56e-10 ***
# diagnosissubtype        -0.0053893  0.0102848  -0.524 0.600718    
# pc1                     -0.0060242  0.1939495  -0.031 0.975245    
# pc2                      0.0332261  0.1639865   0.203 0.839595    
# pc3                     -0.0235667  0.1843471  -0.128 0.898376    
# pc4                     -0.0492010  0.1480144  -0.332 0.739851    
# `AX-40361177`:diagnosis -0.1128706  0.0161278  -6.998 2.21e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1446 on 259 degrees of freedom
#   (4 observations deleted due to missingness)
# Multiple R-squared:  0.2385,	Adjusted R-squared:  0.2061 
# F-statistic: 7.374 on 11 and 259 DF,  p-value: 5.658e-11


min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-40361177'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-40361177`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs627635 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-40361177`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs627635 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-40361177`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs627635 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT53_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-40361177`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-40361177` <- as.factor(df$`AX-40361177`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-40361177`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs627635 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-40361177`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs627635 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL53_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#54 AX-51270082	TC18000998.hg.1	54	-6.91422754	3.56E-11	0.003493401	-0.226996729	SMAD4 	chr18	48556583	48611415	Coding	rs9779183	X	13009957	T	C	T	C

x<-merge_v3$`AX-51270082`
# 0   1   2  NA
#258   8   7

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC18000998.hg.1~`AX-51270082`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-51270082`*diagnosis, data = merge_v3)
# summary (fit)
# Call:
# lm(formula = TC18000998.hg.1 ~ `AX-51270082` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-51270082` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -2.21625 -0.09695  0.00509  0.12093  0.85453 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             10.695500   0.138207  77.388  < 2e-16 ***
# `AX-51270082`            0.145035   0.081526   1.779 0.076404 .  
# age                     -0.001899   0.001308  -1.453 0.147519    
# race                    -0.031691   0.019534  -1.622 0.105932    
# gender                   0.008437   0.033964   0.248 0.804017    
# diagnosis               -0.019573   0.032673  -0.599 0.549662    
# diagnosissubtype         0.019140   0.018677   1.025 0.306408    
# pc1                      0.136568   0.341215   0.400 0.689308    
# pc2                     -0.366396   0.308765  -1.187 0.236445    
# pc3                      0.029706   0.343075   0.087 0.931066    
# pc4                      1.083963   0.303100   3.576 0.000415 ***
# `AX-51270082`:diagnosis -0.226812   0.032812  -6.912 3.65e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.264 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:   0.25,	Adjusted R-squared:  0.2184 
# F-statistic: 7.911 on 11 and 261 DF,  p-value: 7.436e-12


min(merge_v3$TC18000998.hg.1) #6.752242
max(merge_v3$TC18000998.hg.1) #10.89588
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-51270082'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC18000998.hg.1, x = as.factor(x$`AX-51270082`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SMAD4 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("SMAD4")," " , "expression")))+expand_limits(y=c(6,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-51270082`), y=x$TC18000998.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="SMAD4 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ expand_limits(y=c(6,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-51270082`, y=x$TC18000998.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(6,11))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT54_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC18000998.hg.1", groupnames =c("diagnosis","`AX-51270082`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-51270082` <- as.factor(df$`AX-51270082`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-51270082`), y=df$TC18000998.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(6,11))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-51270082`), y=df$TC18000998.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd),width=.2)+ expand_limits(y=c(6,11))

p2 <-p1 + labs(y="SMAD4 expression", x = "rs9779183 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SMAD4")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL54_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#55 AX-15355229	TC0M000010.hg.1	55	6.905644369	3.74E-11	0.00361007	0.425948363	---	chrM	14857	15888	Coding	rs1063632	6	31378510	A	G	G	A

x<-merge_v3$`AX-15355229`
# 0   1   2 NA
#2  31 240  2
#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.844	3	2	6	47	1	0	0	-0.00837881	0.0458948	0.108759	-0.00399842	1
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC0M000010.hg.1~`AX-15355229`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-15355229`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC0M000010.hg.1 ~ `AX-15355229` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-15355229` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -4.8375 -0.0220  0.0744  0.1621  1.3594 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             20.703699   0.381236  54.307  < 2e-16 ***
# `AX-15355229`           -0.452243   0.150981  -2.995    0.003 ** 
# age                     -0.001310   0.002491  -0.526    0.599    
# race                    -0.011084   0.036891  -0.300    0.764    
# gender                  -0.051711   0.064801  -0.798    0.426    
# diagnosis               -0.839815   0.131007  -6.410 6.75e-10 ***
# diagnosissubtype         0.024644   0.035623   0.692    0.490    
# pc1                      0.108357   0.645996   0.168    0.867    
# pc2                      0.218026   0.576965   0.378    0.706    
# pc3                      0.325577   0.642240   0.507    0.613    
# pc4                      0.206404   0.511881   0.403    0.687    
# `AX-15355229`:diagnosis  0.426654   0.061912   6.891 4.13e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.5042 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.2176,	Adjusted R-squared:  0.1846 
# F-statistic: 6.599 on 11 and 261 DF,  p-value: 1.021e-09


min(merge_v3$TC0M000010.hg.1) #12.51037
max(merge_v3$TC0M000010.hg.1) #19.92878
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-15355229'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC0M000010.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("TC0M000010.hg.1")," " , "expression")))+expand_limits(y=c(12,20))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55.jpg",width = 8, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-15355229`), y=x$TC0M000010.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ expand_limits(y=c(12,20))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-15355229`, y=x$TC0M000010.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT55_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC0M000010.hg.1", groupnames =c("diagnosis","`AX-15355229`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-15355229` <- as.factor(df$`AX-15355229`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-15355229`), y=df$TC0M000010.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC0M000010.hg.1-sd, ymax=TC0M000010.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-15355229`), y=df$TC0M000010.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC0M000010.hg.1-sd, ymax=TC0M000010.hg.1+sd),width=.2)+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL55_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#56 AX-11577079	TC04001036.hg.1	57	-6.877826923	4.42E-11	0.004110971	-0.17066165	CLNK 	chr4	10488019	10686489	Coding	rs6665585	1	61090200	A	G	A	G


x<-merge_v3$`AX-11577079`
# 0   1   2 
#235  37   3 
#UCDSS.636	8	2	7	52	1	1	0	-0.00798334	0.0432846	0.0985264	-0.00891958	4
#UCDSS.739	8	1	5	48	1	0	0	-0.0327341	-0.0304762	-0.0108814	0.00640304	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001036.hg.1~`AX-11577079`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11577079`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001036.hg.1 ~ `AX-11577079` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11577079` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.69990 -0.08739 -0.01113  0.07920  2.75538 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.1558861  0.1185811  35.047  < 2e-16 ***
# `AX-11577079`            0.6542702  0.0800256   8.176 1.26e-14 ***
# age                     -0.0001239  0.0011255  -0.110   0.9124    
# race                     0.0225159  0.0168017   1.340   0.1814    
# gender                  -0.0416999  0.0297598  -1.401   0.1623    
# diagnosis                0.0108966  0.0287254   0.379   0.7047    
# diagnosissubtype         0.0051178  0.0164664   0.311   0.7562    
# pc1                      0.7282319  0.2998886   2.428   0.0158 *  
# pc2                      0.0144303  0.2633835   0.055   0.9563    
# pc3                     -0.1364352  0.2959847  -0.461   0.6452    
# pc4                      0.1650275  0.2367089   0.697   0.4863    
# `AX-11577079`:diagnosis -0.1706616  0.0248133  -6.878 4.42e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2332 on 263 degrees of freedom
# Multiple R-squared:  0.2199,	Adjusted R-squared:  0.1873 
# F-statistic:  6.74 on 11 and 263 DF,  p-value: 5.861e-10


min(merge_v3$TC04001036.hg.1) #4.00259
max(merge_v3$TC04001036.hg.1) #7.940702
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-11577079'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC04001036.hg.1, x = as.factor(merge_v3$`AX-11577079`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CLNK expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11577079`), y=merge_v3$TC04001036.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="CLNK expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11577079`, y=merge_v3$TC04001036.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT56_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC04001036.hg.1", groupnames =c("diagnosis","`AX-11577079`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11577079` <- as.factor(df$`AX-11577079`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11577079`), y=df$TC04001036.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11577079`), y=df$TC04001036.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001036.hg.1-sd, ymax=TC04001036.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="CLNK expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CLNK")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("AA", "AG", "GG"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL56_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#57 AX-38578625	TC09001542.hg.1	56	6.878161331	4.41E-11	0.004110971	0.174565276	C5 	chr9	123714614	123837452	Coding	rs148991762	X	13461054	A	C	C	A


x<-merge_v3$`AX-38578625`
# 0   1   2 
#7  11 257 

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC09001542.hg.1~`AX-38578625`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38578625`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC09001542.hg.1 ~ `AX-38578625` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38578625` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.92924 -0.09722 -0.01503  0.07623  2.04657 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.365295   0.188967  33.685  < 2e-16 ***
# `AX-38578625`           -0.658530   0.080585  -8.172 1.29e-14 ***
# age                     -0.001109   0.001072  -1.035  0.30182    
# race                     0.029835   0.016006   1.864  0.06345 .  
# gender                   0.048140   0.028271   1.703  0.08979 .  
# diagnosis               -0.382679   0.058405  -6.552 2.97e-10 ***
# diagnosissubtype         0.012165   0.015692   0.775  0.43889    
# pc1                     -0.612292   0.288562  -2.122  0.03478 *  
# pc2                      0.772656   0.249119   3.102  0.00213 ** 
# pc3                     -0.465469   0.282302  -1.649  0.10038    
# pc4                      0.028612   0.225414   0.127  0.89909    
# `AX-38578625`:diagnosis  0.174565   0.025380   6.878 4.41e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2216 on 263 degrees of freedom
# Multiple R-squared:  0.2725,	Adjusted R-squared:  0.2421 
# F-statistic: 8.955 on 11 and 263 DF,  p-value: 1.542e-13


min(merge_v3$TC09001542.hg.1) #4.803989
max(merge_v3$TC09001542.hg.1) #8.238651
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-11577079'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC09001542.hg.1, x = as.factor(merge_v3$`AX-38578625`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="C5 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CA", "2"="AA"))+scale_y_continuous(expression(paste( italic("C5")," " , "expression")))+expand_limits(y=c(4,8))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-38578625`), y=merge_v3$TC09001542.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="C5 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CA", "AA"))+scale_y_continuous(expression(paste(italic("C5")," " , "expression")))+ expand_limits(y=c(4,8))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-38578625`, y=merge_v3$TC09001542.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="C5 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("C5")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT57_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC09001542.hg.1", groupnames =c("diagnosis","`AX-38578625`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38578625` <- as.factor(df$`AX-38578625`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38578625`), y=df$TC09001542.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC09001542.hg.1-sd, ymax=TC09001542.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="C5 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("C5")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38578625`), y=df$TC09001542.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC09001542.hg.1-sd, ymax=TC09001542.hg.1+sd),width=.2)+ expand_limits(y=c(4,8))

p2 <-p1 + labs(y="C5 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("C5")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL57_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#58 AX-15355229	TC02004959.hg.1	58	-6.871106523	4.60E-11	0.004204776	-0.134159446	MORN2 	chr2	39103103	39109850	Coding	rs1063632	6	31378510	A	G	G	A


x<-merge_v3$`AX-15355229`
# 0   1   2 NA
#2  31 240  2
#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.844	3	2	6	47	1	0	0	-0.00837881	0.0458948	0.108759	-0.00399842	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC02004959.hg.1~`AX-15355229`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-15355229`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC02004959.hg.1 ~ `AX-15355229` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-15355229` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.40613 -0.08960 -0.01337  0.09061  0.94640 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.5689352  0.1205497  29.606  < 2e-16 ***
# `AX-15355229`            0.2024514  0.0477413   4.241 3.10e-05 ***
# age                     -0.0009684  0.0007876  -1.230    0.220    
# race                    -0.0070687  0.0116653  -0.606    0.545    
# gender                   0.0147589  0.0204904   0.720    0.472    
# diagnosis                0.2544263  0.0414253   6.142 3.03e-09 ***
# diagnosissubtype        -0.0033272  0.0112642  -0.295    0.768    
# pc1                     -0.2074298  0.2042687  -1.015    0.311    
# pc2                      0.1379303  0.1824405   0.756    0.450    
# pc3                     -0.0919984  0.2030810  -0.453    0.651    
# pc4                     -0.0286798  0.1618606  -0.177    0.859    
# `AX-15355229`:diagnosis -0.1344837  0.0195769  -6.869 4.71e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1594 on 261 degrees of freedom
#   (2 observations deleted due to missingness)
# Multiple R-squared:  0.1817,	Adjusted R-squared:  0.1472 
# F-statistic: 5.267 on 11 and 261 DF,  p-value: 1.645e-07


min(merge_v3$TC02004959.hg.1) #3.433969
max(merge_v3$TC02004959.hg.1) #5.437461
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-15355229'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC02004959.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="MORN2 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("MORN2")," " , "expression")))+expand_limits(y=c(3,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-15355229`), y=x$TC02004959.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="MORN2 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("MORN2")," " , "expression")))+ expand_limits(y=c(3,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-15355229`, y=x$TC02004959.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,6))

p2 <-p1 + labs(y="MORN2 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("MORN2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT58_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC02004959.hg.1", groupnames =c("diagnosis","`AX-15355229`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-15355229` <- as.factor(df$`AX-15355229`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-15355229`), y=df$TC02004959.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC02004959.hg.1-sd, ymax=TC02004959.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3,6))

p2 <-p1 + labs(y="MORN2 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("MORN2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-15355229`), y=df$TC02004959.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC02004959.hg.1-sd, ymax=TC02004959.hg.1+sd),width=.2)+ expand_limits(y=c(3,6))

p2 <-p1 + labs(y="MORN2 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("MORN2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL58_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#59 AX-38554997	TC01001640.hg.1	59	-6.837093566	5.63E-11	0.005057856	-0.263576607	PTPRC 	chr1	198607801	198726545	Coding	rs149536248	X	11320892	T	C	T	C


x<-merge_v3$`AX-38554997`
# 0   1   2 
#256  15   4
#UCDSS.089	2	1	5	54	1	0	1	-0.0264045	-0.00915837	-0.0208476	0.00082716	1
#UCDSS.437	8	1	7	39	0	0	0	-0.0311912	-0.0293673	-0.0131242	0.00364803	4
#UCDSS.620	2	1	5	65	0	1	0	-0.0312336	-0.0270251	-0.0142701	0.00135788	1
#UCSFCJ.070	4	1	5	41	0	0	0	-0.0251718	-0.0277895	-0.0213072	0.000793302	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01001640.hg.1~`AX-38554997`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38554997`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01001640.hg.1 ~ `AX-38554997` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38554997` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -3.13710 -0.11950  0.01449  0.10716  0.94603 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             17.993636   0.153775 117.013  < 2e-16 ***
# `AX-38554997`            0.270968   0.100923   2.685  0.00772 ** 
# age                      0.002253   0.001468   1.535  0.12607    
# race                    -0.050132   0.021870  -2.292  0.02268 *  
# gender                   0.039717   0.038826   1.023  0.30727    
# diagnosis               -0.019550   0.037110  -0.527  0.59876    
# diagnosissubtype         0.013157   0.021246   0.619  0.53628    
# pc1                     -0.305765   0.387908  -0.788  0.43127    
# pc2                      0.099185   0.340615   0.291  0.77113    
# pc3                      0.863390   0.386341   2.235  0.02627 *  
# pc4                      0.065831   0.307670   0.214  0.83074    
# `AX-38554997`:diagnosis -0.263577   0.038551  -6.837 5.63e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3032 on 263 degrees of freedom
# Multiple R-squared:  0.2621,	Adjusted R-squared:  0.2312 
# F-statistic:  8.49 on 11 and 263 DF,  p-value: 8.459e-13


min(merge_v3$TC01001640.hg.1) #13.08913
max(merge_v3$TC01001640.hg.1) #18.45755
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-38554997'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC01001640.hg.1, x = as.factor(merge_v3$`AX-38554997`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PTPRC expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("PTPRC")," " , "expression")))+expand_limits(y=c(13,19))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-38554997`), y=merge_v3$TC01001640.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="PTPRC expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ expand_limits(y=c(13,19))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-38554997`, y=merge_v3$TC01001640.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQT59_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC01001640.hg.1", groupnames =c("diagnosis","`AX-38554997`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38554997` <- as.factor(df$`AX-38554997`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38554997`), y=df$TC01001640.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01001640.hg.1-sd, ymax=TC01001640.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38554997`), y=df$TC01001640.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01001640.hg.1-sd, ymax=TC01001640.hg.1+sd),width=.2)+ expand_limits(y=c(13,19))

p2 <-p1 + labs(y="PTPRC expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTPRC")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL59_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#60 AX-12423034	TC17000383.hg.1	60	-6.833118712	5.76E-11	0.005092034	-0.358098137	CCL2 	chr17	32582296	32584222	Coding	rs11922093	3	63269389	T	C	T	C


x<-merge_v3$`AX-12423034`
# 0   1   2
#245  30 

```


```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#61 AX-40516485	TC15001756.hg.1	61	-6.818708473	6.28E-11	0.005454382	-0.123569734	AP3B2 	chr15	83328033	83378666	Coding	rs6113722	20	22557099	A	G	G	A


x<-merge_v3$`AX-40516485`
# 0   1   2
# 1  35 239 

```


```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#62 AX-38913193	TC03002054.hg.1	62	6.813513954	6.47E-11	0.005533762	0.406578376	LAMP3 	chr3	182840001	182881627	Coding	rs7081076	10	97174537	A	C	C	A


x<-merge_v3$`AX-38913193`
# 0   1   2
# 1  26 248 

```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#63 AX-42755741	TC01002971.hg.1	63	6.808659966	6.66E-11	0.005604385	0.213550656	SLC16A4 	chr1	110905470	110933704	Coding	rs12399124	X	3544089	A	G	G	A


x<-merge_v3$`AX-42755741`
# 0   1   2 NA
# 7  13 254 1

```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01002971.hg.1~`AX-42755741`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-42755741`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01002971.hg.1 ~ `AX-42755741` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-42755741` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.0627 -0.1086 -0.0223  0.0930  3.1768 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.8432369  0.2375969  28.802  < 2e-16 ***
# `AX-42755741`           -0.8418851  0.0998037  -8.435 2.25e-15 ***
# age                      0.0009914  0.0013749   0.721    0.471    
# race                     0.0019001  0.0204446   0.093    0.926    
# gender                  -0.0266483  0.0361956  -0.736    0.462    
# diagnosis               -0.4447338  0.0714997  -6.220 1.95e-09 ***
# diagnosissubtype         0.0070424  0.0201734   0.349    0.727    
# pc1                     -0.2825680  0.3647948  -0.775    0.439    
# pc2                     -0.1781749  0.3208944  -0.555    0.579    
# pc3                     -0.1036498  0.3607652  -0.287    0.774    
# pc4                     -0.1156777  0.2883741  -0.401    0.689    
# `AX-42755741`:diagnosis  0.2135698  0.0314242   6.796 7.21e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2841 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.2422,	Adjusted R-squared:  0.2104 
# F-statistic: 7.614 on 11 and 262 DF,  p-value: 2.21e-11


min(merge_v3$TC01002971.hg.1) #4.827518
max(merge_v3$TC01002971.hg.1) #9.654251
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-42755741'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC01002971.hg.1, x = as.factor(x$`AX-42755741`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SLC16A4 expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("SLC16A4")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-42755741`), y=x$TC01002971.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-42755741`, y=x$TC01002971.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC01002971.hg.1", groupnames =c("diagnosis","`AX-42755741`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-42755741` <- as.factor(df$`AX-42755741`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-42755741`), y=df$TC01002971.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-42755741`), y=df$TC01002971.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd),width=.2)+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL63_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#No figure for this
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#64 AX-35735671	TC02004959.hg.1	64	-6.804526215	6.82E-11	0.005653169	-0.135659575	MORN2 	chr2	39103103	39109850	Coding	rs1051785	6	31378388	A	G	G	A


x<-merge_v3$`AX-35735671`
# 0   1   2
#  1  32 240

```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#65	AX-38554997	TC0M000010.hg.1	65	-6.798910345	7.05E-11	0.005753829	-0.433595632	---	chrM	14857	15888	Coding	rs149536248	X	11320892	T	C	T	C

x<-merge_v3$`AX-38554997`
# 0   1   2
# 256 15  4

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC0M000010.hg.1~`AX-38554997`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38554997`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC0M000010.hg.1 ~ `AX-38554997` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38554997` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -4.7621 -0.0355  0.0775  0.1708  1.3101 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             19.757577   0.254388  77.667  < 2e-16 ***
# `AX-38554997`            0.490585   0.166956   2.938  0.00359 ** 
# age                     -0.001213   0.002429  -0.500  0.61784    
# race                    -0.034319   0.036179  -0.949  0.34370    
# gender                   0.047159   0.064229   0.734  0.46346    
# diagnosis               -0.005835   0.061391  -0.095  0.92435    
# diagnosissubtype         0.034428   0.035147   0.980  0.32822    
# pc1                     -0.082664   0.641710  -0.129  0.89760    
# pc2                      0.309250   0.563474   0.549  0.58359    
# pc3                      0.231317   0.639118   0.362  0.71769    
# pc4                      0.093750   0.508974   0.184  0.85400    
# `AX-38554997`:diagnosis -0.433596   0.063774  -6.799 7.05e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.5015 on 263 degrees of freedom
# Multiple R-squared:  0.2206,	Adjusted R-squared:  0.188 
# F-statistic: 6.767 on 11 and 263 DF,  p-value: 5.285e-10


min(merge_v3$TC0M000010.hg.1) #12.51037
max(merge_v3$TC0M000010.hg.1) #19.92878
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-38554997'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC0M000010.hg.1, x = as.factor(merge_v3$`AX-38554997`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("TC0M000010.hg.1")," " , "expression")))+expand_limits(y=c(12,20))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-38554997`), y=merge_v3$TC0M000010.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ expand_limits(y=c(12,20))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-38554997`, y=merge_v3$TC0M000010.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC0M000010.hg.1", groupnames =c("diagnosis","`AX-38554997`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38554997` <- as.factor(df$`AX-38554997`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38554997`), y=df$TC0M000010.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC0M000010.hg.1-sd, ymax=TC0M000010.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38554997`), y=df$TC0M000010.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC0M000010.hg.1-sd, ymax=TC0M000010.hg.1+sd),width=.2)+ expand_limits(y=c(12,20))

p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("TC0M000010.hg.1")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL65_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#66	AX-11552584	TC09000560.hg.1	66	6.78660488	7.58E-11	0.006076483	0.315262295	UGCG 	chr9	114659046	114697649	Coding	rs5955819	X	19599813	T	C	C	T


x<-merge_v3$`AX-11552584`
# 0   1   2
# 18  12 245

#0
#UCDSS.211	8	1	7	80	1	0	1	-0.0139458	0.0412199	0.330391	-0.0186824	4
#UCDSS.225	4	1	7	76	0	0	0	-0.0114419	0.0331026	0.0890845	-0.00165415	1
#UCDSS.245	2	1	5	63	1	0	0	-0.0321194	-0.0298252	-0.0169677	0.000904222	1
#UCDSS.294	8	1	5	48	1	0	1	-0.0325009	-0.0290727	-0.017021	0.00196963	4
#UCDSS.526	8	2	5	70	1	0	1	-0.0332491	-0.0270117	-0.0119473	0.00293329	4
#UCDSS.531	8	1	5	61	1	1	1	-0.0310296	-0.0298946	-0.0288656	-0.0194769	4
#UCDSS.534	8	1	5	87	1	0	1	-0.0278813	-0.0251681	-0.0151512	0.000736494	4
#UCDSS.573	7	1	5	74	0	0	0	-0.0330446	-0.0322942	-0.0143157	0.00197388	4
#UCDSS.584	8	1	5	87	0	1	1	-0.032718	-0.0295884	-0.017582	-3.20E-05	4
#UCDSS.636	8	2	7	52	1	1	0	-0.00798334	0.0432846	0.0985264	-0.00891958	4
#UCDSS.674	3	1	6	43	1	0	1	-0.0193745	0.00799197	0.0527321	0.00054771	1
#UCDSS.676	8	1	5	56	1	0	1	-0.0288386	-0.0246101	-0.014345	0.0032324	4
#UCDSS.739	8	1	5	48	1	0	0	-0.0327341	-0.0304762	-0.0108814	0.00640304	4
#UCDSS.812	8	1	6	45	1	0	1	-0.0134626	0.037001	0.0989184	0.000276983	4
#UCDSS.854	8	1	6	34	0	0	0	-0.00312709	0.0524738	0.133219	-0.0139748	4
#UCDSSTourettes.TSFS623CT	8	2	6	47	0	1	0	-0.0259315	-0.015799	0.00805229	0.00426135	4
#UCSFCJ.128	8	1	6	91	1	0	1	-0.01013	0.00301463	0.0420307	-0.00780142	4
#UofASS.002	4	1	5	68	1	0	1	-0.0317127	-0.0293228	-0.0169263	0.000271324	1


#1
#UCDSS.208	8	2	7	58	1	0	0	-0.0155297	0.0373228	0.330817	-0.0138419	4
#UCDSS.209	8	2	7	56	1	0	0	-0.0128046	0.0488689	0.343504	-0.0156587	4
#UCDSS.210	8	2	6	54	1	0	1	-0.0143168	0.0440876	0.329573	-0.0172749	4
#UCDSS.315	7	2	5	59	0	0	0	-0.0337728	-0.0304845	-0.018475	0.000709451	4
#UCDSS.331	8	2	5	57	0	0	0	-0.0332142	-0.0293081	-0.0111536	0.00536828	4
#UCDSS.364	8	2	5	77	0	0	0	-0.0244142	-0.0280246	-0.0109571	-2.16E-05	4
#UCDSS.552	8	2	3	62	1	1	0	0.161348	-0.0344033	-0.00370047	-0.0234773	4
#UCDSS.555	8	2	5	85	1	0	1	-0.0315911	-0.0276138	-0.0168431	7.76E-06	4
#UCDSS.721	8	2	5	49	0	0	1	-0.0327703	-0.0284608	-0.0167504	0.00327062	4
#UCDSS.738	7	2	5	66	0	0	0	-0.0316139	-0.0284217	-0.0124966	0.00314247	4
#UCDSS.744	8	2	6	55	1	0	0	-0.00926499	0.0294275	0.0985531	-0.0119912	4
#UCDSS.779	8	2	5	51	0	1	1	-0.0320623	-0.0294835	-0.019245	-0.00359494	4


```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC09000560.hg.1~`AX-11552584`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11552584`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC09000560.hg.1 ~ `AX-11552584` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11552584` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.68368 -0.23763 -0.05263  0.12908  2.88347 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             11.162745   0.413179  27.017  < 2e-16 ***
# `AX-11552584`           -1.255784   0.166563  -7.539 7.66e-13 ***
# age                     -0.001937   0.002339  -0.828   0.4083    
# race                    -0.055551   0.034835  -1.595   0.1120    
# gender                   0.126101   0.062014   2.033   0.0430 *  
# diagnosis               -0.591049   0.103591  -5.706 3.12e-08 ***
# diagnosissubtype        -0.055281   0.033876  -1.632   0.1039    
# pc1                     -1.243620   0.618040  -2.012   0.0452 *  
# pc2                     -0.569659   0.542516  -1.050   0.2947    
# pc3                     -0.058199   0.626840  -0.093   0.9261    
# pc4                     -0.146676   0.490177  -0.299   0.7650    
# `AX-11552584`:diagnosis  0.315262   0.046454   6.787 7.58e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.4828 on 263 degrees of freedom
# Multiple R-squared:  0.224,	Adjusted R-squared:  0.1916 
# F-statistic: 6.902 on 11 and 263 DF,  p-value: 3.171e-10

min(merge_v3$TC09000560.hg.1) #7.226324
max(merge_v3$TC09000560.hg.1) #11.38889
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-11552584'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC09000560.hg.1, x = as.factor(merge_v3$`AX-11552584`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="UGCG expression", x = "rs5955819 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("UGCG")," " , "expression")))+expand_limits(y=c(7,12))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11552584`), y=merge_v3$TC09000560.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="UGCG expression", x = "rs5955819 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ expand_limits(y=c(7,12))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11552584`, y=merge_v3$TC09000560.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(7,12))

p2 <-p1 + labs(y="UGCG expression", x = "rs5955819 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC09000560.hg.1", groupnames =c("diagnosis","`AX-11552584`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11552584` <- as.factor(df$`AX-11552584`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11552584`), y=df$TC09000560.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC09000560.hg.1-sd, ymax=TC09000560.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(7,12))

p2 <-p1 + labs(y="UGCG expression", x = "rs5955819 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_boxerrorbar.jpg",width = 6, height = 4)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11552584`), y=df$TC09000560.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC09000560.hg.1-sd, ymax=TC09000560.hg.1+sd),width=.2)+ expand_limits(y=c(7,12))

p2 <-p1 + labs(y="UGCG expression", x = "rs5955819 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("UGCG")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL66_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#67 AX-13456741	TC01002971.hg.1	67	6.784520907	7.68E-11	0.006076483	0.208621576	SLC16A4 	chr1	110905470	110933704	Coding	rs77599711	19	56425689	A	G	G	A


x<-merge_v3$`AX-13456741`
# 0   1   2
# 3  33 239
#UCDSS.380	8	2	5	74	1	0	1	-0.029441	-0.0316534	-0.0287257	-0.00900693	4
#UCDSS.531	8	1	5	61	1	1	1	-0.0310296	-0.0298946	-0.0288656	-0.0194769	4
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC01002971.hg.1~`AX-13456741`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-13456741`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC01002971.hg.1 ~ `AX-13456741` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-13456741` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -0.7527 -0.1070 -0.0107  0.0991  3.2610 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.609449   0.219115  30.164  < 2e-16 ***
# `AX-13456741`           -0.799661   0.095816  -8.346 4.05e-15 ***
# age                      0.001656   0.001375   1.205    0.229    
# race                     0.023592   0.020745   1.137    0.256    
# gender                  -0.035365   0.036282  -0.975    0.331    
# diagnosis               -0.399140   0.067367  -5.925 9.76e-09 ***
# diagnosissubtype        -0.008284   0.019940  -0.415    0.678    
# pc1                     -0.118724   0.369764  -0.321    0.748    
# pc2                      0.035460   0.323993   0.109    0.913    
# pc3                     -0.217730   0.364776  -0.597    0.551    
# pc4                     -0.067810   0.289076  -0.235    0.815    
# `AX-13456741`:diagnosis  0.208622   0.030750   6.785 7.68e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2847 on 263 degrees of freedom
# Multiple R-squared:  0.2364,	Adjusted R-squared:  0.2045 
# F-statistic: 7.402 on 11 and 263 DF,  p-value: 4.827e-11

min(merge_v3$TC01002971.hg.1) #4.827518
max(merge_v3$TC01002971.hg.1) #9.654251
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-13456741'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC01002971.hg.1, x = as.factor(merge_v3$`AX-13456741`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="SLC16A4 expression", x = "rs77599711 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("SLC16A4")," " , "expression")))+expand_limits(y=c(4,10))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-13456741`), y=merge_v3$TC01002971.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs77599711 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ expand_limits(y=c(4,10))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-13456741`, y=merge_v3$TC01002971.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs77599711 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC01002971.hg.1", groupnames =c("diagnosis","`AX-13456741`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-13456741` <- as.factor(df$`AX-13456741`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-13456741`), y=df$TC01002971.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs77599711 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-13456741`), y=df$TC01002971.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01002971.hg.1-sd, ymax=TC01002971.hg.1+sd),width=.2)+ expand_limits(y=c(4,10))

p2 <-p1 + labs(y="SLC16A4 expression", x = "rs77599711 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("SLC16A4")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL67_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#68	AX-38475535	TC17001827.hg.1	68	6.776920717	8.03E-11	0.006261296	0.322396223	ABCA6 	chr17	67074847	67138015	Coding	rs117781420	9	19355687	A	G	G	A


x<-merge_v3$`AX-38475535`
# 0   1   2 NA
# 2 27  245 1
#UCSFCJ.099	2	1	5	67	1	0	0	-0.0324391	-0.0290529	-0.0153278	0.00276611	1
#UCDSS.196	8	2	5	56	1	0	1	-0.0311587	-0.0281442	-0.0114177	0.000141421	4
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC17001827.hg.1~`AX-38475535`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-38475535`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC17001827.hg.1 ~ `AX-38475535` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-38475535` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.2157 -0.0936 -0.0173  0.0804  4.9693 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              6.3765765  0.3327531  19.163  < 2e-16 ***
# `AX-38475535`           -1.2483980  0.1475129  -8.463 1.87e-15 ***
# age                      0.0032658  0.0019078   1.712   0.0881 .  
# race                     0.0078941  0.0281705   0.280   0.7795    
# gender                  -0.0091947  0.0504169  -0.182   0.8554    
# diagnosis               -0.6307862  0.1056427  -5.971 7.65e-09 ***
# diagnosissubtype        -0.0004362  0.0275811  -0.016   0.9874    
# pc1                      0.1684659  0.5051471   0.333   0.7390    
# pc2                      0.1330721  0.4437943   0.300   0.7645    
# pc3                      0.1800970  0.4987777   0.361   0.7183    
# pc4                     -0.0409260  0.3972650  -0.103   0.9180    
# `AX-38475535`:diagnosis  0.3220495  0.0476315   6.761 8.86e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.3914 on 262 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.233,	Adjusted R-squared:  0.2008 
# F-statistic: 7.236 on 11 and 262 DF,  p-value: 9.109e-11

min(merge_v3$TC17001827.hg.1) #3.87316
max(merge_v3$TC17001827.hg.1) #10.95106
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-38475535'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17001827.hg.1, x = as.factor(x$`AX-38475535`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))+expand_limits(y=c(3,11))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-38475535`), y=x$TC17001827.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="ABCA6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ expand_limits(y=c(3,11))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-38475535`, y=x$TC17001827.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3,11))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC17001827.hg.1", groupnames =c("diagnosis","`AX-38475535`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-38475535` <- as.factor(df$`AX-38475535`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-38475535`), y=df$TC17001827.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3,11))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-38475535`), y=df$TC17001827.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC17001827.hg.1-sd, ymax=TC17001827.hg.1+sd),width=.2)+ expand_limits(y=c(3,11))

p2 <-p1 + labs(y="ABCA6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("ABCA6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL68_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#69 AX-11371289	TC07001734.hg.1	69	-6.764279753	8.65E-11	0.006647151	-0.190926839	PUS7 	chr7	105080108	105162714	Coding	rs2158937	19	40132472	T	C	C	T


x<-merge_v3$`AX-11371289`
# 0   1   2 
# 2  25 248 
#UCDSS.610	4	2	5	82	1	0	1	-0.0327288	-0.0299031	-0.0117765	0.00424891	1
#UCDSS.744	8	2	6	55	1	0	0	-0.00926499	0.0294275	0.0985531	-0.0119912	4
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC07001734.hg.1~`AX-11371289`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-11371289`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC07001734.hg.1 ~ `AX-11371289` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-11371289` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.77130 -0.10832 -0.00984  0.09938  2.04665 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.964765   0.189329  26.223  < 2e-16 ***
# `AX-11371289`            0.257327   0.080068   3.214  0.00147 ** 
# age                     -0.003849   0.001107  -3.478  0.00059 ***
# race                     0.005633   0.016539   0.341  0.73369    
# gender                   0.047626   0.029210   1.630  0.10420    
# diagnosis                0.365804   0.059090   6.191 2.29e-09 ***
# diagnosissubtype         0.002066   0.016033   0.129  0.89759    
# pc1                      0.082018   0.294770   0.278  0.78104    
# pc2                     -0.082084   0.258245  -0.318  0.75085    
# pc3                      0.073202   0.290696   0.252  0.80138    
# pc4                     -0.157229   0.231998  -0.678  0.49855    
# `AX-11371289`:diagnosis -0.190927   0.028226  -6.764 8.65e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2286 on 263 degrees of freedom
# Multiple R-squared:  0.2445,	Adjusted R-squared:  0.2129 
# F-statistic: 7.739 on 11 and 263 DF,  p-value: 1.371e-11

min(merge_v3$TC07001734.hg.1) #4.973848
max(merge_v3$TC07001734.hg.1) #8.414429
```

```{r}
#x <-merge_v3[!is.na(merge_v3$'AX-11371289'), ] #remove samples with NA
p1 <-ggplot(aes(y = merge_v3$TC07001734.hg.1, x = as.factor(merge_v3$`AX-11371289`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PUS7 expression", x = "rs2158937 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("PUS7")," " , "expression")))+expand_limits(y=c(4,9))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-11371289`), y=merge_v3$TC07001734.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="PUS7 expression", x = "rs2158937 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("PUS7")," " , "expression")))+ expand_limits(y=c(4,9))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-11371289`, y=merge_v3$TC07001734.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,9))

p2 <-p1 + labs(y="PUS7 expression", x = "rs2158937 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PUS7")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(merge_v3,varname = "TC07001734.hg.1", groupnames =c("diagnosis","`AX-11371289`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11371289` <- as.factor(df$`AX-11371289`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-11371289`), y=df$TC07001734.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC07001734.hg.1-sd, ymax=TC07001734.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,9))

p2 <-p1 + labs(y="PUS7 expression", x = "rs2158937 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PUS7")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-11371289`), y=df$TC07001734.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC07001734.hg.1-sd, ymax=TC07001734.hg.1+sd),width=.2)+ expand_limits(y=c(4,9))

p2 <-p1 + labs(y="PUS7 expression", x = "rs2158937 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PUS7")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL69_scattererrorbar.jpg",width = 6, height = 4)
```


```{r}
#Trans-eQTL in "C:\eQTL\R_eQTL_Stroke\Input_Output\eQTL_result\LINEAR_CROSS\mRNA\Samples_275\covariate_subdiagnosis_sex_race_age_diagnosis_pc"

#70 AX-34315977	TC15001756.hg.1	70	-6.752410844	9.28E-11	0.007025676	-0.095151553	AP3B2 	chr15	83328033	83378666	Coding	rs58232949	3	40693259	A	G	G	A


x<-merge_v3$`AX-34315977`
# 0   1   2 NA
# 6  41 222 6
#UCDSS.180	7	2	3	39	1	0	0	0.121282	-0.0321256	-0.01014	-0.0427665	4
#UCDSS.708	8	2	3	41	1	1	1	0.161344	-0.0350806	0.0073334	-0.0420488	4
#UCDSS.774	6	1	5	58	0	0	1	-0.0294796	-0.0289014	-0.0121061	-0.000155022	1
#UCDSS.782	6	1	3	42	0	0	0	0.164189	-0.0371417	0.00127105	-0.0370388	1
#UCDSS.863	8	1	3	41	1	1	0	0.142348	-0.0309692	0.00612607	-0.0206519	4
#UCSFCJ.005	3	2	3	53	1	1	0	0.102322	-0.0350191	-0.0147273	-0.0238308	1

```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC15001756.hg.1~`AX-34315977`+age+race+gender+diagnosis+diagnosissubtype+pc1+pc2+pc3+pc4+`AX-34315977`*diagnosis, data = merge_v3)
summary (fit)
#Call:
# lm(formula = TC15001756.hg.1 ~ `AX-34315977` + age + race + gender + 
#     diagnosis + diagnosissubtype + pc1 + pc2 + pc3 + pc4 + `AX-34315977` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.48800 -0.08134 -0.02188  0.04516  0.88179 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              3.9420036  0.1065773  36.987  < 2e-16 ***
# `AX-34315977`            0.1376594  0.0396295   3.474 0.000602 ***
# age                      0.0013937  0.0007546   1.847 0.065889 .  
# race                     0.0004066  0.0108607   0.037 0.970165    
# gender                   0.0043248  0.0193831   0.223 0.823618    
# diagnosis                0.1867081  0.0294978   6.330 1.09e-09 ***
# diagnosissubtype        -0.0070253  0.0109819  -0.640 0.522930    
# pc1                      0.0299547  0.2066237   0.145 0.884846    
# pc2                     -0.1238823  0.1736929  -0.713 0.476353    
# pc3                      0.1527684  0.1920020   0.796 0.426964    
# pc4                     -0.0440167  0.1546338  -0.285 0.776141    
# `AX-34315977`:diagnosis -0.0946182  0.0142255  -6.651 1.74e-10 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1503 on 257 degrees of freedom
#   (6 observations deleted due to missingness)
# Multiple R-squared:  0.2005,	Adjusted R-squared:  0.1663 
# F-statistic:  5.86 on 11 and 257 DF,  p-value: 1.758e-08

min(merge_v3$TC15001756.hg.1) #4.034248
max(merge_v3$TC15001756.hg.1) #5.587059
```

```{r}
x <-merge_v3[!is.na(merge_v3$'AX-34315977'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC15001756.hg.1, x = as.factor(x$`AX-34315977`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="AP3B2 expression", x = "rs58232949 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("AP3B2")," " , "expression")))+expand_limits(y=c(4,6))


ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70.pdf",width = 8, height = 5)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70.jpg",width = 8, height = 5)
```

```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-34315977`), y=x$TC15001756.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="AP3B2 expression", x = "rs58232949 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ expand_limits(y=c(4,6))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_point.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_point.jpg",width = 6, height = 4)
```  

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-34315977`, y=x$TC15001756.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs58232949 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_regline.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_regline.jpg",width = 6, height = 4)

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis) used x to have no NA in data
df <-data_summary(x,varname = "TC15001756.hg.1", groupnames =c("diagnosis","`AX-34315977`")) #NA in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-34315977` <- as.factor(df$`AX-34315977`)

```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=factor(df$`AX-34315977`), y=df$TC15001756.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs58232949 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_boxerrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_boxerrorbar.jpg",width = 6, height = 4)
```


```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=as.factor(df$`AX-34315977`), y=df$TC15001756.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC15001756.hg.1-sd, ymax=TC15001756.hg.1+sd),width=.2)+ expand_limits(y=c(4,6))

p2 <-p1 + labs(y="AP3B2 expression", x = "rs58232949 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("AP3B2")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+ scale_x_discrete( breaks = c("0", "1", "2"),labels = c("GG", "GA", "AA"))

ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_scattererrorbar.pdf",width = 6, height = 4)
ggsave(p2, file="../../Input_Output/Plot/Sample_275/trans-eQTL70_scattererrorbar.jpg",width = 6, height = 4)
```

```{r}
#having multiple genes for same SNO for trans-eQTL
#Align multiple ggplot2 graphs with a common x axis and different y axes, each with different y-axis labels.
#used this https://gist.github.com/tomhopper/faa24797bb44addeba79

#check the positive or negative of beta, case is intercept then see the direction of control samples

#library("reshape2")
#library(grid)
#AX-16372367
x <-merge_v3[!is.na(merge_v3$'AX-16372367'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17001827.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))

p3 <-ggplot(aes(y = x$TC04001036.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="CLNK expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))

p5 <-ggplot(aes(y = x$TC02000318.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p6<- p5 + labs(y="EML6 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))

p7 <-ggplot(aes(y = x$TC0X001190.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p8<- p7 + labs(y="POF1B expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("POF1B")," " , "expression")))

p9 <-ggplot(aes(y = x$TC07000736.hg.1, x = as.factor(x$`AX-16372367`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p10 <-p9 + labs(y="WNT16 expression", x = "rs950391 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("WNT16")," " , "expression")))

#save 
grid.newpage()
grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6),ggplotGrob(p8),ggplotGrob(p10), size = "last"))
ggsave("multi_rs950391_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6),ggplotGrob(p8),ggplotGrob(p10), size = "last")),width = 12, height = 15)
ggsave("multi_rs950391_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6),ggplotGrob(p8),ggplotGrob(p10), size = "last")),width = 12, height = 15)
```

```{r}
#AX-11393384
p1 <-ggplot(aes(y = merge_v3$TC17001827.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))

p3 <-ggplot(aes(y = merge_v3$TC04001036.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="CLNK expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))

p5<-ggplot(aes(y = merge_v3$TC02000318.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p6 <-p5 + labs(y="EML6 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))

p7 <-ggplot(aes(y = merge_v3$TC01002971.hg.1, x = as.factor(merge_v3$`AX-11393384`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p8 <-p7 + labs(y="SLC16A4 expression", x = "rs2464504 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("SLC16A4")," " , "expression")))

#save
ggsave("multi_rs2464504_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6),ggplotGrob(p8), size = "last")),width = 12, height = 15)
ggsave("multi_rs2464504_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6),ggplotGrob(p8), size = "last")),width = 12, height = 15)
```

```{r}
#AX-11173316
p1 <-ggplot(aes(y = merge_v3$TC17001827.hg.1, x = as.factor(merge_v3$`AX-11173316`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <- p1 + labs(y="ABCA6 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))

p3 <-ggplot(aes(y = merge_v3$TC01002971.hg.1, x = as.factor(merge_v3$`AX-11173316`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p4 <- p3 + labs(y="SLC16A4 expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("SLC16A4")," " , "expression")))

p5 <-ggplot(aes(y = merge_v3$TC04001036.hg.1, x = as.factor(merge_v3$`AX-11173316`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p6 <- p5 + labs(y="CLNK expression", x = "rs11758921 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))

#save
ggsave("multi_rs11758921_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)
ggsave("multi_rs11758921_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)

```

```{r}
#AX-16372292
x <-merge_v3[!is.na(merge_v3$'AX-16372292'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17001827.hg.1, x = as.factor(x$`AX-16372292`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))

p3 <-ggplot(aes(y = x$TC02000318.hg.1, x = as.factor(x$`AX-16372292`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="EML6 expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))

p5 <-ggplot(aes(y = x$TC04001036.hg.1, x = as.factor(x$`AX-16372292`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p6 <-p5 + labs(y="CLNK expression", x = "rs2369519 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))

#save
ggsave("multi_rs2369519_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)
ggsave("multi_rs2369519_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)

```

```{r}
#AX-15355229
x <-merge_v3[!is.na(merge_v3$'AX-15355229'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC17000375.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZNF207 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ZNF207")," " , "expression")))

p3 <-ggplot(aes(y = x$TC01001640.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="PTPRC expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("PTPRC")," " , "expression")))

p5 <-ggplot(aes(y = x$TC18000998.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p6 <-p5 + labs(y="SMAD4 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("SMAD4")," " , "expression")))

p7 <-ggplot(aes(y = x$TC0M000010.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p8 <-p7 + labs(y="TC0M000010.hg.1 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("TC0M000010.hg.1")," " , "expression")))

p9 <-ggplot(aes(y = x$TC02004959.hg.1, x = as.factor(x$`AX-15355229`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p10 <-p9 + labs(y="MORN2 expression", x = "rs1063632 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("MORN2")," " , "expression")))

#save
ggsave("multi_rs1063632_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), ggplotGrob(p8), ggplotGrob(p10),size = "last")),width = 12, height = 15)
ggsave("multi_rs1063632_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), ggplotGrob(p8), ggplotGrob(p10), size = "last")),width = 12, height = 15)
```

```{r}
#AX-51270082
x <-merge_v3[!is.na(merge_v3$'AX-51270082'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC0M000010.hg.1, x = as.factor(x$`AX-51270082`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="TC0M000010.hg.1 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("TC0M000010.hg.1")," " , "expression")))

p3 <-ggplot(aes(y = x$TC01001640.hg.1, x = as.factor(x$`AX-51270082`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="PTPRC expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("PTPRC")," " , "expression")))

p5 <-ggplot(aes(y = x$TC18000998.hg.1, x = as.factor(x$`AX-51270082`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p6 <-p5 + labs(y="SMAD4 expression", x = "rs6787784 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("SMAD4")," " , "expression")))

#save
ggsave("multi_rs6787784_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)
ggsave("multi_rs6787784_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)
```

```{r}
#AX-38475535
x <-merge_v3[!is.na(merge_v3$'AX-38475535'), ]
p1 <-ggplot(aes(y = x$TC04001036.hg.1, x = as.factor(x$`AX-38475535`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="CLNK expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("CLNK")," " , "expression")))

p3 <-ggplot(aes(y = x$TC02000318.hg.1, x = as.factor(x$`AX-38475535`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="EML6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("EML6")," " , "expression")))

p5 <-ggplot(aes(y = x$TC17001827.hg.1, x = as.factor(x$`AX-38475535`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p6 <-p5 + labs(y="ABCA6 expression", x = "rs117781420 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))

#save
ggsave("multi_rs117781420_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)
ggsave("multi_rs117781420_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), ggplotGrob(p6), size = "last")),width = 12, height = 15)

```

```{r}
#AX-38578625
p1 <-ggplot(aes(y = merge_v3$TC01000514.hg.1, x = as.factor(merge_v3$`AX-38578625`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ZNF684 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CA", "2"="AA"))+scale_y_continuous(expression(paste( italic("ZNF684")," " , "expression")))

p3 <-ggplot(aes(y = merge_v3$TC09001542.hg.1, x = as.factor(merge_v3$`AX-38578625`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="C5 expression", x = "rs148991762 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CA", "2"="AA"))+scale_y_continuous(expression(paste( italic("C5")," " , "expression")))

#save
ggsave("multi_rs148991762_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
ggsave("multi_rs148991762_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
```

```{r}
p1 <-ggplot(aes(y = merge_v3$TC17001827.hg.1, x = as.factor(merge_v3$`AX-11577079`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="ABCA6 expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("ABCA6")," " , "expression")))

p3 <-ggplot(aes(y = merge_v3$TC04001036.hg.1, x = as.factor(merge_v3$`AX-11577079`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="CLNK expression", x = "rs6665585 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "AA", "1" = "AG", "2"="GG"))+scale_y_continuous(expression(paste( italic("CLNK ")," " , "expression")))

#save
ggsave("multi_rs6665585_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
ggsave("multi_rs6665585_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
```

```{r}
#AX-38554997
p1 <-ggplot(aes(y = merge_v3$TC01001640.hg.1, x = as.factor(merge_v3$`AX-38554997`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PTPRC expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("PTPRC")," " , "expression")))

p3 <-ggplot(aes(y = merge_v3$TC0M000010.hg.1, x = as.factor(merge_v3$`AX-38554997`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3 ,xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="TC0M000010.hg.1 expression", x = "rs149536248 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("TC0M000010.hg.1")," " , "expression")))

#save
ggsave("multi_rs149536248_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
ggsave("multi_rs149536248_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
```

```{r}
#AX-42755741
x <-merge_v3[!is.na(merge_v3$'AX-42755741'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC09000560.hg.1, x = as.factor(x$`AX-42755741`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="UGCG expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("UGCG")," " , "expression")))


p3 <-ggplot(aes(y = x$TC01002971.hg.1, x = as.factor(x$`AX-42755741`), fill = as.factor(x$diagnosis)), data = x ,xlim(c(0,1,2))) + geom_boxplot()
p4 <-p3 + labs(y="SLC16A4 expression", x = "rs12399124 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "GG", "1" = "GA", "2"="AA"))+scale_y_continuous(expression(paste( italic("SLC16A4")," " , "expression")))

#save
ggsave("multi_rs12399124_trans-eQTL.png", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
ggsave("multi_rs12399124_trans-eQTL.pdf", path = "../../Input_Output/Plot/Sample_275/Multi_Figure/", plot = grid.draw(rbind(ggplotGrob(p2), ggplotGrob(p4), size = "last")),width = 12, height = 15)
```

```{r}
#check the eQTL for the High LD SNPs with the 32 SNPs from Stroke GWAS, For now, I don't remove any SNPs based on MAF, P-value, CR.
#import the SNP file, only this file is different
#uased all others matrix for 275 samples
SNPs_Picked_From_LD_v2_Unique <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Samples_275/SNP_MatchWith32StrokeGWAS_05-13-20/SNPs_Picked_From_LD_v2_Unique", row.names=1, stringsAsFactors = FALSE ) #11262   275

write.table(SNPs_Picked_From_LD_v2_Unique, file = "../../Input_Output/Samples_275/SNP_MatchWith32StrokeGWAS_05-13-20/SNPs_Picked_From_LD_v2_Unique.txt", col.names = T, sep = "\t")

SNP_file_name = paste(base.dir, "/Input_Output/Samples_275/SNP_MatchWith32StrokeGWAS_05-13-20/SNPs_Picked_From_LD_v2_Unique.txt", sep="") #11262
covariates_file_name = paste(base.dir, "/Input_Output/Samples_275/Covariate_Final_NO_Risk_Factors.txt", sep="")
```

```{r}
#Note to myself
#SNPs before filtering: 11262 (keep this results)
#SNPs after filtering: 10905 (if use this set, no cis and trans for 1st try and ~1700 cis, no trans for 2nd try)
```

```{r}
#1st Try
useModel =  modelLINEAR_CROSS

cisDist = 1e6
eQTL = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = output_file_name_tra,
pvOutputThreshold = 1e-10,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, 
output_file_name.cis = output_file_name_cis,
pvOutputThreshold.cis = 1e-5,
snpspos = snpspos, 
genepos = genepos,
cisDist = cisDist,
pvalue.hist = "qqplot",
min.pv.by.genesnp = FALSE,
noFDRsaveMemory = FALSE)

unlink(output_file_name_tra);
unlink(output_file_name_cis); 
```


```{r}
#save the result
write.csv(eQTL$trans$eqtls,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/SNP_MatchWith32StrokeGWAS/eQTL_trans_1e6.csv")
```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/SNP_MatchWith32StrokeGWAS/eQTL_trans_1e6.csv")
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #20

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#20
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/SNP_MatchWith32StrokeGWAS/SNPeQTL_trans_1e6_merge.csv") 
```

```{r}
#2nd Try
useModel =  modelLINEAR_CROSS

cisDist = 1e6
eQTL = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = output_file_name_tra,
pvOutputThreshold = 1e-10,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, 
output_file_name.cis = output_file_name_cis,
pvOutputThreshold.cis = 1e-02,
snpspos = snpspos, 
genepos = genepos,
cisDist = cisDist,
pvalue.hist = "qqplot",
min.pv.by.genesnp = FALSE,
noFDRsaveMemory = FALSE)

unlink(output_file_name_tra);
unlink(output_file_name_cis); 
```


```{r}
#save the result
write.csv(eQTL$cis$eqtls,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/SNP_MatchWith32StrokeGWAS/eQTL_cis_1e6_pvalue1e-2.csv")
```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_cis_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/SNP_MatchWith32StrokeGWAS/eQTL_cis_1e6_pvalue1e-2.csv")
eQTL_cis_1e6_merge <- merge(eQTL_cis_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #1807

eQTL_cis_1e6_merge_v2 <- merge(eQTL_cis_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#20
write.csv(eQTL_cis_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/Samples_275/covariate_subdiagnosis_sex_race_age_diagnosis_pc/SNP_MatchWith32StrokeGWAS/SNPeQTL_cia_1e6_pvalue1e-2_merge.csv") 
```
