---
title: "eQTL_Stroke_MatrixEQTL_022019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#start with installing MatrixEQTL "http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/runit.html"
#install.packages("MatrixEQTL")
#library("MatrixEQTL")
base.dir <- ("C:/eQTL/R_eQTL_Stroke/eQTL_Stroke")
#set the parameters such as linearcross model and names of the genotype and expression data files
# based on this doc, "http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/html/Matrix_eQTL_main.html"
#Set useModel = modelLINEAR_CROSS to add a new term to the modelequal to the product of genotype and the last covariate; the significance of this term is then tested using t-statistic.
useModel = modelLINEAR_CROSS # modelANOVA or modelLINEAR or modelLINEAR_CROSS 
SNP_file_name = paste(base.dir, "/Input_Output/SNP.txt", sep="")
expression_file_name = paste(base.dir, "/Input_Output/GE.txt", sep="") 
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final.txt", sep="")
gene_location_file_name = paste(base.dir, "/Input_Output/Gene_Location.txt", sep="")
snps_location_file_name = paste(base.dir, "/Input_Output/SNP_Location.txt", sep="")
output_file_name_cis = tempfile()
output_file_name_tra = tempfile()
```

```{r}
#The p-value threshold determines which gene-SNP associations are saved in the output file output_file_name. Note that for larger datasets the threshold should be lower. Setting the threshold to a high value for a large dataset may cause excessively large output files.
#pvOutputThreshold_cis = 2e-2 #defualt
#pvOutputThreshold_tras = 1e-2 #defualt
pvOutputThreshold_cis = 0
pvOutputThreshold_trans = 1e-20 #Only associations significant at this level will be saved. After talking with Brad decide to set as 1e-20
```

```{r}
# Finally, define the covariance matrix for the error term. This parameter is rarely used. If the covariance matrix is a multiple of identity, set it to numeric().
errorCovariance = numeric() #have to be sure about this?????

```

```{r}
# Distance for local gene-SNP pairs, numeric. SNP-gene pairs within this distance are considered local. The distance is measured from the nearest end of the gene. SNPs within a gene are always considered local.
cisDist = 1e6 #for test
```

```{r}
# The next section of the sample code contains three very similar parts loading the files with genotype, gene expression, and covariates. In each part one can set the file delimiter (i.e. tabulation "\t", comma ",", or space " "), the string representation for missing values, the number of rows with column labels, and the number of columns with row labels.	Finally, one can change the number of the variables in a slice for the file reading procedure (do not change if not sure).
## Load genotype data
snps = SlicedData$new() #SlicedData object with genotype information.
snps$fileDelimiter = "\t"      # the TAB character
snps$fileOmitCharacters = "NA" # denote missing values;
snps$fileSkipRows = 1          # one row of column labels
snps$fileSkipColumns = 1       # one column of row labels
snps$fileSliceSize = 710000      # read file in slices of over 600000 rows
snps$LoadFile(SNP_file_name) #Number of columns: 351, Number of rows: 628679
#SNP_Sample[SNP_Sample == -1] <- NA # -1 in our data is equal missing data

```

```{r}
# Load gene expression data
gene = SlicedData$new(); #SlicedData object with gene expression information. Must have columns matching those of snps, create gene including row and column
gene$fileDelimiter = "\t"      # the TAB characte
gene$fileOmitCharacters = "NA" # denote missing values;
gene$fileSkipRows = 1          # one row of column labels
gene$fileSkipColumns = 1       # one column of row labels
gene$fileSliceSize = 710000      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name) #Number of columns: 351, Number of rows: 70523 
```

```{r}
# Load covariates
cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t"      # the TAB character
cvrt$fileOmitCharacters = "NA" # denote missing values;
cvrt$fileSkipRows = 1         # one row of column labels
cvrt$fileSkipColumns = 1      # one column of row labels
if(length(covariates_file_name)>0) {
cvrt$LoadFile(covariates_file_name)
}
```
```{r}
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE);

```

```{r}

```

