---
title: "eQTL_Stroke_MatrixEQTL_mRNA_032519"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#by checking the coding type of the probset ID (TC) in "gene_expression_annotation_351.SST-RMA-GENE-FULL - Group 1", found some of them as a non-coding RNA, snoRNA, lncRNA. So in this script we want to filter genes based on biotype == mRNA to see as a real coding genes

```{r}
#cat gene_expression_annotation_351.SST-RMA-GENE-FULL\ -\ Group\ 1.TXT | grep "mRNA" > gene_expression_annotation_351.SST-RMA_mRNA_like.txt, then remove mRNA like such as lncoding RNA by keeping only mRNA and set the new output's name to gene_expression_annotation_351.SST-RMA_mRNA.txt #(27337 genes)
gene_expression_annotation_351.SST.RMA_mRNA.txt <- read.delim("C:/eQTL/GENE/Picked_Files/gene_expression_351/gene_expression_annotation_351.SST-RMA_mRNA.txt.csv", row.names=1) #20328 364
colnames(gene_expression_annotation_351.SST.RMA_mRNA.txt) <- sub(x = colnames(gene_expression_annotation_351.SST.RMA_mRNA.txt), pattern = "_HTA*.*", replacement ="") 
gene_expression_annotation_351.SST.RMA_mRNA_small <- gene_expression_annotation_351.SST.RMA_mRNA.txt[,c(1:351)] # dim 20328 351
GE<- gene_expression_annotation_351.SST.RMA_mRNA_small[,c(-1,-2,-3)] # 20328   351
write.table(gene_expression_annotation_351.SST.RMA_mRNA_small, file = "../../Input_Output/GE_mRNA_20328.txt",col.names =T, sep = "\t") 
```

```{r}
expression_file_name = paste(base.dir, "/Input_Output/GE_mRNA_20328.txt", sep="") 
#back to other script to run eQTL function for this GE dataset
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e6.csv")

```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e6.csv")
#load( "../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #62

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#62
write.csv(eQTL_cis_dist_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e6.csv")
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #33

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#33
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e6_merge.csv")
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_5e5.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_5e5.csv")
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e7.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e7.csv")

```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_cis_dist_1e7 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e7.csv")
#load( "../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e7_merge <- merge(eQTL_cis_dist_1e7,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #513

eQTL_cis_dist_1e7_merge_v2 <- merge(eQTL_cis_dist_1e7_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#513
write.csv(eQTL_cis_dist_1e7_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e7_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_trans_1e7 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e7.csv")
eQTL_trans_1e7_merge <- merge(eQTL_trans_1e7,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #33

eQTL_trans_1e7_merge_v2 <- merge(eQTL_trans_1e7_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#33
write.csv(eQTL_trans_1e7_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e7_merge.csv")
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_5e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_5e6.csv")

```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_cis_dist_5e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_5e6.csv")
#load( "../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_5e6_merge <- merge(eQTL_cis_dist_5e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #296

eQTL_cis_dist_5e6_merge_v2 <- merge(eQTL_cis_dist_5e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#62
write.csv(eQTL_cis_dist_5e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_5e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_trans_5e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_5e6.csv")
eQTL_trans_5e6_merge <- merge(eQTL_trans_5e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #33

eQTL_trans_5e6_merge_v2 <- merge(eQTL_trans_5e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#33
write.csv(eQTL_trans_5e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_5e6_merge.csv")
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_5e5.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_5e5.csv")
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_2e7.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_2e7.csv")

```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_cis_dist_2e7 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_2e7.csv")
#load( "../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_2e7_merge <- merge(eQTL_cis_dist_2e7,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #876

eQTL_cis_dist_2e7_merge_v2 <- merge(eQTL_cis_dist_2e7_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#876
write.csv(eQTL_cis_dist_2e7_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_2e7_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_trans_2e7 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_2e7.csv")
eQTL_trans_2e7_merge <- merge(eQTL_trans_2e7,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #33

eQTL_trans_2e7_merge_v2 <- merge(eQTL_trans_2e7_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#33
write.csv(eQTL_trans_2e7_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_2e7_merge.csv")
```


```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_5e7.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_5e7.csv")
#merge the file like others
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e8.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e8.csv")

```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_cis_dist_1e8 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e8.csv")
#load( "../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e8_merge <- merge(eQTL_cis_dist_1e8,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #2667

eQTL_cis_dist_1e8_merge_v2 <- merge(eQTL_cis_dist_1e8_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#2667
write.csv(eQTL_cis_dist_1e8_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_1e8_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_trans_1e8 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e8.csv")
eQTL_trans_1e8_merge <- merge(eQTL_trans_1e8,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #29

eQTL_trans_1e8_merge_v2 <- merge(eQTL_trans_1e8_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#29
write.csv(eQTL_trans_1e8_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e8_merge.csv")
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_2e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_2e6.csv")

```


```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_cis_dist_2e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_2e6.csv")
#load( "../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_2e6_merge <- merge(eQTL_cis_dist_2e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #114

eQTL_cis_dist_2e6_merge_v2 <- merge(eQTL_cis_dist_2e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#114
write.csv(eQTL_cis_dist_2e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_cis_dist_2e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
#start with coding folder
eQTL_trans_2e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_2e6.csv")
eQTL_trans_2e6_merge <- merge(eQTL_trans_2e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #33

eQTL_trans_2e6_merge_v2 <- merge(eQTL_trans_2e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#33
write.csv(eQTL_trans_2e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_2e6_merge.csv")
```

# redo analysis for x-linked and y-linked genes from mRNA list not all coding and noncoding genes 

```{r}
#cat gene_expression_annotation_351.SST-RMA-GENE-FULL\ -\ Group\ 1.TXT | grep "mRNA" > gene_expression_annotation_351.SST-RMA_mRNA_like.txt, then remove mRNA like such as lncoding RNA by keeping only mRNA and set the new output's name to gene_expression_annotation_351.SST-RMA_mRNA.txt #(27337 genes)
gene_expression_annotation_351.SST.RMA_mRNA_sex <- read.csv("C:/eQTL/GENE/Picked_Files/gene_expression_351/gene_expression_annotation_351.SST-RMA_mRNA_sex.csv", row.names=1) #898 364

colnames(gene_expression_annotation_351.SST.RMA_mRNA_sex) <- sub(x = colnames(gene_expression_annotation_351.SST.RMA_mRNA_sex), pattern = "_HTA*.*", replacement ="") 
GE <- gene_expression_annotation_351.SST.RMA_mRNA_sex[,c(1:351)] # dim 898 351
write.table(GE, file = "../../Input_Output/GE_mRNA_sex_898.txt",col.names =T, sep = "\t") 
```


```{r}
expression_file_name = paste(base.dir, "/Input_Output/GE_mRNA_sex_898.txt", sep="") 
#back to other script to run eQTL function for this GE dataset
```

```{r}
write.csv(eQTL$trans$eqtls, "../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_5e6.csv")
write.csv(eQTL$cis$eqtls, "../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dis_5e6.csv")
```

```{r}
#sex folder in mRNA folder to compare with another sex folder
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex//eQTL_cis_dist_1e6.csv") #1
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #1

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#62
write.csv(eQTL_cis_dist_1e6_merge_v2,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dist_1e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_1e6.csv") #5
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #5

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#5
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_1e6_merge.csv") 
```

```{r}
#sex folder in mRNA folder to compare with another sex folder
eQTL_cis_dist_2e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dist_2e6.csv") #3
eQTL_cis_dist_2e6_merge <- merge(eQTL_cis_dist_2e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #3

eQTL_cis_dist_2e6_merge_v2 <- merge(eQTL_cis_dist_2e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#62
write.csv(eQTL_cis_dist_2e6_merge_v2,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dist_2e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_2e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_2e6.csv") #5
eQTL_trans_2e6_merge <- merge(eQTL_trans_2e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #5

eQTL_trans_2e6_merge_v2 <- merge(eQTL_trans_2e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#5
write.csv(eQTL_trans_2e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_2e6_merge.csv")
```

```{r}
#sex folder in mRNA folder to compare with another sex folder
eQTL_cis_dist_5e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dist_5e6.csv") #6
eQTL_cis_dist_5e6_merge <- merge(eQTL_cis_dist_5e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #6

eQTL_cis_dist_5e6_merge_v2 <- merge(eQTL_cis_dist_5e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#6
write.csv(eQTL_cis_dist_5e6_merge_v2,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dist_5e6_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_5e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_5e6.csv") #5
eQTL_trans_5e6_merge <- merge(eQTL_trans_5e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #5

eQTL_trans_5e6_merge_v2 <- merge(eQTL_trans_5e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#5
write.csv(eQTL_trans_5e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_5e6_merge.csv")
```

```{r}
#sex folder in mRNA folder to compare with another sex folder
eQTL_cis_dist_1e7 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dist_1e7.csv") #6
eQTL_cis_dist_1e7_merge <- merge(eQTL_cis_dist_1e7,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #6

eQTL_cis_dist_1e7_merge_v2 <- merge(eQTL_cis_dist_1e7_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#6
write.csv(eQTL_cis_dist_1e7_merge_v2,"../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_cis_dist_1e7_merge.csv")
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e7 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_1e7.csv") #5
eQTL_trans_1e7_merge <- merge(eQTL_trans_1e7,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #5

eQTL_trans_1e7_merge_v2 <- merge(eQTL_trans_1e7_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#5
write.csv(eQTL_trans_1e7_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/sex/eQTL_trans_1e7_merge.csv")
```

```{r}
#save the SNP and expression matrix which ws used for final analysis
#SNP_CRmore95_MAFmore0.05_HWEmore0.0000005 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", row.names=1)
#GE_mRNA_20328 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/GE_mRNA_20328.txt", row.names=1)
#Covariates_Final_No_PC_NO_Subtype <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Covariates_Final_No_PC_NO_Subtype.txt", row.names=1)
save(GE_mRNA_20328, SNP_CRmore95_MAFmore0.05_HWEmore0.0000005, file = "../../Input_Output/RData/GE_mRNA_SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.RData")
```

```{r}
#compare lm() result with MatrixeQTL function "https://rdrr.io/cran/MatrixEQTL/man/modelLINEAR_CROSS.html"
#snp <- as.matrix(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005)
#ge <- as.matrix(GE_mRNA_20328)
#cvr <-as.matrix(Covariates_Final_No_PC_NO_Subtype)
#for lm() finction needs to transpose dataset
#snps.mat <- t(snp) # 351 273688
#gene.mat <-t(ge) #  351 20328
#cvrt.mat <- t(cvr) #351 4
# snps.mat2 <- as.matrix(snps.mat[,1])
# class(snps.mat2)
# gene.mat2 <- as.matrix(gene.mat[,1])


beta = eQTL$all$eqtls$beta;
tstat = eQTL$all$eqtls$statistic;
pvalue = eQTL$all$eqtls$pvalue;
rez = c(beta = beta, tstat = tstat, pvalue = pvalue)
# And compare to those from the linear regression in R
{
    cat('\n\n Matrix eQTL: \n'); 
    print(rez);
    cat('\n R summary(lm()) output: \n')
   lmodel = lm(gene.mat2~ snps.mat2 + cvrt.mat + snps.mat2*cvrt.mat[,4]); #4 refers to the last column. here means diagnosis
                   
    lmout = tail(summary(lmodel)$coefficients,1)[, c("Estimate", "t value", "Pr(>|t|)")];
    print( tail(lmout) );
}
# Results from Matrix eQTL and 'lm' must agree
stopifnot(all.equal(lmout, rez, check.attributes = FALSE));
#should see something like as follows:
#  Matrix eQTL: 
#         beta        tstat       pvalue 
# 1.092349e+00 1.258344e+01 2.844406e-08 
# 
#  R summary(lm()) output: 
#     Estimate      t value     Pr(>|t|) 
# 1.092349e+00 1.258344e+01 2.844406e-08 
```

```{r}
par(mfrow=c(2,2))
plot(lmodel$coefficients)
abline(lmodel)
```

```{r}
cvrt2 <- as.data.frame(cvrt.mat)
cvrt.stroke <- cvrt2[cvrt2$diagnosis == 4,] #subset only stroke group #183
cvrt.control <- cvrt2[cvrt2$diagnosis == 1,] #subset only control group #168
snps.mat2.stroke <-as.matrix(snps.mat2[rownames(cvrt.stroke),])
snps.mat2.control <-as.matrix(snps.mat2[rownames(cvrt.control),])
#gene.mat2.stroke <-as.matrix(gene.mat2[rownames(cvrt.stroke),])
gene.mat2.control <-as.matrix(gene.mat2[rownames(cvrt.control),])
plot(snps.mat2.control,gene.mat2.control)
```


```{r}
#To create dataset for plotting interaction "https://cran.r-project.org/web/packages/sjPlot/vignettes/plot_interactions.html"
#SNP_CRmore95_MAFmore0.05_HWEmore0.0000005 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", row.names=1)
#View(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005)
SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t <- t(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005)
#dim(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#Covariates_Final_No_PC_NO_Subtype <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Covariates_Final_No_PC_NO_Subtype.txt", row.names=1)
#View(Covariates_Final_No_PC_NO_Subtype)
Covariates_Final_No_PC_NO_Subtype_t <-t(Covariates_Final_No_PC_NO_Subtype)
#class(Covariates_Final_No_PC_NO_Subtype_t)
#class(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#class(GE_mRNA_20328_t)
merge <- cbind(GE_mRNA_20328_t,SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#dim(merge)
#dim(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#dim(GE_mRNA_20328_t) 273688 + 20328
merge_v2 <- cbind(merge,Covariates_Final_No_PC_NO_Subtype_t)
#dim(merge_v2) #final dataset
#write.csv(merge_v2, "../../Input_Output/merge_final.txt", col.names = T, sep = "\t")
merge_v3 <- as.data.frame(merge_v2)
save(merge_v3, file = "../../Input_Output/merge_final.RData")
```

```{r}
#required library
# library(sjPlot)
# library(sjmisc)
# library(ggplot2)
```

```{r}
#make categorical
# merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
# #fit model with interaction picked one pair
# fit <- lm(TC01000003.hg.1~`AFFX-KIT-000001` +age+race+sex +`AFFX-KIT-000001`*diagnosis, data = merge_v3)
# #plotting
# plot_model(fit, type = "pred", terms = c("AFFX-KIT-000001","diagnosis"))
# plot_model(fit, type = "pred", terms = c("diagnosis","AFFX-KIT-000001"))

```

```{r}
#check the plot for one significant SNP-gene pair from eQTL_dist_1e6
#   X        snps            gene statistic       pvalue        FDR        beta
# 1   1 AX-14758778 TC04001305.hg.1 -5.985654 5.417431e-09 0.02260843 -0.45290434
# 2   2 AX-40592209 TC20001033.hg.1 -5.631148 3.723028e-08 0.06760264 -0.06751612
# 3   3 AX-11336441 TC02001095.hg.1  5.580785 4.859682e-08 0.06760264  0.09543061
# 4   4 AX-29907155 TC11001143.hg.1  5.452492 9.499488e-08 0.09910994  0.25824924
fit <- lm(TC11001143.hg.1~`AX-29907155` +age+race+sex +`AX-29907155`*diagnosis, data = merge_v3)
plot_model(fit, type = "pred", terms = c("AX-29907155","diagnosis"))
# 5   5 AX-39724859 TC01000514.hg.1  5.405367 1.211415e-07 0.10111133  0.15726662
# 6   6 AX-15964685 TC08000305.hg.1  5.198619 3.451509e-07 0.20825690  0.30886447
fit <- lm(TC08000305.hg.1~`AX-15964685` +age+race+sex +`AX-15964685`*diagnosis, data = merge_v3)
plot_model(fit, type = "pred", terms = c("AX-15964685","diagnosis"))
# 7   7 AX-11701696 TC18000998.hg.1 -5.196212 3.493176e-07 0.20825690 -0.14708963
fit <- lm(TC18000998.hg.1~`AX-11701696` +age+race+sex +`AX-11701696`*diagnosis, data = merge_v3)
plot_model(fit, type = "pred", terms = c("AX-11701696","diagnosis"))
#fit <- lm(TC04001305.hg.1~`AX-14758778` +age+race+sex +`AX-14758778`*diagnosis, data = merge_v3) # The beta with this command is not same as MatrixEQTL function because diagnosis is factor here
#lmodel = lm(TC04001305.hg.1~`AX-14758778` + cvrt.mat + `AX-14758778`*cvrt.mat[,4], data = merge_v3) ## The beta with this command is same because the diagnosis is not factor here
#plotting
# summary(lmodel)
# lm(formula = TC04001305.hg.1 ~ `AX-14758778` + cvrt.mat + `AX-14758778` * 
#     cvrt.mat[, 4], data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.8064 -0.3122 -0.0944  0.2065  7.3009 
# 
# Coefficients: (1 not defined because of singularities)
#                              Estimate Std. Error t value Pr(>|t|)    
# (Intercept)                  4.225558   0.269986  15.651  < 2e-16 ***
# `AX-14758778`                1.729474   0.231276   7.478 6.44e-13 ***
# cvrt.matsex                  0.014063   0.080626   0.174    0.862    
# cvrt.matrace                 0.036018   0.030259   1.190    0.235    
# cvrt.matage                  0.002444   0.002885   0.847    0.398    
# cvrt.matdiagnosis            0.013258   0.027731   0.478    0.633    
# cvrt.mat[, 4]                      NA         NA      NA       NA    
# `AX-14758778`:cvrt.mat[, 4] -0.451942   0.075903  -5.954 6.50e-09 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.7119 on 341 degrees of freedom
#   (3 observations deleted due to missingness)
# Multiple R-squared:  0.1494,	Adjusted R-squared:  0.1344 
#F-statistic:  9.98 on 6 and 341 DF,  p-value: 3.696e-10
#plot_model(fit, type = "pred", terms = c("AX-14758778","diagnosis")) #this function predict the model 
```

```{r}
#START
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
fit <- lm(TC04001305.hg.1~`AX-14758778` +age+race+sex+diagnosis +`AX-14758778`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001305.hg.1 ~ `AX-14758778` + age + race + sex + 
#     diagnosis + `AX-14758778` * diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.8064 -0.3122 -0.0944  0.2065  7.3009 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.225558   0.269986  15.651  < 2e-16 ***
# `AX-14758778`            1.729474   0.231276   7.478 6.44e-13 ***
# age                      0.002444   0.002885   0.847    0.398    
# race                     0.036018   0.030259   1.190    0.235    
# sex                      0.014063   0.080626   0.174    0.862    
# diagnosis                0.013258   0.027731   0.478    0.633    
# `AX-14758778`:diagnosis -0.451942   0.075903  -5.954 6.50e-09 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.7119 on 341 degrees of freedom
#   (3 observations deleted due to missingness)
# Multiple R-squared:  0.1494,	Adjusted R-squared:  0.1344 
# F-statistic:  9.98 on 6 and 341 DF,  p-value: 3.696e-10
plot_model(fit, type = "pred", terms = c("AX-14758778","diagnosis"))
```



```{r}
#check the plot for one significant SNP-gene pair from eQTL_dist_1e6

#plot exact point
#devtools::install_github("hadley/devtools")
#library(devtools)
#install_github("kassambara/easyGgplot2")
#library("ggplot2")
#library("easyGgplot2") do not this library for now
```

```{r}
#   X        snps            gene statistic       pvalue        FDR        beta
# 1   1 AX-14758778 TC04001305.hg.1 -5.985654 5.417431e-09 0.02260843 -0.45290434
#merge_v3$`AX-14758778`  
# 0   1   2 NA
# 312  33   3 3

x <-merge_v3[!is.na(merge_v3$'AX-14758778'), ] #remove samples with NA
p1 <-ggplot(aes(y = x$TC04001305, x = as.factor(x$`AX-14758778`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p1 + labs(title = "cis-eQTL", y="CXCL10 expression", x = "rs78046578 genotype")

```



```{r}
#to see the scatterplot for each SNP-gene pairs
p <- ggplot(merge_v3, aes(x=merge_v3$`AX-14758778`, y =merge_v3$TC04001305, color = diagnosis))+
  geom_boxplot()

p1 <- ggplot(data = merge_v3, aes(x=merge_v3$`AX-14758778`, y=merge_v3$TC04001305)) + geom_point(aes(colour =factor (merge_v3$diagnosis))) 
p1 <- ggplot(data = merge_v3, aes(x=merge_v3$`AX-14758778`, color = merge_v3$diagnosis, group=merge_v3$diagnosis)) + geom_point(aes(y=merge_v3$TC04001305), alpha =0.5)

#to see reg line for all sample
reg1<-lm(TC04001305.hg.1~`AX-14758778`, data = merge_v3)
with(merge_v3,plot(as.factor(`AX-14758778`),TC04001305.hg.1))
abline(reg1)
#again scatterplot
ggplot(x,aes(x=as.factor(x$`AX-14758778`), y=x$TC04001305), shape =as.factor(diagnosis)) +
    geom_point(aes(colour = as.factor(diagnosis)))
```


```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
p1 <- ggplot(x, aes(x=x$`AX-14758778`, y=x$TC04001305.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p1 + labs(title = "cis-eQTL", y="CXCL10 expression", x = "rs78046578 genotype")
```


```{r}
#function for having error bar for different groups

data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}

```


```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(x,varname = "TC04001305.hg.1", groupnames =c("diagnosis","`AX-14758778`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-14758778` <- as.factor(df$`AX-14758778`)
```

```{r}
#Barplot with error bars
p<- ggplot(df, aes(x=df$`AX-14758778`, y=df$TC04001305.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001305.hg.1-sd, ymax=TC04001305.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="CXCL10 expression", x = "rs78046578 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-14758778`, y=df$TC04001305.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001305.hg.1-sd, ymax=TC04001305.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="CXCL10 expression", x= "rs78046578 genotype")

```


```{r}
# 2   2 AX-40592209 TC20001033.hg.1 -5.631148 3.723028e-08 0.06760264 -0.06751612
fit <- lm(TC20001033.hg.1~`AX-40592209` +age+race+sex +`AX-40592209`*diagnosis, data = merge_v3)
plot_model(fit, type = "pred", terms = c("AX-40592209","diagnosis"), axis.title = c("rs12480811 genotype","PTK6 expression"),axis.labels = c("rs12480811 genotype","PTK6 expression"))

#merge_v3$`AX-40592209`  
# 0   1   2 
# 2  48   301 
plot(as.factor(merge_v3$`AX-40592209`),merge_v3$TC20001033.hg.1) # to see plot for all samples
ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-40592209`), y=merge_v3$TC20001033.hg.)) +    geom_point(shape=1) +  geom_smooth(method=lm , color="red", se=FALSE) #to see scatterplot for all samples

p1 <-ggplot(aes(y = merge_v3$TC20001033.hg.1, x = as.factor(merge_v3$`AX-40592209`), fill = as.factor(merge_v3$diagnosis)), data =merge_v3, xlim(c(0,1,2))) + geom_boxplot()
p1 + labs(title = "cis-eQTL", y="PTK6 expression", x = "rs12480811 genotype")
```

```{r}
#scatterplot with drawing line
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-40592209`, y=merge_v3$TC20001033.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p1 + labs(title = "cis-eQTL", y="PTK6 expression", x = "rs17712238 genotype")
```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(x,varname = "TC20001033.hg.1", groupnames =c("diagnosis","`AX-40592209`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-14758778` <- as.factor(df$`AX-40592209`)
```

```{r}
#Barplot with error bars
p<- ggplot(df, aes(x=df$`AX-40592209`, y=df$TC20001033.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC20001033.hg.1-sd, ymax=TC20001033.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="PTK6 expression", x = "rs17712238 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-40592209`, y=df$TC20001033.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC20001033.hg.1-sd, ymax=TC20001033.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="PTK6 expression", x= "rs17712238 genotype")

```




```{r}
#   X        snps            gene statistic       pvalue        FDR        beta
#3   3 AX-11336441 TC02001095.hg.1  5.580785 4.859682e-08 0.06760264  0.09543061
#merge_v3$`AX-11336441`  
  # 0   1   2 
  # 3  38 310
fit <- lm(TC02001095.hg.1~`AX-11336441` +age+race+sex+diagnosis +`AX-11336441`*diagnosis, data = merge_v3)
# summary (fit)
# Call:
# lm(formula = TC02001095.hg.1 ~ `AX-11336441` + age + race + sex + 
#     diagnosis + `AX-11336441` * diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.44356 -0.09706 -0.01818  0.06927  1.19661 
# 
# Coefficients:
#                           Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              5.1103028  0.1222451  41.804  < 2e-16 ***
# `AX-11336441`           -0.3488233  0.0552303  -6.316 8.31e-10 ***
# age                      0.0009570  0.0006626   1.444    0.150    
# race                     0.0059026  0.0069812   0.846    0.398    
# sex                      0.0003352  0.0183888   0.018    0.985    
# diagnosis               -0.2018213  0.0327035  -6.171 1.91e-09 ***
# `AX-11336441`:diagnosis  0.0954306  0.0170999   5.581 4.86e-08 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1646 on 344 degrees of freedom
# Multiple R-squared:  0.1395,	Adjusted R-squared:  0.1245 
# F-statistic: 9.297 on 6 and 344 DF,  p-value: 1.874e-09

plot_model(fit, type = "pred", terms = c("AX-11336441","diagnosis"),axis.title = c("rs17712238 genotype","ZNF804A expression"), axis.labels = c("rs17712238 genotype","ZNF804A expression"))
```

```{r}
p1 <-ggplot(aes(y = merge_v3$TC02001095.hg.1, x = as.factor(merge_v3$`AX-11336441`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p1 + labs(title = "cis-eQTL", y="ZNF804A expression", x = "rs17712238 genotype")

```



```{r}
#scatterplot with drawing line 
p1 <- ggplot(merge_v3, aes(x = merge_v3$`AX-11336441`, y=merge_v3$TC02001095.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p1 + labs(title = "cis-eQTL", y="ZNF804A expression", x = "rs17712238 genotype")

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(merge_v3,varname = "TC02001095.hg.1", groupnames =c("diagnosis","`AX-11336441`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-11336441` <- as.factor(df$`AX-11336441`)
```

```{r}
#Barplot with error bars
p<- ggplot(df, aes(x=df$`AX-11336441`, y=df$TC02001095.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC02001095.hg.1-sd, ymax=TC02001095.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="ZNF804A expression", x = "rs17712238 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-11336441`, y=df$TC02001095.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC02001095.hg.1-sd, ymax=TC02001095.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="ZNF804A expression", x= "rs17712238 genotype")

```



```{r}
# 4   4 AX-29907155 TC11001143.hg.1  5.452492 9.499488e-08 0.09910994  0.25824924
merge_v3$`AX-29907155`
#0   1   2 
#19 118 214 

fit <- lm(TC11001143.hg.1~`AX-29907155` +age+race+sex +`AX-29907155`*diagnosis, data = merge_v3)
summary(fit)
# Call:
# lm(formula = TC11001143.hg.1 ~ `AX-29907155` + age + race + sex + 
#     `AX-29907155` * diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -1.97039 -0.50962 -0.06581  0.35667  3.13849 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              8.384963   0.348192  24.081  < 2e-16 ***
# `AX-29907155`           -0.632831   0.131633  -4.808 2.29e-06 ***
# age                      0.003736   0.003153   1.185   0.2368    
# race                    -0.040871   0.033232  -1.230   0.2196    
# sex                      0.159699   0.087495   1.825   0.0688 .  
# diagnosis               -0.348016   0.079243  -4.392 1.50e-05 ***
# `AX-29907155`:diagnosis  0.258249   0.047364   5.452 9.50e-08 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.7844 on 344 degrees of freedom
# Multiple R-squared:  0.1071,	Adjusted R-squared:  0.09157 
# F-statistic:  6.88 on 6 and 344 DF,  p-value: 6.515e-07

plot_model(fit, type = "pred", terms = c("AX-29907155","diagnosis"),axis.title = c("rs56348411 genotype","NRGN expression"), axis.labels = c("rs56348411 genotype","NRGN expression"))

```

```{r}
p1 <-ggplot(aes(y = merge_v3$TC11001143.hg.1, x = as.factor(merge_v3$`AX-29907155`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p1 + labs(title = "cis-eQTL", y="NRGN expression", x = "rs56348411 genotype")

```



```{r}
#scatterplot with drawing line 
p1 <- ggplot(merge_v3, aes(x = merge_v3$`AX-29907155`, y=merge_v3$TC11001143.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p1 + labs(title = "cis-eQTL", y="NRGN expression", x = "rs56348411 genotype")

```

```{r}
p <- ggplot(merge_v3, aes(x = merge_v3$`AX-29907155`, y=merge_v3$TC11001143.hg.1, group = as.factor(diagnosis),color=as.factor(diagnosis)))+
  #geom_line()
  geom_point()
  geom_errorbar(aes(ymin =merge_v3$TC11001143.hg.1-sd, ymax= merge_v3$TC11001143.hg.1+sd), width =0.2, color ="black",
                position = position_dodge(0.05))

sd <-sd(merge_v3$TC11001143.hg.1)
  lower <- sd  
ggplot() + 
geom_errorbar(data=merge_v3, mapping=aes(x=merge_v3$`AX-29907155`, ymin= merge_v3$TC11001143.hg.1-sd, ymax= merge_v3$TC11001143.hg.1+sd), width=0.2, size=1, color="blue") + 
geom_point(data=merge_v3, mapping=aes(x=merge_v3$`AX-29907155`, y=merge_v3$TC11001143.hg.1), size=4, shape=21, fill="white")
```


```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(merge_v3,varname = "TC11001143.hg.1", groupnames =c("diagnosis","`AX-29907155`"))
# Convert genotype to a factor variable
df$`AX-29907155` <- as.factor(df$`AX-29907155`)
```

```{r}
#Barplot with error bars
p<- ggplot(df, aes(x=df$`AX-29907155`, y=df$TC11001143.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC11001143.hg.1-sd, ymax=TC11001143.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p+ labs(title = "cis-eQTL", y="NRGN expression", x = "rs56348411 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-29907155`, y=df$TC11001143.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC11001143.hg.1-sd, ymax=TC11001143.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="NRGN expression", x= "rs56348411 genotype")

```


```{r}
# 5   5 AX-39724859 TC01000514.hg.1  5.405367 1.211415e-07 0.10111133  0.15726662

merge_v3$`AX-39724859`
#0   1   2 
#4  30 317  
#load merge_v3
fit <- lm(TC01000514.hg.1~`AX-39724859`+age+race+sex+`AX-39724859`*diagnosis, data = merge_v3)
# summary(fit)
# Call:
# lm(formula = TC01000514.hg.1 ~ `AX-39724859` + age + race + sex + 
#     `AX-39724859` * diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -0.9019 -0.1398 -0.0201  0.1211  3.5326 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              7.075905   0.184079  38.440  < 2e-16 ***
# `AX-39724859`           -0.593189   0.083668  -7.090 7.67e-12 ***
# age                     -0.001173   0.001127  -1.041    0.299    
# race                     0.003796   0.011924   0.318    0.750    
# sex                      0.002856   0.031489   0.091    0.928    
# diagnosis               -0.320217   0.055941  -5.724 2.27e-08 ***
# `AX-39724859`:diagnosis  0.157267   0.029095   5.405 1.21e-07 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2817 on 344 degrees of freedom
# Multiple R-squared:  0.1449,	Adjusted R-squared:   0.13 
# F-statistic: 9.714 on 6 and 344 DF,  p-value: 6.873e-10

plot_model(fit, type = "pred", terms = c("AX-39724859","diagnosis"),axis.title = c("rs11809423 genotype","ZNF684 expression"), axis.labels = c("rs11809423 genotype","ZNF684 expression"))

```

```{r}
p1 <-ggplot(aes(y = merge_v3$TC01000514.hg.1, x = as.factor(merge_v3$`AX-39724859`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p1 + labs(title = "cis-eQTL", y="ZNF684 expression", x = "rs11809423 genotype")

```


```{r}
#scatterplot with drawing line 
p1 <- ggplot(merge_v3, aes(x = merge_v3$`AX-39724859`, y=merge_v3$TC01000514.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p2 <- p1 + labs(title = "cis-eQTL", y="ZNF684 expression", x = "rs11809423 genotype")

```

```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(merge_v3,varname = "TC01000514.hg.1", groupnames =c("diagnosis","`AX-39724859`")) #No NA
# Convert genotype to a factor variable
df$`AX-39724859` <- as.factor(df$`AX-39724859`)
```

```{r}
#Barplot with error bars
df$diagnosis <- as.factor(df$diagnosis)
p<- ggplot(df, aes(x=df$`AX-39724859`, y=df$TC01000514.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="ZNF684 expression", x = "rs11809423 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-39724859`, y=df$TC01000514.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC01000514.hg.1-sd, ymax=TC01000514.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="ZNF684 expression", x= "rs11809423 genotype")

```


```{r}
# 6   6 AX-15964685 TC08000305.hg.1  5.198619 3.451509e-07 0.20825690  0.30886447

merge_v3$`AX-15964685`
#0   1   2  NA
#296  46   3  6
#load merge_v3
# Call:
# lm(formula = TC08000305.hg.1 ~ `AX-15964685` + age + race + sex + 
#     `AX-15964685` * diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.1697 -0.3746 -0.1071  0.2429  4.6769 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              5.101116   0.208593  24.455  < 2e-16 ***
# `AX-15964685`           -0.478323   0.137117  -3.488  0.00055 ***
# age                      0.003007   0.002280   1.319  0.18806    
# race                     0.003104   0.023957   0.130  0.89700    
# sex                      0.080258   0.063427   1.265  0.20661    
# diagnosis               -0.015505   0.022562  -0.687  0.49242    
# `AX-15964685`:diagnosis  0.307687   0.059748   5.150 4.44e-07 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.5608 on 338 degrees of freedom
#   (6 observations deleted due to missingness)
# Multiple R-squared:  0.09078,	Adjusted R-squared:  0.07464 
# F-statistic: 5.625 on 6 and 338 DF,  p-value: 1.395e-05

plot_model(fit, type = "pred", terms = c("AX-15964685","diagnosis"),axis.title = c("rs72639679 genotype","IDO1 expression"), axis.labels = c("rs72639679 genotype","IDO1 expression"))

```

```{r}
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
x <-merge_v3[!is.na(merge_v3$'AX-15964685'), ] #remove samples with NAs dim 345 294020
p1 <-ggplot(aes(y = x$TC08000305.hg.1, x = as.factor(x$`AX-15964685`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <- p1 + labs(title = "cis-eQTL", y="IDO1 expression", x = "rs72639679 genotype")

```


```{r}
#scatterplot with drawing line 
p1 <- ggplot(x, aes(x = x$`AX-15964685`, y=x$TC08000305.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p2 <- p1 + labs(title = "cis-eQTL", y="IDO1 expression", x = "rs72639679 genotype")

```

```{r}
#for merge_v3 data
#x$diagnosis <- as.factor(x$diagnosis)
df <-data_summary(x,varname = "TC08000305.hg.1", groupnames =c("diagnosis","`AX-15964685`")) #No NA
# Convert genotype to a factor variable
df$`AX-15964685` <- as.factor(df$`AX-15964685`)
```

```{r}
#Barplot with error bars
df$diagnosis <- as.factor(df$diagnosis)
p<- ggplot(df, aes(x=df$`AX-15964685`, y=df$TC08000305.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC08000305.hg.1-sd, ymax=TC08000305.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="IDO1 expression", x = "rs72639679 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-15964685`, y=df$TC08000305.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC08000305.hg.1-sd, ymax=TC08000305.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="IDO1 expression", x= "rs72639679 genotype")

```



```{r}
# 7   7 AX-11701696 TC18000998.hg.1 -5.196212 3.493176e-07 0.20825690 -0.14708963

merge_v3$`AX-11701696`
#0   1   2  
#304  45   2 
#load merge_v3
# fit<-lm(formula = TC18000998.hg.1 ~ `AX-11701696` + age + race + sex +`AX-11701696`* diagnosis, data = merge_v3)
# summary(fit)
# Call:
# lm(formula = TC18000998.hg.1 ~ `AX-11701696` + age + race + sex + 
#     `AX-11701696` * diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -2.94315 -0.10771  0.00998  0.13634  0.62031 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)             10.643293   0.104979 101.385  < 2e-16 ***
# `AX-11701696`            0.188991   0.076982   2.455   0.0146 *  
# age                     -0.001310   0.001127  -1.163   0.2457    
# race                    -0.021581   0.011975  -1.802   0.0724 .  
# sex                     -0.020971   0.031513  -0.665   0.5062    
# diagnosis                0.018058   0.011278   1.601   0.1102    
# `AX-11701696`:diagnosis -0.147090   0.028307  -5.196 3.49e-07 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.2813 on 344 degrees of freedom
# Multiple R-squared:  0.1225,	Adjusted R-squared:  0.1072 
# F-statistic: 8.006 on 6 and 344 DF,  p-value: 4.229e-08

plot_model(fit, type = "pred", terms = c("AX-11701696","diagnosis"),axis.title = c("rs975903 genotype","SMAD4 expression"), axis.labels = c("rs975903 genotype","SMAD4 expression"))

```

```{r}
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
p1 <-ggplot(aes(y = merge_v3$TC18000998.hg.1, x = as.factor(merge_v3$`AX-11701696`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <- p1 + labs(title = "cis-eQTL", y="SMAD4 expression", x = "rs975903 genotype")

```


```{r}
#scatterplot with drawing line 
p1 <- ggplot(merge_v3, aes(x = merge_v3$`AX-11701696`, y=merge_v3$TC18000998.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p2 <- p1 + labs(title = "cis-eQTL", y="SMAD4 expression", x = "rs975903 genotype")

```

```{r}
#for merge_v3 data
#x$diagnosis <- as.factor(x$diagnosis)
df <-data_summary(merge_v3,varname = "TC18000998.hg.1", groupnames =c("diagnosis","`AX-11701696`")) #No NA
# Convert genotype to a factor variable
df$`AX-11701696` <- as.factor(df$`AX-11701696`)
```

```{r}
#Barplot with error bars
df$diagnosis <- as.factor(df$diagnosis)
p<- ggplot(df, aes(x=df$`AX-11701696`, y=df$TC18000998.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="SMAD4 expression", x = "rs975903 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-11701696`, y=df$TC18000998.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC18000998.hg.1-sd, ymax=TC18000998.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="SMAD4 expression", x= "rs975903 genotype")

```



```{r}
#create the figures for some of trans-eQTL 
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/eQTL_trans_1e6.csv", row.names=1)
#AX-35207585 TC03002054.hg.1 -7.001953 1.331282e-11 4.934046e-03 -0.35400376
fit <- lm(TC03002054.hg.1~`AX-35207585` +age+race+sex+diagnosis +`AX-35207585`*diagnosis, data = merge_v3)
plot_model(fit, type = "pred", terms = c("AX-35207585","diagnosis"))
#AX-12869101 TC03000801.hg.1  6.704439 8.302696e-11 1.538588e-02  0.39215365
fit <- lm(TC03000801.hg.1~`AX-12869101` +age+race+sex+diagnosis +`AX-12869101`*diagnosis, data = merge_v3)
```

```{r}
#AX-35207585 TC03002054.hg.1 -7.001953 1.331282e-11 4.934046e-03 -0.35400376
fit <- lm(TC03002054.hg.1~`AX-35207585` +age+race+sex+diagnosis +`AX-35207585`*diagnosis, data = merge_v3)
merge_v3$`AX-35207585`
#0   1   2  NA  
#316  32  2  1
x <-merge_v3[!is.na(merge_v3$'AX-35207585'), ] #remove samples with NAs dim 350 294920

#load merge_v3
# Call:
# lm(formula = TC03002054.hg.1 ~ `AX-35207585` + age + race + sex + 
#     diagnosis + `AX-35207585` * diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.2053 -0.1453 -0.0557  0.0629  5.4931 
# 
# Coefficients:
#                          Estimate Std. Error t value Pr(>|t|)    
# (Intercept)              4.482052   0.164702  27.213  < 2e-16 ***
# `AX-35207585`            1.368519   0.161445   8.477 6.96e-16 ***
# age                     -0.002535   0.001782  -1.422    0.156    
# race                    -0.017511   0.018911  -0.926    0.355    
# sex                      0.062109   0.049531   1.254    0.211    
# diagnosis                0.015210   0.017176   0.886    0.376    
# `AX-35207585`:diagnosis -0.353915   0.050559  -7.000 1.35e-11 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.4423 on 343 degrees of freedom
#   (1 observation deleted due to missingness)
# Multiple R-squared:  0.1792,	Adjusted R-squared:  0.1648 
# F-statistic: 12.48 on 6 and 343 DF,  p-value: 9.921e-13
plot_model(fit, type = "pred", terms = c("AX-35207585","diagnosis"),axis.title = c("rs77696886 genotype","LAMP3 expression"), axis.labels = c("rs77696886 genotype","LAMP3 expression"))

```

```{r}
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
x$diagnosis <- as.factor(x$diagnosis)
p1 <-ggplot(aes(y = x$TC03002054.hg.1, x = as.factor(x$`AX-35207585`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <- p1 + labs(title = "cis-eQTL", y="LAMP3 expression", x = "rs77696886 genotype")

```


```{r}
#scatterplot with drawing line 
p1 <- ggplot(x, aes(x = x$`AX-35207585`, y=x$TC03002054.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p2 <- p1 + labs(title = "cis-eQTL", y="LAMP3 expression", x = "rs77696886 genotype")

```

```{r}
#for merge_v3 data
#x$diagnosis <- as.factor(x$diagnosis)
df <-data_summary(x,varname = "TC03002054.hg.1", groupnames =c("diagnosis","`AX-35207585`")) #No NA
# Convert genotype to a factor variable
df$`AX-35207585` <- as.factor(df$`AX-35207585`)
```

```{r}
#Barplot with error bars
df$diagnosis <- as.factor(df$diagnosis)
p<- ggplot(df, aes(x=df$`AX-35207585`, y=df$TC03002054.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC03002054.hg.1-sd, ymax=TC03002054.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="LAMP3 expression", x = "rs77696886 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-35207585`, y=df$TC03002054.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC03002054.hg.1-sd, ymax=TC03002054.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="LAMP3 expression", x= "rs77696886 genotype")

```



```{r}
#AX-12869101 TC03000801.hg.1  6.704439 8.302696e-11 1.538588e-02  0.39215365
#load merge_v3
fit <- lm(TC03000801.hg.1~`AX-12869101`+age+race+sex+diagnosis +`AX-12869101`*diagnosis, data = merge_v3)

merge_v3$`AX-12869101`
#0   1   2   
#2  37  312 



plot_model(fit, type = "pred", terms = c("AX-12869101","diagnosis"),axis.title = c("rs8003997 genotype","GYG1 expression"), axis.labels = c("rs8003997 genotype","GYG1 expression"))

```

```{r}
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
p1 <-ggplot(aes(y = merge_v3$TC03000801.hg.1, x = as.factor(merge_v3$`AX-12869101`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <- p1 + labs(title = "cis-eQTL", y="GYG1 expression", x = "rs8003997 genotype")

```


```{r}
#scatterplot with drawing line 
p1 <- ggplot(merge_v3, aes(x = merge_v3$`AX-12869101`, y=merge_v3$TC03000801.hg.1, color=as.factor(diagnosis), shape=as.factor(diagnosis)))+
  geom_point(aes(size=as.factor(diagnosis))) + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_shape_manual(values=c(3, 16, 17))+ 
  scale_color_manual(values=c('#999999','#E69F00', '#56B4E9'))+
  scale_size_manual(values=c(2,3,4))+
  theme(legend.position="top")
p2 <- p1 + labs(title = "cis-eQTL", y="GYG1 expression", x = "rs8003997 genotype")

```

```{r}
#for merge_v3 data
df <-data_summary(x,varname = "TC03000801.hg.1", groupnames =c("diagnosis","`AX-12869101`")) #No NA
# Convert genotype to a factor variable
df$`AX-12869101` <- as.factor(df$`AX-12869101`)
```

```{r}
#Barplot with error bars
df$diagnosis <- as.factor(df$diagnosis)
p<- ggplot(df, aes(x=df$`AX-12869101`, y=df$TC03000801.hg.1, fill=df$diagnosis)) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC03000801.hg.1-sd, ymax=TC03000801.hg.1+sd), width=.2,
                 position=position_dodge(.9)) 

p2<-p+ labs(title = "cis-eQTL", y="GYG1 expression", x = "rs8003997 genotype")
```

```{r}
#scatterplot with error bars
p<- ggplot(df, aes(x=df$`AX-12869101`, y=df$TC03000801.hg.1, group=df$diagnosis, color =df$diagnosis))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC03000801.hg.1-sd, ymax=TC03000801.hg.1+sd),width=.2)
p2<- p+ labs(title = "cis-eQTL", y="GYG1 expression", x= "rs8003997 genotype")
```

```{r}
#since demographic variables for ischemic stroke (IS) and vascular risk factor control (VRFC) patients are significant, add these variables into model and redo analysis
SNP_file_name = paste(base.dir, "/Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", sep="")
gene_location_file_name = paste(base.dir, "/Input_Output/Gene_Location_Exclude_Weird_Position.txt", sep="")
snps_location_file_name = paste(base.dir, "/Input_Output/SNP_Location.txt", sep="") #include all SNPs location
expression_file_name = paste(base.dir, "/Input_Output/GE_mRNA_19309.txt", sep="") #only coding gene
#check different covariate matrix
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_NO_Subtype.txt", sep="")
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_No_PC_NO_Subtype.txt", sep="")
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_NO_Batch.txt", sep="") # new subdiagnosis
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_No_PC.txt", sep="")
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_No_Race_NO_PC.txt", sep="")
```

```{r}
#load genotype data 273688
#load exp data 65988 for all genes and 19309 for only protein-coding genes
#load covariate data 
```


```{r}
useModel =  modelLINEAR_CROSS
cisDist = 1e6
eQTL = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = output_file_name_tra,
pvOutputThreshold = 1e-10,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, 
output_file_name.cis = output_file_name_cis,
pvOutputThreshold.cis = 1e-5,
snpspos = snpspos, 
genepos = genepos,
cisDist = cisDist,
pvalue.hist = "qqplot",
min.pv.by.genesnp = FALSE,
noFDRsaveMemory = FALSE)

unlink(output_file_name_tra);
unlink(output_file_name_cis);
```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_sex_race_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_sex_race_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_trans_1e6.csv")
```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_sex_race_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_sex_race_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_trans_1e6.csv")
```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_trans_1e6.csv") # #corrected for last updated covariate matrix, the covariate matrix corrected for this analysis (after changing race coding)
```

```{r}
#Standard errors for the estimates.
eQTL$cis$eqtls$beta_se <- eQTL$cis$eqtls$beta / eQTL$cis$eqtls$statistic
# eQTL$cis$eqtls$beta_se
#  [1] 0.075854727 0.012066198 0.017297127 0.047162212 0.059563955 0.029166231
#  [7] 0.010477433 0.024182143 0.008352869 0.032107944 0.010133248 0.009137178
# [13] 0.028726676 0.009626007 0.039579878 0.010700809 0.007630901 0.009797138
# [19] 0.024453765 0.014638791 0.005133608 0.040457040 0.008509307 0.043775565
# [25] 0.017979140 0.007390872 0.009437438 0.032973273 0.007725246 0.023255304
# [31] 0.021296291 0.007457942 0.068455040 0.028361994 0.010223230 0.039667200
# [37] 0.031463367 0.031505299 0.007024834 0.007812243 0.006770492 0.008740375
# [43] 0.008355497 0.011992802 0.010980923 0.009595175 0.005663774 0.010755003
# [49] 0.013926929 0.013926873 0.062989805 0.006775567 0.098087008 0.013612401
# [55] 0.023703287 0.008452552 0.032737538 0.020449920 0.021851710 0.008500891
# [61] 0.037508396 0.010354264 0.048032828 0.035203528 0.038934214 0.010344384
#[67] 0.016473400
eQTL$trans$eqtls$beta_se <- eQTL$trans$eqtls$beta / eQTL$trans$eqtls$statistic
# eQTL$trans$eqtls$beta_se
#  [1] 0.02358744 0.01588290 0.01415431 0.01808981 0.02472100 0.02007244
#  [7] 0.02914444 0.02744014 0.02459031 0.03709459 0.03906363 0.01646312
# [13] 0.03473560 0.03937634 0.04050171 0.02435498 0.01738336 0.01513311
# [19] 0.05060625 0.01444129 0.01962192 0.03692866 0.02943508 0.03643836
# [25] 0.01404486 0.03407944 0.03987007 0.01552397 0.02997063
```

```{r}
#Raw correlations from which Matrix eQTL calculates t-statistics and p-values.
dfFull = eQTL$param$dfFull
tstat = eQTL$cis$eqtls$statistic
r = tstat / sqrt( dfFull + tstat^2 )
R2 = r^2
#  R2
#  [1] 0.09510262 0.08511459 0.07968084 0.07945413 0.07434535 0.07420156
#  [7] 0.07087332 0.07017246 0.06962320 0.06949803 0.06932230 0.06912241
# [13] 0.06903879 0.06726197 0.06698603 0.06634255 0.06627773 0.06584486
# [19] 0.06472357 0.06456517 0.06395140 0.06380791 0.06297481 0.06215805
# [25] 0.06208506 0.06185348 0.06176478 0.06158587 0.06147160 0.06138077
# [31] 0.06109520 0.06102370 0.06074230 0.06050882 0.06041129 0.06040314
# [37] 0.06032564 0.06011694 0.05997462 0.05994304 0.05981198 0.05962986
# [43] 0.05955519 0.05919489 0.05918110 0.05916915 0.05901745 0.05851090
# [49] 0.05809745 0.05805584 0.05804623 0.05787786 0.05782754 0.05758840
# [55] 0.05718037 0.05717668 0.05711226 0.05711130 0.05693809 0.05692471
# [61] 0.05686321 0.05666130 0.05657320 0.05652570 0.05606978 0.05597175
# [67] 0.05590293

dfFull = eQTL$param$dfFull
tstat = eQTL$trans$eqtls$statistic
r = tstat / sqrt( dfFull + tstat^2 )
R2 = r^2
#  R2
#  [1] 0.2777432 0.1505736 0.1472243 0.1432146 0.1410028 0.1370104 0.1359646
#  [8] 0.1354437 0.1352983 0.1334349 0.1325515 0.1301655 0.1253817 0.1250485
# [15] 0.1244907 0.1242009 0.1214836 0.1211350 0.1206759 0.1199084 0.1197227
# [22] 0.1184184 0.1183419 0.1179460 0.1167708 0.1166493 0.1164784 0.1163501
# [29] 0.1160688
```


```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_pc1_pc2_pc3_hypertension_diabates_cholestrolemia _diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_pc1_pc2_pc3_hypertension_diabates_cholestrolemia _diagnosis/eQTL_trans_1e6.csv") # highest number of eQTL pass FDR here
```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_sex_race_age_pc1_pc2_pc3_hypertension_diabates_cholestrolemia_diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_sex_race_age_pc1_pc2_pc3_hypertension_diabates_cholestrolemia_diagnosis/eQTL_trans_1e6.csv") # highest number of eQTL pass FDR here
```

```{r}
#check for coding genes
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_age_hypertension_diabates_cholestrolemia _diagnosis/eQTL_trans_1e6.csv") # highest number of eQTL pass FDR here
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_cis_dist_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_cis_dist_1e6.csv")
#load("../Input_Output/RData/gene_expression_annotation_SNP_calls_small.RData")
eQTL_cis_dist_1e6_merge <- merge(eQTL_cis_dist_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #67

eQTL_cis_dist_1e6_merge_v2 <- merge(eQTL_cis_dist_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#67
write.csv(eQTL_cis_dist_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_cis_dist_1e6_merge.csv") #corrected for last updated covariate matrix
```

```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*)
eQTL_trans_1e6 <- read.csv("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_trans_1e6.csv")
eQTL_trans_1e6_merge <- merge(eQTL_trans_1e6,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #29

eQTL_trans_1e6_merge_v2 <- merge(eQTL_trans_1e6_merge ,SNP_calls_small, by = "snps", all.x = TRUE, sort = TRUE)#29
write.csv(eQTL_trans_1e6_merge_v2, "../../Input_Output/eQTL_result/LINEAR_CROSS/mRNA/covariate_diagnosissubtype_sex_race_age_hypertension_diabates_cholestrolemia_diagnosis/eQTL_trans_1e6_merge.csv") #corrected for last updated covariate matrix
```

```{r}
#updated merge_v3 for this analysis, additional varibales should be added as a row in covariate matrix

#SNP_CRmore95_MAFmore0.05_HWEmore0.0000005 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", row.names=1)
#View(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005)
SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t <- t(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005)
#dim(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#Covariates_Final_NO_PC <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/Covariates_Final_NO_PC.txt", row.names=1)
#View(Covariates_Final_NO_PC)
Covariates_Final_NO_PC_t <-t(Covariates_Final_NO_PC)
#GE_mRNA_19309 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/GE_mRNA_19309.txt", row.names=1)
GE_mRNA_19309_t <-t(GE_mRNA_19309)
#class(GE_mRNA_19309_t)
merge <- cbind(GE_mRNA_19309_t,SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#dim(merge)
#dim(SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_t)
#dim(GE_mRNA_19309_t) 273688 + 19309
merge_v2 <- cbind(merge,Covariates_Final_NO_PC_t)
#dim(merge_v2) #final dataset  351 293005
merge_v3 <- as.data.frame(merge_v2)
save(merge_v3, file = "../../Input_Output/RData/merge_final.RData")
```

```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
#          snps            gene statistic       pvalue        FDR        beta
#1  AX-14758778 TC04001305.hg.1 -5.977727 5.719097e-09 0.02386736 -0.45343881
fit <- lm(TC04001305.hg.1~`AX-14758778` +age+race+sex+diagnosis+historyhypertension +historydiabetesmellitus+historyhypercholestrolemia+diagnosissubtype +`AX-14758778`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC04001305.hg.1 ~ `AX-14758778` + age + race + sex + 
#     diagnosis + historyhypertension + historydiabetesmellitus + 
#     historyhypercholestrolemia + diagnosissubtype + `AX-14758778` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#     Min      1Q  Median      3Q     Max 
# -1.8468 -0.3359 -0.0773  0.2357  7.0694 
# 
# Coefficients:
#                              Estimate Std. Error t value Pr(>|t|)    
# (Intercept)                 4.2958432  0.2807849  15.299  < 2e-16 ***
# `AX-14758778`               1.7522919  0.2314126   7.572 3.56e-13 ***
# age                         0.0003242  0.0031125   0.104   0.9171    
# race                        0.0370903  0.0299812   1.237   0.2169    
# sex                         0.0132518  0.0807882   0.164   0.8698    
# diagnosis                   0.0514983  0.0742076   0.694   0.4882    
# historyhypertension         0.1638475  0.0905787   1.809   0.0714 .  
# historydiabetesmellitus    -0.2215654  0.0950071  -2.332   0.0203 *  
# historyhypercholestrolemia  0.0694459  0.0841493   0.825   0.4098    
# diagnosissubtype           -0.0231168  0.0413524  -0.559   0.5765    
# `AX-14758778`:diagnosis    -0.4525145  0.0761100  -5.946 6.89e-09 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.7068 on 337 degrees of freedom
#   (3 observations deleted due to missingness)
# Multiple R-squared:  0.1713,	Adjusted R-squared:  0.1467 
# F-statistic: 6.967 on 10 and 337 DF,  p-value: 6.101e-10

#draw Figures
plot_model(fit, type = "pred", terms = c("AX-14758778","diagnosis"), axis.title = c("rs78046578","CXCL10 expression"))
```


```{r}
#plot exact point
#devtools::install_github("hadley/devtools")
#library(devtools)
#install_github("kassambara/easyGgplot2")
#library("ggplot2")
#library("easyGgplot2") do not this library for now
```

```{r}
#          snps            gene statistic       pvalue        FDR        beta
#1  AX-14758778 TC04001305.hg.1 -5.977727 5.719097e-09 0.02386736 -0.45343881
#merge_v3$`AX-14758778`  
# 0   1   2 NA
# 312  33   3 3
x <- merge_v3$`AX-14758778`
# 2 203 (UCDSS.674),233 (UCDSS.738), 347("UofASS.032") 

#diagnosissubtype	gender	race	age	diagnosis	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia

#UCDSS.674	3	1	6	43	1	1	0	1
#UCDSS.738	7	2	5	66	4	0	0	0
#UofASS.032	7	2	5	63	4	0	0	0


x <-merge_v3[!is.na(merge_v3$'AX-14758778'), ] #remove samples with NA

p1 <-ggplot(aes(y = x$TC04001305, x = as.factor(x$`AX-14758778`), fill = as.factor(x$diagnosis)), data = x,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(title = "cis-eQTL", y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "TT", "1" = "TC", "2"="CC"))+scale_y_continuous(expression(paste( italic("CXCL10")," " , "expression")))+expand_limits(y=c(3.5,14.5))
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1.jpg",width = 7, height = 5)
```



```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = x, aes(x=as.factor(x$`AX-14758778`), y=x$TC04001305)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(x$diagnosis))) 

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression")))+ expand_limits(y=c(3.5,14.5))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_point.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_point.jpg",width = 7, height = 5)

#to see reg line for all sample
reg1<-lm(TC04001305.hg.1~`AX-14758778`, data = x)
with(x,plot(as.factor(`AX-14758778`),TC04001305.hg.1))
abline(reg1)

```

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(x, aes(x=x$`AX-14758778`, y=x$TC04001305.hg.1, colour=as.factor(x$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(3.5,14.5))

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_regline.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_regline.jpg",width = 7, height = 5)

```


```{r}
#function for having error bar for different groups

data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}
```


```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(x,varname = "TC04001305.hg.1", groupnames =c("diagnosis","`AX-14758778`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-14758778` <- as.factor(df$`AX-14758778`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=df$`AX-14758778`, y=df$TC04001305.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC04001305.hg.1-sd, ymax=TC04001305.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(3.5,14.5))

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_boxerrorbar.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_boxerrorbar.jpg",width = 7, height = 5)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=df$`AX-14758778`, y=df$TC04001305.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC04001305.hg.1-sd, ymax=TC04001305.hg.1+sd),width=.2)+ expand_limits(y=c(3.5,14.5))

p2 <-p1 + labs(y="CXCL10 expression", x = "rs78046578 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("CXCL10")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_scattererrorbar.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL1_scattererrorbar.jpg",width = 7, height = 5)
```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
#snps	gene	statistic	pvalue	FDR	beta	beta_se
#2	AX-40592209	TC20001033.hg.1	-5.624165652	3.89E-08	0.081273047	-0.067862296	0.012066198

fit <- lm(TC20001033.hg.1~`AX-40592209` +age+race+sex+diagnosis+historyhypertension +historydiabetesmellitus+historyhypercholestrolemia+diagnosissubtype +`AX-40592209`*diagnosis, data = merge_v3)
summary (fit)
# Call:
# lm(formula = TC20001033.hg.1 ~ `AX-40592209` + age + race + sex + 
#     diagnosis + historyhypertension + historydiabetesmellitus + 
#     historyhypercholestrolemia + diagnosissubtype + `AX-40592209` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.34439 -0.08239 -0.01746  0.08080  0.58921 
# 
# Coefficients:
#                              Estimate Std. Error t value Pr(>|t|)    
# (Intercept)                 5.1214672  0.0794586  64.455  < 2e-16 ***
# `AX-40592209`               0.1156792  0.0341456   3.388 0.000787 ***
# age                         0.0005837  0.0005423   1.076 0.282595    
# race                        0.0085822  0.0052564   1.633 0.103455    
# sex                         0.0248181  0.0141710   1.751 0.080791 .  
# diagnosis                   0.1313776  0.0262676   5.002 9.13e-07 ***
# historyhypertension        -0.0013833  0.0158859  -0.087 0.930660    
# historydiabetesmellitus    -0.0300203  0.0166762  -1.800 0.072716 .  
# historyhypercholestrolemia  0.0023073  0.0148082   0.156 0.876271    
# diagnosissubtype           -0.0027488  0.0072555  -0.379 0.705028    
# `AX-40592209`:diagnosis    -0.0678623  0.0120662  -5.624 3.89e-08 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1245 on 340 degrees of freedom
# Multiple R-squared:  0.1265,	Adjusted R-squared:  0.1008 
# F-statistic: 4.924 on 10 and 340 DF,  p-value: 1.154e-06
```


```{r}
#snps	gene	statistic	pvalue	FDR	beta	
#2	AX-40592209	TC20001033.hg.1	-5.624165652	3.89E-08	0.081273047	-0.067862296	
#merge_v3$`AX-40592209`  
#  0   1   2 
#  2  48 301 
x <- merge_v3$`AX-40592209`
# 2 47 (UCDSS.219), 198 (UCDSS.642)
#diagnosissubtype	gender	race	age	diagnosis	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia
#UCDSS.219	8	1	5	62	4	1	1	1
#UCDSS.642	3	1	5	64	1	1	0	0

#min(merge_v3$TC20001033.hg.1)
#[1] 5.144266
# max(merge_v3$TC20001033.hg.1)
#[1] 6.203747

p1 <-ggplot(aes(y = merge_v3$TC20001033.hg.1, x = as.factor(merge_v3$`AX-40592209`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("PTK6")," " , "expression")))+expand_limits(y=c(5.5,6.5))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2.jpg",width = 7, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-40592209`), y=merge_v3$TC20001033.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression")))+ expand_limits(y=c(5.5,6.5))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_point.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_point.jpg",width = 7, height = 5)

```

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-40592209`, y=merge_v3$TC20001033.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5.5,6.5))

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_regline.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_regline.jpg",width = 7, height = 5)

```


```{r}
#function for having error bar for different groups

data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}
```


```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(merge_v3,varname = "TC20001033.hg.1", groupnames =c("diagnosis","`AX-40592209`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-40592209` <- as.factor(df$`AX-40592209`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=df$`AX-40592209`, y=df$TC20001033.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC20001033.hg.1-sd, ymax=TC20001033.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5.5,6.5))

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_boxerrorbar.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_boxerrorbar.jpg",width = 7, height = 5)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=df$`AX-40592209`, y=df$TC20001033.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC20001033.hg.1-sd, ymax=TC20001033.hg.1+sd),width=.2)+ expand_limits(y=c(5,6))

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_scattererrorbar.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_scattererrorbar.jpg",width = 7, height = 5)
```


```{r}
#Try without running merge_v3$diagnosis <- to_factor(merge_v3$diagnosis)
#          snps            gene statistic       pvalue        FDR        beta
#3  AX-11336441 TC02001095.hg.1  5.425593 1.099361e-07 0.11976284  0.09384717
fit <- lm(TC02001095.hg.1~`AX-11336441` +age+race+sex+diagnosis+historyhypertension +historydiabetesmellitus+historyhypercholestrolemia+diagnosissubtype +`AX-11336441`*diagnosis, data = merge_v3)
summary (fit)

# Call:
# lm(formula = TC02001095.hg.1 ~ `AX-11336441` + age + race + sex + 
#     diagnosis + historyhypertension + historydiabetesmellitus + 
#     historyhypercholestrolemia + diagnosissubtype + `AX-11336441` * 
#     diagnosis, data = merge_v3)
# 
# Residuals:
#      Min       1Q   Median       3Q      Max 
# -0.44424 -0.09503 -0.02709  0.07378  1.18019 
# 
# Coefficients:
#                              Estimate Std. Error t value Pr(>|t|)    
# (Intercept)                 5.1031951  0.1234103  41.351  < 2e-16 ***
# `AX-11336441`              -0.3459965  0.0556597  -6.216 1.49e-09 ***
# age                         0.0009954  0.0007215   1.380    0.169    
# race                        0.0067030  0.0069730   0.961    0.337    
# sex                        -0.0024119  0.0186376  -0.129    0.897    
# diagnosis                  -0.1909470  0.0363247  -5.257 2.60e-07 ***
# historyhypertension         0.0176967  0.0213655   0.828    0.408    
# historydiabetesmellitus    -0.0104799  0.0221402  -0.473    0.636    
# historyhypercholestrolemia -0.0117846  0.0195947  -0.601    0.548    
# diagnosissubtype           -0.0044340  0.0096043  -0.462    0.645    
# `AX-11336441`:diagnosis     0.0938472  0.0172971   5.426 1.10e-07 ***
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# 
# Residual standard error: 0.1651 on 340 degrees of freedom
# Multiple R-squared:  0.1441,	Adjusted R-squared:  0.1189 
# F-statistic: 5.724 on 10 and 340 DF,  p-value: 5.981e-08
```


```{r}
#          snps            gene statistic       pvalue        FDR        beta
#3  AX-11336441 TC02001095.hg.1  5.425593 1.099361e-07 0.11976284  0.09384717
#merge_v3$`AX-11336441`  
# 0   1   2 
# 3  38 310 
x <- merge_v3$`AX-11336441`
# 0 106 (UCDSS.438),120 (UCDSS.478),323 (UCSFCJ.099)**HERE**
#diagnosissubtype	gender	race	age	diagnosis	historyhypertension	historydiabetesmellitus	historyhypercholestrolemia
#UUCDSS.438	8	1	5	68	4	0	0	0
#UCDSS.478	8	2	5	44	4	0	0	1
#UCSFCJ.099	2	1	5	67	1	1	0	0

#min(merge_v3$TC02001095.hg.1)
#[1] 5.144266
# max(merge_v3$TC02001095.hg.1)
#[1] 6.203747

p1 <-ggplot(aes(y = merge_v3$TC20001033.hg.1, x = as.factor(merge_v3$`AX-40592209`), fill = as.factor(merge_v3$diagnosis)), data = merge_v3,  xlim(c(0,1,2))) + geom_boxplot()
p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete(labels =c("0" = "CC", "1" = "CT", "2"="TT"))+scale_y_continuous(expression(paste( italic("PTK6")," " , "expression")))+expand_limits(y=c(5.5,6.5))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2.jpg",width = 7, height = 5)
```


```{r}
#to see the scatterplot for each SNP-gene pairs
cols <- c("1" = "red", "4" = "blue")
p1 <- ggplot(data = merge_v3, aes(x=as.factor(merge_v3$`AX-40592209`), y=merge_v3$TC20001033.hg.1)) + geom_point(alpha = 0.5, size = 3, aes(colour =as.factor(merge_v3$diagnosis))) 

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_colour_manual(name ="diagnosis", values = cols, breaks = c("1", "4"),labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("CC", "CT", "TT"))+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression")))+ expand_limits(y=c(5.5,6.5))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_point.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_point.jpg",width = 7, height = 5)

```

```{r}
#scatterplot with drawing line "http://www.sthda.com/english/wiki/ggplot2-scatter-plots-quick-start-guide-r-software-and-data-visualization"
#change that to show the case and control only based on the colour not shape
p1 <- ggplot(merge_v3, aes(x=merge_v3$`AX-40592209`, y=merge_v3$TC20001033.hg.1, colour=as.factor(merge_v3$diagnosis), xlim(c(0,1,2))))+
  geom_point() + 
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)+
  scale_color_manual(values=cols, name = "diagnosis",breaks = c("1", "4"),labels = c("IS", "VRFC"))+
  theme(legend.position="top")+ expand_limits(y=c(5.5,6.5))

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression"))) 

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_regline.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_regline.jpg",width = 7, height = 5)

```


```{r}
#function for having error bar for different groups

data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}
```


```{r}
#for merge_v3 data
#merge_v3$diagnosis <- as.factor(merge_v3$diagnosis)
df <-data_summary(merge_v3,varname = "TC20001033.hg.1", groupnames =c("diagnosis","`AX-40592209`")) #Na in sd is due to small number of samples in that case, so we could see the NA
# Convert genotype to a factor variable
df$`AX-40592209` <- as.factor(df$`AX-40592209`)
```

```{r}
#Barplot with error bars
p1<- ggplot(df, aes(x=df$`AX-40592209`, y=df$TC20001033.hg.1, fill=as.factor(df$diagnosis))) + 
  geom_bar(stat="identity", color="black", 
           position=position_dodge()) +
  geom_errorbar(aes(ymin=TC20001033.hg.1-sd, ymax=TC20001033.hg.1+sd), width=.2,
                 position=position_dodge(.9))+ expand_limits(y=c(5.5,6.5))

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression")))+ scale_fill_discrete(name = "diagnosis", labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_boxerrorbar.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_boxerrorbar.jpg",width = 7, height = 5)
```

```{r}
#scatterplot with error bars
p1 <- ggplot(df, aes(x=df$`AX-40592209`, y=df$TC20001033.hg.1, group=as.factor(df$diagnosis), color =as.factor(df$diagnosis)))+
geom_line(aes(group=df$diagnosis))+
geom_errorbar(aes(ymin=TC20001033.hg.1-sd, ymax=TC20001033.hg.1+sd),width=.2)+ expand_limits(y=c(5,6))

p2 <-p1 + labs(y="PTK6 expression", x = "rs12480811 genotype", fill ="diagnosis")+scale_y_continuous(expression(paste(italic("PTK6")," " , "expression")))+ scale_color_discrete(name = "diagnosis", breaks = c("1","4"), labels = c("IS", "VRFC"))+scale_x_discrete( breaks = c("0", "1", "2"),labels = c("TT", "TC", "CC"))

ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_scattererrorbar.pdf",width = 7, height = 5)
ggsave(p2, file="../../Input_Output/Plot/cis-eQTL2_scattererrorbar.jpg",width = 7, height = 5)
```
