---
title: "eQTL_Stroke_MatrixEQTL_Two_Steps_030319"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Want to check the FDR with the method which is presented in Insight into Genotype-Phenotype Associations through eQTL Mapping in Multiple Cell Types in Health and Immun and the main paper could be found in https://academic.oup.com/aje/article/169/2/219/97415 and compare with another methods: https://academic.oup.com/aje/article/175/3/177/105981


```{r}
base.dir <- ("C:/eQTL/R_eQTL_Stroke/")
```

```{r}
#Outlier in genotype. call rate (CR), MAF & H.W.p.value
SNP_calls_351_table <- read.delim("C:/eQTL/SNP/SNP_calls_351_eQTL/SNP_calls_351_table.txt", row.names = 1) #628679 31, including MAF, CR, HW and SNP id for each SNPs
#filter based on other references
SNP_calls_351_table_filter <- SNP_calls_351_table[SNP_calls_351_table$MinorAlleleFrequency > 0.05 & SNP_calls_351_table$CR > 95 & SNP_calls_351_table$H.W.p.Value > 0.0000005,] #I think we need to filter based on H.W (sample size), #273688 30
SNP_calls_351_table_filter_id <- row.names(SNP_calls_351_table_filter) 
write.table(SNP_calls_351_table_filter_id, file = "../../Input_Output/SNP_calls_351_table_filter_id", col.names = T, sep = "\t")
#subset snps id after filtering from SNP.txt file
SNP <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/SNP.txt", row.names = 1)
SNP_after_filter <- SNP[SNP_calls_351_table_filter_id2,] 
write.table(SNP_after_filter, file = "../../Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", col.names = T, sep = "\t")
```

```{r}
SNP_file_name = paste(base.dir, "/Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", sep="")
#covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final.txt", sep="") #need sample as colnames
#covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_No_PC.txt", sep="")
gene_location_file_name = paste(base.dir, "/Input_Output/Gene_Location_Exclude_Weird_Position.txt", sep="")
SNP_Location <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/SNP_Location.txt", row.names=1) #include all SNPs location
SNP_location_filter <- SNP_Location[rownames(SNP_after_filter),] #273688 2
write.table(SNP_location_filter, file = "../../Input_Output/SNP_Location_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", col.names = T, sep = "\t")
snps_location_file_name = paste(base.dir, "/Input_Output/SNP_Location_CRmore95_MAFmore0.05_HWEmore0.0000005.txt", sep="") #remove two rows because of NA (no info from annotation file)
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_No_PC_NO_Subtype.txt", sep="")
#covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_No_PC_NO_Subtype_No_Diagnosis.txt", sep="")
```

```{r}
GE <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/GE.txt", row.names=1)
GE <- GE[rownames(Gene_Location_Exclude_Weird_Position),] #65988
write.table(GE, file = "../../Input_Output/GE_Exclude_Weird_Position.txt", sep = "\t")
expression_file_name = paste(base.dir, "/Input_Output/GE_Exclude_Weird_Position.txt", sep="") 
```

```{r}
output_file_name_cis = tempfile()
#output_file_name_trans = tempfile()
output_file_name = tempfile()
```


```{r}
#The p-value threshold determines which gene-SNP associations are saved in the output file output_file_name. Note that for larger datasets the threshold should be lower. Setting the threshold to a high value for a large dataset may cause excessively large output files.
#pvOutputThreshold_cis = 2e-2 #defualt
#pvOutputThreshold_tras = 1e-2 #defualt
pvOutputThreshold_cis =  1e-05
#pvOutputThreshold_trans = 1e-20  #Only associations significant at this level will be saved. 
#pvOutputThreshold_trans = 1e-10
```

```{r}
errorCovariance = numeric()
```

```{r}
#cisDist = 1e6 #keep this value as a cis out of this distance conder as a trans
```

```{r}
## Load genotype data
snps = SlicedData$new() #SlicedData object with genotype information.
snps$fileDelimiter = "\t"      # the TAB character
snps$fileOmitCharacters = "NA" # denote missing values;
snps$fileSkipRows = 1          # one row of column labels
snps$fileSkipColumns = 1       # one column of row labels
snps$fileSliceSize = 2000    # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name) #Number of columns: 351, Number of rows: 273688 

#Rows read:  273688  done. This is correct, we have  273688  SNPs id
```

```{r}
# Load gene expression data
gene = SlicedData$new(); #SlicedData object with gene expression information. Must have columns matching those of snps, create gene including row and column
gene$fileDelimiter = "\t"      # the TAB characte
gene$fileOmitCharacters = "NA" # denote missing values;
gene$fileSkipRows = 1          # one row of column labels
gene$fileSkipColumns = 1       # one column of row labels
gene$fileSliceSize = 2000      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name) #Number of columns: 351, Number of rows: 65988

#Rows read: 65988 done. This is correct, we have 65988 genes
```

```{r}
# Load covariates
cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t"      # the TAB character
cvrt$fileOmitCharacters = "NA" # denote missing values;
cvrt$fileSkipRows = 1         # one row of column labels
cvrt$fileSkipColumns = 1      # one column of row labels
if(length(covariates_file_name)>0) {
cvrt$LoadFile(covariates_file_name)
}
#Rows read: 4 done. correct
```

```{r}
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE) #dim  65988   4
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE) #dim 273686 3
```

```{r}
#save
save(snpspos,genepos, file = "../../Input_Output/snpspos_genepos_filter.Rdata")
```

```{r}
#The first main Matrix eQTL function is called:
useModel =  modelLINEAR # modelANOVA or modelLINEAR or modelLINEAR_CROSS
# Set parameter useModel = modelLINEAR in the call of Matrix_eQTL_main to indicate that the effect of genotype on expression should be assumed to be additive linear.
# SNP-gene pairs passing the step 1 threshold α1 were eligible for step 2 testing. We used a p-value of 5×10-5 for α1.
# A more liberal α1 allows more SNP-gene pairs through to step 2, but at the price of more multiple testing and a more stringent threshold in step 2.
eQTL_step1 = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name ="./eQTL_step1_test3",
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, #logical. Set to TRUE to display more detailed report about the progress
pvOutputThreshold = pvOutputThreshold,
snpspos = snpspos, 
genepos = genepos,
#cisDist = cisDist,
pvalue.hist = "qqplot", #logical, numerical, or "qqplot".Defines whether and how the distribution of p-values is recorded in the returned object.If pvalue.hist = FALSE, the information is not recorded and the analysis is performed faster. Alternatively, set pvalue.hist = "qqplot" to record information sufficient to create a QQ-plot of the p-values (use plot on the returned object to create the plot).To record information for a histogram set pvalue.hist to the desired number of bins of equal size.	Finally, pvalue.hist can also be set to a custom set of bin edges.
min.pv.by.genesnp = FALSE, #logical. Set min.pv.by.genesnp = TRUE to record the minimum p-value for each SNP and each gene in the returned object. The minimum p-values are recorded even if if they are above the corresponding thresholds of pvOutputThreshold and pvOutputThreshold.cis. The analysis runs faster when the parameter is set to FALSE.
noFDRsaveMemory = FALSE) 
#got 1,731,587 eQTLs
```

```{r}
#prepare input for step2
#save the eQTL_step1 as a table then export the snpid and gened for next step
geneid_uniq <- read.table("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/geneid_uniq.txt", row.names=1, quote="\"", comment.char="") #65944
snpid_uniq <- read.table("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/snpid_uniq.txt", row.names=1, quote="\"", comment.char="", col.names = TRUE) #169733
```

```{r}
#To reduce more #SNP (169733) from step1, we want to try linear model same as first step just for cis-eQT
#prepare SNP.txt for this step
#SNP <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/SNP.txt", row.names = 1)
#SNP_after_filter <- SNP[SNP_calls_351_table_filter_id,] 
SNP_after_filter_step1 <- SNP_after_filter[rownames(snpid_uniq),] # 169733    351
write.table(SNP_after_filter_step1, file = "../../Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.txt", col.names = T, sep = "\t")
SNP_file_name = paste(base.dir, "/Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.txt", sep="")
#SNP_location_filter <- SNP_Location[rownames(SNP_after_filter),] # 273688 
SNP_location_filter_step1 <- SNP_location_filter[rownames(snpid_uniq),] #169733    2
write.table(SNP_location_filter_step1, file = "../../Input_Output/SNP_Location_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.txt", col.names = T, sep = "\t")
snps_location_file_name = paste(base.dir, "/Input_Output/SNP_Location_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.txt", sep="")
expression_file_name = paste(base.dir, "/Input_Output/GE_Exclude_Weird_Position.txt", sep="") 
gene_location_file_name = paste(base.dir, "/Input_Output/Gene_Location_Exclude_Weird_Position.txt", sep="")
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE) #dim  65988   4
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE)
output_file_name = tempfile()
```

```{r}
pvOutputThreshold_cis = 1e-5
cisDist = 1e6
```

```{r}
#load genotype data, expression data and covariates data
```

```{r}
#The first main Matrix eQTL function is called:
useModel =  modelLINEAR 

eQTL_step1.a = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = output_file_name,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, #logical. Set to TRUE to display more detailed report about the progress
pvOutputThreshold = pvOutputThreshold_cis,
snpspos = snpspos, 
genepos = genepos,
cisDist = cisDist,
pvalue.hist = "qqplot", #logical, numerical, or "qqplot".Defines whether and how the distribution of p-values is recorded in the returned object.If pvalue.hist = FALSE, the information is not recorded and the analysis is performed faster. Alternatively, set pvalue.hist = "qqplot" to record information sufficient to create a QQ-plot of the p-values (use plot on the returned object to create the plot).To record information for a histogram set pvalue.hist to the desired number of bins of equal size.	Finally, pvalue.hist can also be set to a custom set of bin edges.
min.pv.by.genesnp = FALSE, #logical. Set min.pv.by.genesnp = TRUE to record the minimum p-value for each SNP and each gene in the returned object. The minimum p-values are recorded even if if they are above the corresponding thresholds of pvOutputThreshold and pvOutputThreshold.cis. The analysis runs faster when the parameter is set to FALSE.
noFDRsaveMemory = FALSE)
```

```{r}
snpid_step1.a_uniq <- read.table("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/snpid_step1.a_uniq.txt", row.names=1, quote="\"", comment.char="")#87927

```

```{r}
SNP_after_filter_step1.a <- SNP_after_filter[rownames(snpid_step1.a_uniq),] #87927   351
write.table(SNP_after_filter_step1.a, file = "../../Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.a.txt", col.names = T, sep = "\t")
SNP_file_name = paste(base.dir, "/Input_Output/SNP_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.a.txt", sep="")
#SNP_location_filter <- SNP_Location[rownames(SNP_after_filter),] # 273688 
SNP_location_filter_step1.a <- SNP_location_filter[rownames(snpid_step1.a_uniq),] #87927    2
write.table(SNP_location_filter_step1.a, file = "../../Input_Output/SNP_Location_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.a.txt", col.names = T, sep = "\t")
snps_location_file_name = paste(base.dir, "/Input_Output/SNP_Location_CRmore95_MAFmore0.05_HWEmore0.0000005_step1.a.txt", sep="")
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE) #87927
```

```{r}
#load genotype data, expression data
```

```{r}
#load covariates data with cinsidering diagnosis row as a last row of covariate matrix
covariates_file_name = paste(base.dir, "/Input_Output/Covariates_Final_No_PC_NO_Subtype.txt", sep="")
```


```{r}
#load genotype data from step1_test2 file

eQTL_step1_test2 <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/eQTL_step1_test3") #17264 only cis-eQTL pvOutputThreshold_cis = 1e-5 & cisDist =  1e+06
eQTL_step1_test3_uniq <- unique(eQTL_step1_test3$SNP) #9162
SNP_step1_test3_uniq<- SNP_after_filter[eQTL_step1_test3_uniq,] #9162 351
SNP_step1_test3_uniq_sort <- SNP_step1_test3_uniq[sort(rownames(SNP_step1_test3_uniq)),]
write.table(SNP_step1_test3_uniq_sort, file = "../../Input_Output/SNP_step1_test3_uniq_sort.txt", col.names = T, sep = "\t")
SNP_file_name = paste(base.dir, "/Input_Output/SNP_step1_test3_uniq_sort.txt", sep="")
```

```{r}
#The second main Matrix eQTL function is called:
useModel =  modelLINEAR_CROSS 

eQTL_step2 = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = "./eQTL_step2_test16_trans",
output_file_name.cis = "./eQTL_step2_test16",
pvOutputThreshold = 1e-10,
useModel =  modelLINEAR_CROSS, 
errorCovariance = errorCovariance, 
verbose = TRUE, #logical. Set to TRUE to display more detailed report about the progress
pvOutputThreshold.cis = 1e-5,
snpspos = snpspos, 
genepos = genepos,
cisDist = 5e+07,
pvalue.hist = "qqplot", #logical, numerical, or "qqplot".Defines whether and how the distribution of p-values is recorded in the returned object.If pvalue.hist = FALSE, the information is not recorded and the analysis is performed faster. Alternatively, set pvalue.hist = "qqplot" to record information sufficient to create a QQ-plot of the p-values (use plot on the returned object to create the plot).To record information for a histogram set pvalue.hist to the desired number of bins of equal size.	Finally, pvalue.hist can also be set to a custom set of bin edges.
min.pv.by.genesnp = FALSE, #logical. Set min.pv.by.genesnp = TRUE to record the minimum p-value for each SNP and each gene in the returned object. The minimum p-values are recorded even if if they are above the corresponding thresholds of pvOutputThreshold and pvOutputThreshold.cis. The analysis runs faster when the parameter is set to FALSE.
noFDRsaveMemory = FALSE)
```



```{r}
#merge two dataframe to add more info. on the final result (eQTL_step*), run this for all the final result
eQTL_step2_file <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/eQTL_step2_file")
gene_expression_annotation_351.SST.RMA.GENE.FULL...Group.1 <- read.delim("C:/eQTL/GENE/Picked_Files/gene_expression_351/gene_expression_annotation_351.SST-RMA-GENE-FULL - Group 1.TXT")
gene_expression_annotation_small <-gene_expression_annotation_351.SST.RMA.GENE.FULL...Group.1[c(1:67527), c(1,353,358,360,361,365)] #70523 6
colnames(gene_expression_annotation_small) <- c("gene", "Gene.Symbol" , "Chromosome" ,"Start" , "Stop", "locus.type")
eQTL_step2_merge <- merge(eQTL_step2_file,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #88671 

Axiom_BioBank1.na35.annot_v2 <- read.csv("C:/eQTL/SNP/SNP_calls_351_eQTL/Axiom_BioBank1.na35.annot_v2.csv")
SNP_calls_small <- Axiom_BioBank1.na35.annot_v2[,c(1,3,5,6,12,13,14,15)] #718212 8
colnames(SNP_calls_small)[1] <- "SNP"
eQTL_step2_merge_v2 <- merge(eQTL_step2_merge,SNP_calls_small, by = "SNP", all.x = TRUE, sort = TRUE) #88671
```

```{r}
eQTL_step2a_file <- read.delim("C:/eQTL/R_eQTL_Stroke/Input_Output/eQTL_result/eQTL_step2a_file")

eQTL_step2a_merge <- merge(eQTL_step2a_file,gene_expression_annotation_small,by ="gene", all.x = TRUE, sort =TRUE) #88671 

eQTL_step2a_merge_v2 <- merge(eQTL_step2a_merge,SNP_calls_small, by = "SNP", all.x = TRUE, sort = TRUE) #88671
```


```{r}
#filter genes listby removing non-coding genes
gene_expression_annotation_351.SST.RMA.GENE.Coding <- read.delim("C:/eQTL/GENE/Picked_Files/gene_expression_351/gene_expression_annotation_351.SST-RMA-GENE-Coding.txt", header = TRUE, row.names=1) #44699 351
#View(gene_expression_annotation_351.SST.RMA.GENE.Coding)
#colnames(gene_expression_annotation_351.SST.RMA.GENE.Coding) <- sub(x = colnames(gene_expression_annotation_351.SST.RMA.GENE.Coding), pattern = "_HTA*.*", replacement ="")
write.table(gene_expression_annotation_351.SST.RMA.GENE.Coding, file = "../../Input_Output/GE_coding.txt", col.names = T, sep = "\t")
#load all input files
expression_file_name = paste(base.dir, "/Input_Output/GE_coding.txt", sep="")
```

```{r}
#run LINEAR_CROSS model directly.

#output_file_name_cis = tempfile()
#output_file_name_tra = tempfile()

useModel =  modelLINEAR_CROSS
cisDist = 1e6
eQTL = Matrix_eQTL_main(
snps = snps, 
gene = gene, 
cvrt = cvrt,
output_file_name = output_file_name_tra,
pvOutputThreshold = 1e-10,
useModel = useModel, 
errorCovariance = errorCovariance, 
verbose = TRUE, 
output_file_name.cis = output_file_name_cis,
pvOutputThreshold.cis = 1e-5,
snpspos = snpspos, 
genepos = genepos,
cisDist = cisDist,
pvalue.hist = "qqplot",
min.pv.by.genesnp = FALSE,
noFDRsaveMemory = FALSE)

unlink(output_file_name_tra);
unlink(output_file_name_cis);
```

```{r}
cat('Analysis done in: ', eQTL$time.in.sec, ' seconds', '\n');
cat('Detected local eQTLs:', '\n');
show(eQTL$cis$eqtls)
cat('Detected distant eQTLs:', '\n');
show(eQTL$trans$eqtls)
```

```{r}
plot(eQTL)

png("../../Input_Output/eQTL_result/plot_eQTL_cis_dist_1e7.png")
plot(eQTL)
dev.off()
```

```{r}
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/sex/eQTL_cis_dist_1e6.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/sex/eQTL_trans_1e6.csv")
```

```{r}
#I want to try for all genes 
expression_file_name = paste(base.dir, "/Input_Output/GE_Exclude_Weird_Position.txt", sep="") 
```

```{r}
#filter genes list by kepping only sex genes
gene_expression_annotation_351.SST.RMA.GENE.Sex_Chromosome <- read.delim("C:/eQTL/GENE/Picked_Files/gene_expression_351/gene_expression_annotation_351.SST-RMA-GENE-Sex_Chromosome.txt", row.names=1) #2686 351
colnames(gene_expression_annotation_351.SST.RMA.GENE.Sex_Chromosome) <- sub(x = colnames(gene_expression_annotation_351.SST.RMA.GENE.Sex_Chromosome), pattern = "_HTA*.*", replacement ="")
write.table(gene_expression_annotation_351.SST.RMA.GENE.Sex_Chromosome, file = "../../Input_Output/GE_sex.txt", col.names = T, sep = "\t")
#load all input files
expression_file_name = paste(base.dir, "/Input_Output/GE_sex.txt", sep="")
#load gene file and run Matrix_eQTL_main function

#save the result
write.csv(eQTL$cis$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/sex/eQTL_cis_dist_1e7.csv")
write.csv(eQTL$trans$eqtls, "../../Input_Output/eQTL_result/LINEAR_CROSS/sex/eQTL_trans_1e7.csv")
```


